// pskov/ver-nodejs/pskov-ver-nodejs.js
(function (factory) {
  if (typeof define === 'function' && define.amd)
    define(['exports', './kotlin-kotlin-stdlib.js'], factory);
  else if (typeof exports === 'object')
    factory(module.exports, require('./kotlin-kotlin-stdlib.js'));
  else {
    if (typeof globalThis['kotlin-kotlin-stdlib'] === 'undefined') {
      throw new Error("Error loading module 'pskov:ver-nodejs'. Its dependency 'kotlin-kotlin-stdlib' was not found. Please, check whether 'kotlin-kotlin-stdlib' is loaded prior to 'pskov:ver-nodejs'.");
    }
    globalThis['pskov:ver-nodejs'] = factory(typeof globalThis['pskov:ver-nodejs'] === 'undefined' ? {} : globalThis['pskov:ver-nodejs'], globalThis['kotlin-kotlin-stdlib']);
  }
}(function (_, kotlin_kotlin) {
  'use strict';
  //region block: imports
  var imul = Math.imul;
  var protoOf = kotlin_kotlin.$_$.t;
  var initMetadataForInterface = kotlin_kotlin.$_$.q;
  var Unit_getInstance = kotlin_kotlin.$_$.e;
  var ArrayList_init_$Create$ = kotlin_kotlin.$_$.b;
  var removeFirst = kotlin_kotlin.$_$.i;
  var initMetadataForClass = kotlin_kotlin.$_$.p;
  var defineProp = kotlin_kotlin.$_$.k;
  var VOID = kotlin_kotlin.$_$.a;
  var emptyMap = kotlin_kotlin.$_$.g;
  var THROW_CCE = kotlin_kotlin.$_$.d1;
  var KtMap = kotlin_kotlin.$_$.f;
  var isInterface = kotlin_kotlin.$_$.s;
  var isArray = kotlin_kotlin.$_$.r;
  var toString = kotlin_kotlin.$_$.u;
  var getStringHashCode = kotlin_kotlin.$_$.n;
  var hashCode = kotlin_kotlin.$_$.o;
  var getBooleanHashCode = kotlin_kotlin.$_$.m;
  var equals = kotlin_kotlin.$_$.l;
  var substring = kotlin_kotlin.$_$.a1;
  var startsWith = kotlin_kotlin.$_$.z;
  var take = kotlin_kotlin.$_$.c1;
  var charSequenceLength = kotlin_kotlin.$_$.j;
  var ensureNotNull = kotlin_kotlin.$_$.e1;
  var joinToString = kotlin_kotlin.$_$.h;
  var LinkedHashMap_init_$Create$ = kotlin_kotlin.$_$.c;
  var split = kotlin_kotlin.$_$.y;
  var Default_getInstance = kotlin_kotlin.$_$.d;
  var decodeToString = kotlin_kotlin.$_$.v;
  var substring_0 = kotlin_kotlin.$_$.b1;
  var endsWith = kotlin_kotlin.$_$.x;
  var encodeToByteArray = kotlin_kotlin.$_$.w;
  //endregion
  //region block: pre-declaration
  function fieldAny(name) {
    return this.field(name);
  }
  initMetadataForInterface(CLDContext, 'CLDContext');
  initMetadataForClass(CLDController, 'CLDController');
  initMetadataForClass(AppContext, 'AppContext', AppContext, VOID, [CLDContext]);
  initMetadataForClass(FSFile, 'FSFile', FSFile);
  initMetadataForClass(NetRequest, 'NetRequest', NetRequest);
  initMetadataForClass(NetResponse, 'NetResponse', NetResponse);
  initMetadataForClass(SrvContext, 'SrvContext', SrvContext, VOID, [CLDContext]);
  //endregion
  function CLDContext() {
  }
  function CLDController$registerFieldCallback$lambda($fieldName, $cb) {
    return function (c) {
      var tmp;
      if (c.recentField === $fieldName) {
        tmp = $cb(c);
      }
      return Unit_getInstance();
    };
  }
  function CLDController(context) {
    this.context = context;
    var tmp = this;
    // Inline function 'kotlin.collections.mutableListOf' call
    tmp.callbacks_1 = ArrayList_init_$Create$();
    var tmp_0 = this;
    // Inline function 'kotlin.collections.mutableListOf' call
    tmp_0.functions_1 = ArrayList_init_$Create$();
    this.isProcessingQueue = false;
    var tmp_1 = this;
    // Inline function 'kotlin.collections.mutableListOf' call
    tmp_1.queue_1 = ArrayList_init_$Create$();
  }
  protoOf(CLDController).set_context_nlv17e_k$ = function (_set____db54di) {
    this.context = _set____db54di;
  };
  protoOf(CLDController).get_context_h02k06_k$ = function () {
    return this.context;
  };
  protoOf(CLDController).set_callbacks_8gq60i_k$ = function (_set____db54di) {
    this.callbacks_1 = _set____db54di;
  };
  protoOf(CLDController).get_callbacks_9fvbjy_k$ = function () {
    return this.callbacks_1;
  };
  protoOf(CLDController).set_functions_r1h97z_k$ = function (_set____db54di) {
    this.functions_1 = _set____db54di;
  };
  protoOf(CLDController).get_functions_xrpcc5_k$ = function () {
    return this.functions_1;
  };
  protoOf(CLDController).set_isProcessingQueue_ex9c7y_k$ = function (_set____db54di) {
    this.isProcessingQueue = _set____db54di;
  };
  protoOf(CLDController).get_isProcessingQueue_dg022z_k$ = function () {
    return this.isProcessingQueue;
  };
  protoOf(CLDController).set_queue_pqz836_k$ = function (_set____db54di) {
    this.queue_1 = _set____db54di;
  };
  protoOf(CLDController).get_queue_ecg85b_k$ = function () {
    return this.queue_1;
  };
  protoOf(CLDController).executeFunctions = function () {
    var c = removeFirst(this.queue_1);
    this.context.set_recentField_5jv888_k$(c.recentField);
    this.context.setField(c.recentField, c.fieldAny(c.recentField));
    var _iterator__ex2g4s = this.functions_1.iterator_jk1svi_k$();
    while (_iterator__ex2g4s.hasNext_bitz1p_k$()) {
      var f = _iterator__ex2g4s.next_20eer_k$();
      var ctx = f(this.context.selfCopy());
      if (!(ctx.recentField === 'none')) {
        this.queue_1.add_utx5q5_k$(ctx);
      }
    }
    this.reportContext();
  };
  protoOf(CLDController).processQueue = function () {
    if (this.isProcessingQueue) {
      return Unit_getInstance();
    }
    this.isProcessingQueue = true;
    while (this.queue_1.get_size_woubt6_k$() > 0) {
      this.executeFunctions();
    }
    this.isProcessingQueue = false;
  };
  protoOf(CLDController).registerCallback = function (cb) {
    this.callbacks_1.add_utx5q5_k$(cb);
  };
  protoOf(CLDController).registerFieldCallback = function (fieldName, cb) {
    var tmp = this.callbacks_1;
    tmp.add_utx5q5_k$(CLDController$registerFieldCallback$lambda(fieldName, cb));
  };
  protoOf(CLDController).registerFunction = function (f) {
    this.functions_1.add_utx5q5_k$(f);
  };
  protoOf(CLDController).reportContext = function () {
    var _iterator__ex2g4s = this.callbacks_1.iterator_jk1svi_k$();
    while (_iterator__ex2g4s.hasNext_bitz1p_k$()) {
      var cb = _iterator__ex2g4s.next_20eer_k$();
      cb(this.context);
    }
  };
  protoOf(CLDController).set = function (fieldName, value) {
    var c = this.context.selfCopy();
    c.setField(fieldName, value);
    c.set_recentField_5jv888_k$(fieldName);
    this.queue_1.add_utx5q5_k$(c);
    this.processQueue();
  };
  function get_CONST_API_LIST() {
    return CONST_API_LIST;
  }
  var CONST_API_LIST;
  function get_CONST_API_PATH() {
    return CONST_API_PATH;
  }
  var CONST_API_PATH;
  function get_CONST_API_READ() {
    return CONST_API_READ;
  }
  var CONST_API_READ;
  function get_CONST_API_WRITE() {
    return CONST_API_WRITE;
  }
  var CONST_API_WRITE;
  function get_CONST_EXT_MD() {
    return CONST_EXT_MD;
  }
  var CONST_EXT_MD;
  function get_CONST_GET() {
    return CONST_GET;
  }
  var CONST_GET;
  function get_CONST_POST() {
    return CONST_POST;
  }
  var CONST_POST;
  function AppContext(baseURL, cfg, didClickEditorTab, didClickFilesTab, didClickRenderTab, didClickSaveBtn, didLaunch, didResize, didSaveEditedFiles, didSaveFile, editedContents, editedFileContents, editorContents, header, inputDirFiles, inputDirs, inputMDFiles, installEditor, listInputDirId, projectPath, readFile, readFileContents, request, resizeEditor, response, responseError, saveFileId, saveFiles, selectedFileId, selectedFileName, selectedTabId, splashTimeout, recentField) {
    baseURL = baseURL === VOID ? '' : baseURL;
    var tmp;
    if (cfg === VOID) {
      // Inline function 'kotlin.collections.mapOf' call
      tmp = emptyMap();
    } else {
      tmp = cfg;
    }
    cfg = tmp;
    didClickEditorTab = didClickEditorTab === VOID ? false : didClickEditorTab;
    didClickFilesTab = didClickFilesTab === VOID ? false : didClickFilesTab;
    didClickRenderTab = didClickRenderTab === VOID ? false : didClickRenderTab;
    didClickSaveBtn = didClickSaveBtn === VOID ? false : didClickSaveBtn;
    didLaunch = didLaunch === VOID ? false : didLaunch;
    didResize = didResize === VOID ? false : didResize;
    didSaveEditedFiles = didSaveEditedFiles === VOID ? false : didSaveEditedFiles;
    didSaveFile = didSaveFile === VOID ? false : didSaveFile;
    editedContents = editedContents === VOID ? '' : editedContents;
    var tmp_0;
    if (editedFileContents === VOID) {
      // Inline function 'kotlin.collections.mapOf' call
      tmp_0 = emptyMap();
    } else {
      tmp_0 = editedFileContents;
    }
    editedFileContents = tmp_0;
    editorContents = editorContents === VOID ? '' : editorContents;
    var tmp_1;
    if (header === VOID) {
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp_1 = [];
    } else {
      tmp_1 = header;
    }
    header = tmp_1;
    var tmp_2;
    if (inputDirFiles === VOID) {
      // Inline function 'kotlin.collections.mapOf' call
      tmp_2 = emptyMap();
    } else {
      tmp_2 = inputDirFiles;
    }
    inputDirFiles = tmp_2;
    var tmp_3;
    if (inputDirs === VOID) {
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp_3 = [];
    } else {
      tmp_3 = inputDirs;
    }
    inputDirs = tmp_3;
    var tmp_4;
    if (inputMDFiles === VOID) {
      // Inline function 'kotlin.collections.mapOf' call
      tmp_4 = emptyMap();
    } else {
      tmp_4 = inputMDFiles;
    }
    inputMDFiles = tmp_4;
    installEditor = installEditor === VOID ? false : installEditor;
    listInputDirId = listInputDirId === VOID ? 0 : listInputDirId;
    projectPath = projectPath === VOID ? '' : projectPath;
    readFile = readFile === VOID ? '' : readFile;
    readFileContents = readFileContents === VOID ? '' : readFileContents;
    request = request === VOID ? new NetRequest() : request;
    resizeEditor = resizeEditor === VOID ? false : resizeEditor;
    response = response === VOID ? new NetResponse() : response;
    responseError = responseError === VOID ? new NetResponse() : responseError;
    saveFileId = saveFileId === VOID ? 0 : saveFileId;
    var tmp_5;
    if (saveFiles === VOID) {
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp_5 = [];
    } else {
      tmp_5 = saveFiles;
    }
    saveFiles = tmp_5;
    var tmp_6;
    if (selectedFileId === VOID) {
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp_6 = [];
    } else {
      tmp_6 = selectedFileId;
    }
    selectedFileId = tmp_6;
    selectedFileName = selectedFileName === VOID ? '' : selectedFileName;
    selectedTabId = selectedTabId === VOID ? 0 : selectedTabId;
    splashTimeout = splashTimeout === VOID ? 0 : splashTimeout;
    recentField = recentField === VOID ? '' : recentField;
    this.baseURL = baseURL;
    this.cfg = cfg;
    this.didClickEditorTab = didClickEditorTab;
    this.didClickFilesTab = didClickFilesTab;
    this.didClickRenderTab = didClickRenderTab;
    this.didClickSaveBtn = didClickSaveBtn;
    this.didLaunch = didLaunch;
    this.didResize = didResize;
    this.didSaveEditedFiles = didSaveEditedFiles;
    this.didSaveFile = didSaveFile;
    this.editedContents = editedContents;
    this.editedFileContents = editedFileContents;
    this.editorContents = editorContents;
    this.header = header;
    this.inputDirFiles = inputDirFiles;
    this.inputDirs = inputDirs;
    this.inputMDFiles = inputMDFiles;
    this.installEditor = installEditor;
    this.listInputDirId = listInputDirId;
    this.projectPath = projectPath;
    this.readFile = readFile;
    this.readFileContents = readFileContents;
    this.request = request;
    this.resizeEditor = resizeEditor;
    this.response = response;
    this.responseError = responseError;
    this.saveFileId = saveFileId;
    this.saveFiles = saveFiles;
    this.selectedFileId = selectedFileId;
    this.selectedFileName = selectedFileName;
    this.selectedTabId = selectedTabId;
    this.splashTimeout = splashTimeout;
    this.recentField_1 = recentField;
  }
  protoOf(AppContext).set_baseURL_590weh_k$ = function (_set____db54di) {
    this.baseURL = _set____db54di;
  };
  protoOf(AppContext).get_baseURL_48hedn_k$ = function () {
    return this.baseURL;
  };
  protoOf(AppContext).set_cfg_i3p1hn_k$ = function (_set____db54di) {
    this.cfg = _set____db54di;
  };
  protoOf(AppContext).get_cfg_18j85h_k$ = function () {
    return this.cfg;
  };
  protoOf(AppContext).set_didClickEditorTab_wrusx9_k$ = function (_set____db54di) {
    this.didClickEditorTab = _set____db54di;
  };
  protoOf(AppContext).get_didClickEditorTab_nlhywm_k$ = function () {
    return this.didClickEditorTab;
  };
  protoOf(AppContext).set_didClickFilesTab_2vlx4f_k$ = function (_set____db54di) {
    this.didClickFilesTab = _set____db54di;
  };
  protoOf(AppContext).get_didClickFilesTab_v0t5io_k$ = function () {
    return this.didClickFilesTab;
  };
  protoOf(AppContext).set_didClickRenderTab_fd3j9g_k$ = function (_set____db54di) {
    this.didClickRenderTab = _set____db54di;
  };
  protoOf(AppContext).get_didClickRenderTab_dlvwh9_k$ = function () {
    return this.didClickRenderTab;
  };
  protoOf(AppContext).set_didClickSaveBtn_p90pc4_k$ = function (_set____db54di) {
    this.didClickSaveBtn = _set____db54di;
  };
  protoOf(AppContext).get_didClickSaveBtn_55q9m5_k$ = function () {
    return this.didClickSaveBtn;
  };
  protoOf(AppContext).set_didLaunch_x1bx8g_k$ = function (_set____db54di) {
    this.didLaunch = _set____db54di;
  };
  protoOf(AppContext).get_didLaunch_bo02hj_k$ = function () {
    return this.didLaunch;
  };
  protoOf(AppContext).set_didResize_ls5fdr_k$ = function (_set____db54di) {
    this.didResize = _set____db54di;
  };
  protoOf(AppContext).get_didResize_8rkivq_k$ = function () {
    return this.didResize;
  };
  protoOf(AppContext).set_didSaveEditedFiles_v88nc6_k$ = function (_set____db54di) {
    this.didSaveEditedFiles = _set____db54di;
  };
  protoOf(AppContext).get_didSaveEditedFiles_2rh7h7_k$ = function () {
    return this.didSaveEditedFiles;
  };
  protoOf(AppContext).set_didSaveFile_msus2u_k$ = function (_set____db54di) {
    this.didSaveFile = _set____db54di;
  };
  protoOf(AppContext).get_didSaveFile_5khjj_k$ = function () {
    return this.didSaveFile;
  };
  protoOf(AppContext).set_editedContents_ez3tu2_k$ = function (_set____db54di) {
    this.editedContents = _set____db54di;
  };
  protoOf(AppContext).get_editedContents_vv69p0_k$ = function () {
    return this.editedContents;
  };
  protoOf(AppContext).set_editedFileContents_15uwio_k$ = function (_set____db54di) {
    this.editedFileContents = _set____db54di;
  };
  protoOf(AppContext).get_editedFileContents_v36xa0_k$ = function () {
    return this.editedFileContents;
  };
  protoOf(AppContext).set_editorContents_9ncjk6_k$ = function (_set____db54di) {
    this.editorContents = _set____db54di;
  };
  protoOf(AppContext).get_editorContents_ed2568_k$ = function () {
    return this.editorContents;
  };
  protoOf(AppContext).set_header_3bn6oo_k$ = function (_set____db54di) {
    this.header = _set____db54di;
  };
  protoOf(AppContext).get_header_e7o2vq_k$ = function () {
    return this.header;
  };
  protoOf(AppContext).set_inputDirFiles_aq0wu_k$ = function (_set____db54di) {
    this.inputDirFiles = _set____db54di;
  };
  protoOf(AppContext).get_inputDirFiles_av0giz_k$ = function () {
    return this.inputDirFiles;
  };
  protoOf(AppContext).set_inputDirs_grpwsx_k$ = function (_set____db54di) {
    this.inputDirs = _set____db54di;
  };
  protoOf(AppContext).get_inputDirs_yalcmx_k$ = function () {
    return this.inputDirs;
  };
  protoOf(AppContext).set_inputMDFiles_yfefbd_k$ = function (_set____db54di) {
    this.inputMDFiles = _set____db54di;
  };
  protoOf(AppContext).get_inputMDFiles_kbyiqn_k$ = function () {
    return this.inputMDFiles;
  };
  protoOf(AppContext).set_installEditor_maric6_k$ = function (_set____db54di) {
    this.installEditor = _set____db54di;
  };
  protoOf(AppContext).get_installEditor_zalolb_k$ = function () {
    return this.installEditor;
  };
  protoOf(AppContext).set_listInputDirId_5r92k1_k$ = function (_set____db54di) {
    this.listInputDirId = _set____db54di;
  };
  protoOf(AppContext).get_listInputDirId_kijjmt_k$ = function () {
    return this.listInputDirId;
  };
  protoOf(AppContext).set_projectPath_c6iemh_k$ = function (_set____db54di) {
    this.projectPath = _set____db54di;
  };
  protoOf(AppContext).get_projectPath_1nfxxn_k$ = function () {
    return this.projectPath;
  };
  protoOf(AppContext).set_readFile_lgagpx_k$ = function (_set____db54di) {
    this.readFile = _set____db54di;
  };
  protoOf(AppContext).get_readFile_oveznf_k$ = function () {
    return this.readFile;
  };
  protoOf(AppContext).set_readFileContents_6a4sbl_k$ = function (_set____db54di) {
    this.readFileContents = _set____db54di;
  };
  protoOf(AppContext).get_readFileContents_s4pvpn_k$ = function () {
    return this.readFileContents;
  };
  protoOf(AppContext).set_request_qynj8c_k$ = function (_set____db54di) {
    this.request = _set____db54di;
  };
  protoOf(AppContext).get_request_jdwg4m_k$ = function () {
    return this.request;
  };
  protoOf(AppContext).set_resizeEditor_4fjall_k$ = function (_set____db54di) {
    this.resizeEditor = _set____db54di;
  };
  protoOf(AppContext).get_resizeEditor_pfef06_k$ = function () {
    return this.resizeEditor;
  };
  protoOf(AppContext).set_response_p6qtoa_k$ = function (_set____db54di) {
    this.response = _set____db54di;
  };
  protoOf(AppContext).get_response_xlk07e_k$ = function () {
    return this.response;
  };
  protoOf(AppContext).set_responseError_vudjwu_k$ = function (_set____db54di) {
    this.responseError = _set____db54di;
  };
  protoOf(AppContext).get_responseError_q48d9a_k$ = function () {
    return this.responseError;
  };
  protoOf(AppContext).set_saveFileId_51ckbb_k$ = function (_set____db54di) {
    this.saveFileId = _set____db54di;
  };
  protoOf(AppContext).get_saveFileId_8syyn7_k$ = function () {
    return this.saveFileId;
  };
  protoOf(AppContext).set_saveFiles_gdm605_k$ = function (_set____db54di) {
    this.saveFiles = _set____db54di;
  };
  protoOf(AppContext).get_saveFiles_b67nm9_k$ = function () {
    return this.saveFiles;
  };
  protoOf(AppContext).set_selectedFileId_e5kq15_k$ = function (_set____db54di) {
    this.selectedFileId = _set____db54di;
  };
  protoOf(AppContext).get_selectedFileId_d81txh_k$ = function () {
    return this.selectedFileId;
  };
  protoOf(AppContext).set_selectedFileName_oda0iz_k$ = function (_set____db54di) {
    this.selectedFileName = _set____db54di;
  };
  protoOf(AppContext).get_selectedFileName_6md2gb_k$ = function () {
    return this.selectedFileName;
  };
  protoOf(AppContext).set_selectedTabId_5uxjow_k$ = function (_set____db54di) {
    this.selectedTabId = _set____db54di;
  };
  protoOf(AppContext).get_selectedTabId_mp2gek_k$ = function () {
    return this.selectedTabId;
  };
  protoOf(AppContext).set_splashTimeout_qup791_k$ = function (_set____db54di) {
    this.splashTimeout = _set____db54di;
  };
  protoOf(AppContext).get_splashTimeout_haycz3_k$ = function () {
    return this.splashTimeout;
  };
  protoOf(AppContext).set_recentField_5jv888_k$ = function (_set____db54di) {
    this.recentField_1 = _set____db54di;
  };
  protoOf(AppContext).get_recentField_jy0xl6_k$ = function () {
    return this.recentField_1;
  };
  protoOf(AppContext).field = function (name) {
    switch (name) {
      case 'baseURL':
        var tmp = this.baseURL;
        return !(tmp == null) ? tmp : THROW_CCE();
      case 'cfg':
        var tmp_0 = this.cfg;
        return !(tmp_0 == null) ? tmp_0 : THROW_CCE();
      case 'didClickEditorTab':
        var tmp_1 = this.didClickEditorTab;
        return !(tmp_1 == null) ? tmp_1 : THROW_CCE();
      case 'didClickFilesTab':
        var tmp_2 = this.didClickFilesTab;
        return !(tmp_2 == null) ? tmp_2 : THROW_CCE();
      case 'didClickRenderTab':
        var tmp_3 = this.didClickRenderTab;
        return !(tmp_3 == null) ? tmp_3 : THROW_CCE();
      case 'didClickSaveBtn':
        var tmp_4 = this.didClickSaveBtn;
        return !(tmp_4 == null) ? tmp_4 : THROW_CCE();
      case 'didLaunch':
        var tmp_5 = this.didLaunch;
        return !(tmp_5 == null) ? tmp_5 : THROW_CCE();
      case 'didResize':
        var tmp_6 = this.didResize;
        return !(tmp_6 == null) ? tmp_6 : THROW_CCE();
      case 'didSaveEditedFiles':
        var tmp_7 = this.didSaveEditedFiles;
        return !(tmp_7 == null) ? tmp_7 : THROW_CCE();
      case 'didSaveFile':
        var tmp_8 = this.didSaveFile;
        return !(tmp_8 == null) ? tmp_8 : THROW_CCE();
      case 'editedContents':
        var tmp_9 = this.editedContents;
        return !(tmp_9 == null) ? tmp_9 : THROW_CCE();
      case 'editedFileContents':
        var tmp_10 = this.editedFileContents;
        return !(tmp_10 == null) ? tmp_10 : THROW_CCE();
      case 'editorContents':
        var tmp_11 = this.editorContents;
        return !(tmp_11 == null) ? tmp_11 : THROW_CCE();
      case 'header':
        var tmp_12 = this.header;
        return !(tmp_12 == null) ? tmp_12 : THROW_CCE();
      case 'inputDirFiles':
        var tmp_13 = this.inputDirFiles;
        return !(tmp_13 == null) ? tmp_13 : THROW_CCE();
      case 'inputDirs':
        var tmp_14 = this.inputDirs;
        return !(tmp_14 == null) ? tmp_14 : THROW_CCE();
      case 'inputMDFiles':
        var tmp_15 = this.inputMDFiles;
        return !(tmp_15 == null) ? tmp_15 : THROW_CCE();
      case 'installEditor':
        var tmp_16 = this.installEditor;
        return !(tmp_16 == null) ? tmp_16 : THROW_CCE();
      case 'listInputDirId':
        var tmp_17 = this.listInputDirId;
        return !(tmp_17 == null) ? tmp_17 : THROW_CCE();
      case 'projectPath':
        var tmp_18 = this.projectPath;
        return !(tmp_18 == null) ? tmp_18 : THROW_CCE();
      case 'readFile':
        var tmp_19 = this.readFile;
        return !(tmp_19 == null) ? tmp_19 : THROW_CCE();
      case 'readFileContents':
        var tmp_20 = this.readFileContents;
        return !(tmp_20 == null) ? tmp_20 : THROW_CCE();
      case 'request':
        var tmp_21 = this.request;
        return !(tmp_21 == null) ? tmp_21 : THROW_CCE();
      case 'resizeEditor':
        var tmp_22 = this.resizeEditor;
        return !(tmp_22 == null) ? tmp_22 : THROW_CCE();
      case 'response':
        var tmp_23 = this.response;
        return !(tmp_23 == null) ? tmp_23 : THROW_CCE();
      case 'responseError':
        var tmp_24 = this.responseError;
        return !(tmp_24 == null) ? tmp_24 : THROW_CCE();
      case 'saveFileId':
        var tmp_25 = this.saveFileId;
        return !(tmp_25 == null) ? tmp_25 : THROW_CCE();
      case 'saveFiles':
        var tmp_26 = this.saveFiles;
        return !(tmp_26 == null) ? tmp_26 : THROW_CCE();
      case 'selectedFileId':
        var tmp_27 = this.selectedFileId;
        return !(tmp_27 == null) ? tmp_27 : THROW_CCE();
      case 'selectedFileName':
        var tmp_28 = this.selectedFileName;
        return !(tmp_28 == null) ? tmp_28 : THROW_CCE();
      case 'selectedTabId':
        var tmp_29 = this.selectedTabId;
        return !(tmp_29 == null) ? tmp_29 : THROW_CCE();
      case 'splashTimeout':
        var tmp_30 = this.splashTimeout;
        return !(tmp_30 == null) ? tmp_30 : THROW_CCE();
    }
    return !('unknown-field-name' == null) ? 'unknown-field-name' : THROW_CCE();
  };
  protoOf(AppContext).selfCopy = function () {
    return this.copy();
  };
  protoOf(AppContext).setField = function (name, value) {
    switch (name) {
      case 'baseURL':
        var tmp = this;
        tmp.baseURL = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'cfg':
        var tmp_0 = this;
        tmp_0.cfg = (!(value == null) ? isInterface(value, KtMap) : false) ? value : THROW_CCE();
        break;
      case 'didClickEditorTab':
        var tmp_1 = this;
        tmp_1.didClickEditorTab = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didClickFilesTab':
        var tmp_2 = this;
        tmp_2.didClickFilesTab = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didClickRenderTab':
        var tmp_3 = this;
        tmp_3.didClickRenderTab = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didClickSaveBtn':
        var tmp_4 = this;
        tmp_4.didClickSaveBtn = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didLaunch':
        var tmp_5 = this;
        tmp_5.didLaunch = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didResize':
        var tmp_6 = this;
        tmp_6.didResize = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didSaveEditedFiles':
        var tmp_7 = this;
        tmp_7.didSaveEditedFiles = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didSaveFile':
        var tmp_8 = this;
        tmp_8.didSaveFile = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'editedContents':
        var tmp_9 = this;
        tmp_9.editedContents = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'editedFileContents':
        var tmp_10 = this;
        tmp_10.editedFileContents = (!(value == null) ? isInterface(value, KtMap) : false) ? value : THROW_CCE();
        break;
      case 'editorContents':
        var tmp_11 = this;
        tmp_11.editorContents = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'header':
        var tmp_12 = this;
        tmp_12.header = (!(value == null) ? isArray(value) : false) ? value : THROW_CCE();
        break;
      case 'inputDirFiles':
        var tmp_13 = this;
        tmp_13.inputDirFiles = (!(value == null) ? isInterface(value, KtMap) : false) ? value : THROW_CCE();
        break;
      case 'inputDirs':
        var tmp_14 = this;
        tmp_14.inputDirs = (!(value == null) ? isArray(value) : false) ? value : THROW_CCE();
        break;
      case 'inputMDFiles':
        var tmp_15 = this;
        tmp_15.inputMDFiles = (!(value == null) ? isInterface(value, KtMap) : false) ? value : THROW_CCE();
        break;
      case 'installEditor':
        var tmp_16 = this;
        tmp_16.installEditor = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'listInputDirId':
        var tmp_17 = this;
        tmp_17.listInputDirId = (!(value == null) ? typeof value === 'number' : false) ? value : THROW_CCE();
        break;
      case 'projectPath':
        var tmp_18 = this;
        tmp_18.projectPath = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'readFile':
        var tmp_19 = this;
        tmp_19.readFile = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'readFileContents':
        var tmp_20 = this;
        tmp_20.readFileContents = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'request':
        var tmp_21 = this;
        tmp_21.request = value instanceof NetRequest ? value : THROW_CCE();
        break;
      case 'resizeEditor':
        var tmp_22 = this;
        tmp_22.resizeEditor = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'response':
        var tmp_23 = this;
        tmp_23.response = value instanceof NetResponse ? value : THROW_CCE();
        break;
      case 'responseError':
        var tmp_24 = this;
        tmp_24.responseError = value instanceof NetResponse ? value : THROW_CCE();
        break;
      case 'saveFileId':
        var tmp_25 = this;
        tmp_25.saveFileId = (!(value == null) ? typeof value === 'number' : false) ? value : THROW_CCE();
        break;
      case 'saveFiles':
        var tmp_26 = this;
        tmp_26.saveFiles = (!(value == null) ? isArray(value) : false) ? value : THROW_CCE();
        break;
      case 'selectedFileId':
        var tmp_27 = this;
        tmp_27.selectedFileId = (!(value == null) ? isArray(value) : false) ? value : THROW_CCE();
        break;
      case 'selectedFileName':
        var tmp_28 = this;
        tmp_28.selectedFileName = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'selectedTabId':
        var tmp_29 = this;
        tmp_29.selectedTabId = (!(value == null) ? typeof value === 'number' : false) ? value : THROW_CCE();
        break;
      case 'splashTimeout':
        var tmp_30 = this;
        tmp_30.splashTimeout = (!(value == null) ? typeof value === 'number' : false) ? value : THROW_CCE();
        break;
    }
  };
  protoOf(AppContext).component1_7eebsc_k$ = function () {
    return this.baseURL;
  };
  protoOf(AppContext).component2_7eebsb_k$ = function () {
    return this.cfg;
  };
  protoOf(AppContext).component3_7eebsa_k$ = function () {
    return this.didClickEditorTab;
  };
  protoOf(AppContext).component4_7eebs9_k$ = function () {
    return this.didClickFilesTab;
  };
  protoOf(AppContext).component5_7eebs8_k$ = function () {
    return this.didClickRenderTab;
  };
  protoOf(AppContext).component6_7eebs7_k$ = function () {
    return this.didClickSaveBtn;
  };
  protoOf(AppContext).component7_7eebs6_k$ = function () {
    return this.didLaunch;
  };
  protoOf(AppContext).component8_7eebs5_k$ = function () {
    return this.didResize;
  };
  protoOf(AppContext).component9_7eebs4_k$ = function () {
    return this.didSaveEditedFiles;
  };
  protoOf(AppContext).component10_gazzfo_k$ = function () {
    return this.didSaveFile;
  };
  protoOf(AppContext).component11_gazzfn_k$ = function () {
    return this.editedContents;
  };
  protoOf(AppContext).component12_gazzfm_k$ = function () {
    return this.editedFileContents;
  };
  protoOf(AppContext).component13_gazzfl_k$ = function () {
    return this.editorContents;
  };
  protoOf(AppContext).component14_gazzfk_k$ = function () {
    return this.header;
  };
  protoOf(AppContext).component15_gazzfj_k$ = function () {
    return this.inputDirFiles;
  };
  protoOf(AppContext).component16_gazzfi_k$ = function () {
    return this.inputDirs;
  };
  protoOf(AppContext).component17_gazzfh_k$ = function () {
    return this.inputMDFiles;
  };
  protoOf(AppContext).component18_gazzfg_k$ = function () {
    return this.installEditor;
  };
  protoOf(AppContext).component19_gazzff_k$ = function () {
    return this.listInputDirId;
  };
  protoOf(AppContext).component20_gazzet_k$ = function () {
    return this.projectPath;
  };
  protoOf(AppContext).component21_gazzes_k$ = function () {
    return this.readFile;
  };
  protoOf(AppContext).component22_gazzer_k$ = function () {
    return this.readFileContents;
  };
  protoOf(AppContext).component23_gazzeq_k$ = function () {
    return this.request;
  };
  protoOf(AppContext).component24_gazzep_k$ = function () {
    return this.resizeEditor;
  };
  protoOf(AppContext).component25_gazzeo_k$ = function () {
    return this.response;
  };
  protoOf(AppContext).component26_gazzen_k$ = function () {
    return this.responseError;
  };
  protoOf(AppContext).component27_gazzem_k$ = function () {
    return this.saveFileId;
  };
  protoOf(AppContext).component28_gazzel_k$ = function () {
    return this.saveFiles;
  };
  protoOf(AppContext).component29_gazzek_k$ = function () {
    return this.selectedFileId;
  };
  protoOf(AppContext).component30_gazzdy_k$ = function () {
    return this.selectedFileName;
  };
  protoOf(AppContext).component31_gazzdx_k$ = function () {
    return this.selectedTabId;
  };
  protoOf(AppContext).component32_gazzdw_k$ = function () {
    return this.splashTimeout;
  };
  protoOf(AppContext).component33_gazzdv_k$ = function () {
    return this.recentField_1;
  };
  protoOf(AppContext).copy_6mdfce_k$ = function (baseURL, cfg, didClickEditorTab, didClickFilesTab, didClickRenderTab, didClickSaveBtn, didLaunch, didResize, didSaveEditedFiles, didSaveFile, editedContents, editedFileContents, editorContents, header, inputDirFiles, inputDirs, inputMDFiles, installEditor, listInputDirId, projectPath, readFile, readFileContents, request, resizeEditor, response, responseError, saveFileId, saveFiles, selectedFileId, selectedFileName, selectedTabId, splashTimeout, recentField) {
    return new AppContext(baseURL, cfg, didClickEditorTab, didClickFilesTab, didClickRenderTab, didClickSaveBtn, didLaunch, didResize, didSaveEditedFiles, didSaveFile, editedContents, editedFileContents, editorContents, header, inputDirFiles, inputDirs, inputMDFiles, installEditor, listInputDirId, projectPath, readFile, readFileContents, request, resizeEditor, response, responseError, saveFileId, saveFiles, selectedFileId, selectedFileName, selectedTabId, splashTimeout, recentField);
  };
  protoOf(AppContext).copy = function (baseURL, cfg, didClickEditorTab, didClickFilesTab, didClickRenderTab, didClickSaveBtn, didLaunch, didResize, didSaveEditedFiles, didSaveFile, editedContents, editedFileContents, editorContents, header, inputDirFiles, inputDirs, inputMDFiles, installEditor, listInputDirId, projectPath, readFile, readFileContents, request, resizeEditor, response, responseError, saveFileId, saveFiles, selectedFileId, selectedFileName, selectedTabId, splashTimeout, recentField, $super) {
    baseURL = baseURL === VOID ? this.baseURL : baseURL;
    cfg = cfg === VOID ? this.cfg : cfg;
    didClickEditorTab = didClickEditorTab === VOID ? this.didClickEditorTab : didClickEditorTab;
    didClickFilesTab = didClickFilesTab === VOID ? this.didClickFilesTab : didClickFilesTab;
    didClickRenderTab = didClickRenderTab === VOID ? this.didClickRenderTab : didClickRenderTab;
    didClickSaveBtn = didClickSaveBtn === VOID ? this.didClickSaveBtn : didClickSaveBtn;
    didLaunch = didLaunch === VOID ? this.didLaunch : didLaunch;
    didResize = didResize === VOID ? this.didResize : didResize;
    didSaveEditedFiles = didSaveEditedFiles === VOID ? this.didSaveEditedFiles : didSaveEditedFiles;
    didSaveFile = didSaveFile === VOID ? this.didSaveFile : didSaveFile;
    editedContents = editedContents === VOID ? this.editedContents : editedContents;
    editedFileContents = editedFileContents === VOID ? this.editedFileContents : editedFileContents;
    editorContents = editorContents === VOID ? this.editorContents : editorContents;
    header = header === VOID ? this.header : header;
    inputDirFiles = inputDirFiles === VOID ? this.inputDirFiles : inputDirFiles;
    inputDirs = inputDirs === VOID ? this.inputDirs : inputDirs;
    inputMDFiles = inputMDFiles === VOID ? this.inputMDFiles : inputMDFiles;
    installEditor = installEditor === VOID ? this.installEditor : installEditor;
    listInputDirId = listInputDirId === VOID ? this.listInputDirId : listInputDirId;
    projectPath = projectPath === VOID ? this.projectPath : projectPath;
    readFile = readFile === VOID ? this.readFile : readFile;
    readFileContents = readFileContents === VOID ? this.readFileContents : readFileContents;
    request = request === VOID ? this.request : request;
    resizeEditor = resizeEditor === VOID ? this.resizeEditor : resizeEditor;
    response = response === VOID ? this.response : response;
    responseError = responseError === VOID ? this.responseError : responseError;
    saveFileId = saveFileId === VOID ? this.saveFileId : saveFileId;
    saveFiles = saveFiles === VOID ? this.saveFiles : saveFiles;
    selectedFileId = selectedFileId === VOID ? this.selectedFileId : selectedFileId;
    selectedFileName = selectedFileName === VOID ? this.selectedFileName : selectedFileName;
    selectedTabId = selectedTabId === VOID ? this.selectedTabId : selectedTabId;
    splashTimeout = splashTimeout === VOID ? this.splashTimeout : splashTimeout;
    recentField = recentField === VOID ? this.recentField_1 : recentField;
    return $super === VOID ? this.copy_6mdfce_k$(baseURL, cfg, didClickEditorTab, didClickFilesTab, didClickRenderTab, didClickSaveBtn, didLaunch, didResize, didSaveEditedFiles, didSaveFile, editedContents, editedFileContents, editorContents, header, inputDirFiles, inputDirs, inputMDFiles, installEditor, listInputDirId, projectPath, readFile, readFileContents, request, resizeEditor, response, responseError, saveFileId, saveFiles, selectedFileId, selectedFileName, selectedTabId, splashTimeout, recentField) : $super.copy_6mdfce_k$.call(this, baseURL, cfg, didClickEditorTab, didClickFilesTab, didClickRenderTab, didClickSaveBtn, didLaunch, didResize, didSaveEditedFiles, didSaveFile, editedContents, editedFileContents, editorContents, header, inputDirFiles, inputDirs, inputMDFiles, installEditor, listInputDirId, projectPath, readFile, readFileContents, request, resizeEditor, response, responseError, saveFileId, saveFiles, selectedFileId, selectedFileName, selectedTabId, splashTimeout, recentField);
  };
  protoOf(AppContext).toString = function () {
    return 'AppContext(baseURL=' + this.baseURL + ', cfg=' + toString(this.cfg) + ', didClickEditorTab=' + this.didClickEditorTab + ', didClickFilesTab=' + this.didClickFilesTab + ', didClickRenderTab=' + this.didClickRenderTab + ', didClickSaveBtn=' + this.didClickSaveBtn + ', didLaunch=' + this.didLaunch + ', didResize=' + this.didResize + ', didSaveEditedFiles=' + this.didSaveEditedFiles + ', didSaveFile=' + this.didSaveFile + ', editedContents=' + this.editedContents + ', editedFileContents=' + toString(this.editedFileContents) + ', editorContents=' + this.editorContents + ', header=' + toString(this.header) + ', inputDirFiles=' + toString(this.inputDirFiles) + ', inputDirs=' + toString(this.inputDirs) + ', inputMDFiles=' + toString(this.inputMDFiles) + ', installEditor=' + this.installEditor + ', listInputDirId=' + this.listInputDirId + ', projectPath=' + this.projectPath + ', readFile=' + this.readFile + ', readFileContents=' + this.readFileContents + ', request=' + this.request.toString() + ', resizeEditor=' + this.resizeEditor + ', response=' + this.response.toString() + ', responseError=' + this.responseError.toString() + ', saveFileId=' + this.saveFileId + ', saveFiles=' + toString(this.saveFiles) + ', selectedFileId=' + toString(this.selectedFileId) + ', selectedFileName=' + this.selectedFileName + ', selectedTabId=' + this.selectedTabId + ', splashTimeout=' + this.splashTimeout + ', recentField=' + this.recentField_1 + ')';
  };
  protoOf(AppContext).hashCode = function () {
    var result = getStringHashCode(this.baseURL);
    result = imul(result, 31) + hashCode(this.cfg) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didClickEditorTab) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didClickFilesTab) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didClickRenderTab) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didClickSaveBtn) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didLaunch) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didResize) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didSaveEditedFiles) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didSaveFile) | 0;
    result = imul(result, 31) + getStringHashCode(this.editedContents) | 0;
    result = imul(result, 31) + hashCode(this.editedFileContents) | 0;
    result = imul(result, 31) + getStringHashCode(this.editorContents) | 0;
    result = imul(result, 31) + hashCode(this.header) | 0;
    result = imul(result, 31) + hashCode(this.inputDirFiles) | 0;
    result = imul(result, 31) + hashCode(this.inputDirs) | 0;
    result = imul(result, 31) + hashCode(this.inputMDFiles) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.installEditor) | 0;
    result = imul(result, 31) + this.listInputDirId | 0;
    result = imul(result, 31) + getStringHashCode(this.projectPath) | 0;
    result = imul(result, 31) + getStringHashCode(this.readFile) | 0;
    result = imul(result, 31) + getStringHashCode(this.readFileContents) | 0;
    result = imul(result, 31) + this.request.hashCode() | 0;
    result = imul(result, 31) + getBooleanHashCode(this.resizeEditor) | 0;
    result = imul(result, 31) + this.response.hashCode() | 0;
    result = imul(result, 31) + this.responseError.hashCode() | 0;
    result = imul(result, 31) + this.saveFileId | 0;
    result = imul(result, 31) + hashCode(this.saveFiles) | 0;
    result = imul(result, 31) + hashCode(this.selectedFileId) | 0;
    result = imul(result, 31) + getStringHashCode(this.selectedFileName) | 0;
    result = imul(result, 31) + this.selectedTabId | 0;
    result = imul(result, 31) + this.splashTimeout | 0;
    result = imul(result, 31) + getStringHashCode(this.recentField_1) | 0;
    return result;
  };
  protoOf(AppContext).equals = function (other) {
    if (this === other)
      return true;
    if (!(other instanceof AppContext))
      return false;
    var tmp0_other_with_cast = other instanceof AppContext ? other : THROW_CCE();
    if (!(this.baseURL === tmp0_other_with_cast.baseURL))
      return false;
    if (!equals(this.cfg, tmp0_other_with_cast.cfg))
      return false;
    if (!(this.didClickEditorTab === tmp0_other_with_cast.didClickEditorTab))
      return false;
    if (!(this.didClickFilesTab === tmp0_other_with_cast.didClickFilesTab))
      return false;
    if (!(this.didClickRenderTab === tmp0_other_with_cast.didClickRenderTab))
      return false;
    if (!(this.didClickSaveBtn === tmp0_other_with_cast.didClickSaveBtn))
      return false;
    if (!(this.didLaunch === tmp0_other_with_cast.didLaunch))
      return false;
    if (!(this.didResize === tmp0_other_with_cast.didResize))
      return false;
    if (!(this.didSaveEditedFiles === tmp0_other_with_cast.didSaveEditedFiles))
      return false;
    if (!(this.didSaveFile === tmp0_other_with_cast.didSaveFile))
      return false;
    if (!(this.editedContents === tmp0_other_with_cast.editedContents))
      return false;
    if (!equals(this.editedFileContents, tmp0_other_with_cast.editedFileContents))
      return false;
    if (!(this.editorContents === tmp0_other_with_cast.editorContents))
      return false;
    if (!equals(this.header, tmp0_other_with_cast.header))
      return false;
    if (!equals(this.inputDirFiles, tmp0_other_with_cast.inputDirFiles))
      return false;
    if (!equals(this.inputDirs, tmp0_other_with_cast.inputDirs))
      return false;
    if (!equals(this.inputMDFiles, tmp0_other_with_cast.inputMDFiles))
      return false;
    if (!(this.installEditor === tmp0_other_with_cast.installEditor))
      return false;
    if (!(this.listInputDirId === tmp0_other_with_cast.listInputDirId))
      return false;
    if (!(this.projectPath === tmp0_other_with_cast.projectPath))
      return false;
    if (!(this.readFile === tmp0_other_with_cast.readFile))
      return false;
    if (!(this.readFileContents === tmp0_other_with_cast.readFileContents))
      return false;
    if (!this.request.equals(tmp0_other_with_cast.request))
      return false;
    if (!(this.resizeEditor === tmp0_other_with_cast.resizeEditor))
      return false;
    if (!this.response.equals(tmp0_other_with_cast.response))
      return false;
    if (!this.responseError.equals(tmp0_other_with_cast.responseError))
      return false;
    if (!(this.saveFileId === tmp0_other_with_cast.saveFileId))
      return false;
    if (!equals(this.saveFiles, tmp0_other_with_cast.saveFiles))
      return false;
    if (!equals(this.selectedFileId, tmp0_other_with_cast.selectedFileId))
      return false;
    if (!(this.selectedFileName === tmp0_other_with_cast.selectedFileName))
      return false;
    if (!(this.selectedTabId === tmp0_other_with_cast.selectedTabId))
      return false;
    if (!(this.splashTimeout === tmp0_other_with_cast.splashTimeout))
      return false;
    if (!(this.recentField_1 === tmp0_other_with_cast.recentField_1))
      return false;
    return true;
  };
  function FSFile(isFile, path) {
    isFile = isFile === VOID ? false : isFile;
    path = path === VOID ? '' : path;
    this.isFile = isFile;
    this.path = path;
  }
  protoOf(FSFile).set_isFile_2v2z1u_k$ = function (_set____db54di) {
    this.isFile = _set____db54di;
  };
  protoOf(FSFile).get_isFile_evxosv_k$ = function () {
    return this.isFile;
  };
  protoOf(FSFile).set_path_tgrnnc_k$ = function (_set____db54di) {
    this.path = _set____db54di;
  };
  protoOf(FSFile).get_path_wos8ry_k$ = function () {
    return this.path;
  };
  protoOf(FSFile).component1_7eebsc_k$ = function () {
    return this.isFile;
  };
  protoOf(FSFile).component2_7eebsb_k$ = function () {
    return this.path;
  };
  protoOf(FSFile).copy_miatnu_k$ = function (isFile, path) {
    return new FSFile(isFile, path);
  };
  protoOf(FSFile).copy = function (isFile, path, $super) {
    isFile = isFile === VOID ? this.isFile : isFile;
    path = path === VOID ? this.path : path;
    return $super === VOID ? this.copy_miatnu_k$(isFile, path) : $super.copy_miatnu_k$.call(this, isFile, path);
  };
  protoOf(FSFile).toString = function () {
    return 'FSFile(isFile=' + this.isFile + ', path=' + this.path + ')';
  };
  protoOf(FSFile).hashCode = function () {
    var result = getBooleanHashCode(this.isFile);
    result = imul(result, 31) + getStringHashCode(this.path) | 0;
    return result;
  };
  protoOf(FSFile).equals = function (other) {
    if (this === other)
      return true;
    if (!(other instanceof FSFile))
      return false;
    var tmp0_other_with_cast = other instanceof FSFile ? other : THROW_CCE();
    if (!(this.isFile === tmp0_other_with_cast.isFile))
      return false;
    if (!(this.path === tmp0_other_with_cast.path))
      return false;
    return true;
  };
  function NetRequest(body, method, url) {
    body = body === VOID ? '' : body;
    method = method === VOID ? '' : method;
    url = url === VOID ? '' : url;
    this.body = body;
    this.method = method;
    this.url = url;
  }
  protoOf(NetRequest).set_body_pyrtp1_k$ = function (_set____db54di) {
    this.body = _set____db54di;
  };
  protoOf(NetRequest).get_body_wojkyz_k$ = function () {
    return this.body;
  };
  protoOf(NetRequest).set_method_j1pto4_k$ = function (_set____db54di) {
    this.method = _set____db54di;
  };
  protoOf(NetRequest).get_method_gl8esq_k$ = function () {
    return this.method;
  };
  protoOf(NetRequest).set_url_caxf3c_k$ = function (_set____db54di) {
    this.url = _set____db54di;
  };
  protoOf(NetRequest).get_url_18iuii_k$ = function () {
    return this.url;
  };
  protoOf(NetRequest).component1_7eebsc_k$ = function () {
    return this.body;
  };
  protoOf(NetRequest).component2_7eebsb_k$ = function () {
    return this.method;
  };
  protoOf(NetRequest).component3_7eebsa_k$ = function () {
    return this.url;
  };
  protoOf(NetRequest).copy_nc7k0r_k$ = function (body, method, url) {
    return new NetRequest(body, method, url);
  };
  protoOf(NetRequest).copy = function (body, method, url, $super) {
    body = body === VOID ? this.body : body;
    method = method === VOID ? this.method : method;
    url = url === VOID ? this.url : url;
    return $super === VOID ? this.copy_nc7k0r_k$(body, method, url) : $super.copy_nc7k0r_k$.call(this, body, method, url);
  };
  protoOf(NetRequest).toString = function () {
    return 'NetRequest(body=' + this.body + ', method=' + this.method + ', url=' + this.url + ')';
  };
  protoOf(NetRequest).hashCode = function () {
    var result = getStringHashCode(this.body);
    result = imul(result, 31) + getStringHashCode(this.method) | 0;
    result = imul(result, 31) + getStringHashCode(this.url) | 0;
    return result;
  };
  protoOf(NetRequest).equals = function (other) {
    if (this === other)
      return true;
    if (!(other instanceof NetRequest))
      return false;
    var tmp0_other_with_cast = other instanceof NetRequest ? other : THROW_CCE();
    if (!(this.body === tmp0_other_with_cast.body))
      return false;
    if (!(this.method === tmp0_other_with_cast.method))
      return false;
    if (!(this.url === tmp0_other_with_cast.url))
      return false;
    return true;
  };
  function NetResponse(contents, req) {
    contents = contents === VOID ? '' : contents;
    req = req === VOID ? new NetRequest() : req;
    this.contents = contents;
    this.req = req;
  }
  protoOf(NetResponse).set_contents_mlm2h9_k$ = function (_set____db54di) {
    this.contents = _set____db54di;
  };
  protoOf(NetResponse).get_contents_tuemzn_k$ = function () {
    return this.contents;
  };
  protoOf(NetResponse).set_req_rk19il_k$ = function (_set____db54di) {
    this.req = _set____db54di;
  };
  protoOf(NetResponse).get_req_18ix1n_k$ = function () {
    return this.req;
  };
  protoOf(NetResponse).component1_7eebsc_k$ = function () {
    return this.contents;
  };
  protoOf(NetResponse).component2_7eebsb_k$ = function () {
    return this.req;
  };
  protoOf(NetResponse).copy_crxm1z_k$ = function (contents, req) {
    return new NetResponse(contents, req);
  };
  protoOf(NetResponse).copy = function (contents, req, $super) {
    contents = contents === VOID ? this.contents : contents;
    req = req === VOID ? this.req : req;
    return $super === VOID ? this.copy_crxm1z_k$(contents, req) : $super.copy_crxm1z_k$.call(this, contents, req);
  };
  protoOf(NetResponse).toString = function () {
    return 'NetResponse(contents=' + this.contents + ', req=' + this.req.toString() + ')';
  };
  protoOf(NetResponse).hashCode = function () {
    var result = getStringHashCode(this.contents);
    result = imul(result, 31) + this.req.hashCode() | 0;
    return result;
  };
  protoOf(NetResponse).equals = function (other) {
    if (this === other)
      return true;
    if (!(other instanceof NetResponse))
      return false;
    var tmp0_other_with_cast = other instanceof NetResponse ? other : THROW_CCE();
    if (!(this.contents === tmp0_other_with_cast.contents))
      return false;
    if (!this.req.equals(tmp0_other_with_cast.req))
      return false;
    return true;
  };
  function SrvContext(arguments_0, browserDir, defaultHTTPPort, didLaunch, didWriteFile, dirFiles, httpPort, inputDir, inputHTTPPort, listDir, projectAbsPath, projectDir, readFile, readFileContents, request, response, url, writeFile, recentField) {
    var tmp;
    if (arguments_0 === VOID) {
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp = [];
    } else {
      tmp = arguments_0;
    }
    arguments_0 = tmp;
    browserDir = browserDir === VOID ? '' : browserDir;
    defaultHTTPPort = defaultHTTPPort === VOID ? 0 : defaultHTTPPort;
    didLaunch = didLaunch === VOID ? false : didLaunch;
    didWriteFile = didWriteFile === VOID ? false : didWriteFile;
    var tmp_0;
    if (dirFiles === VOID) {
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp_0 = [];
    } else {
      tmp_0 = dirFiles;
    }
    dirFiles = tmp_0;
    httpPort = httpPort === VOID ? 0 : httpPort;
    inputDir = inputDir === VOID ? '' : inputDir;
    inputHTTPPort = inputHTTPPort === VOID ? 0 : inputHTTPPort;
    listDir = listDir === VOID ? '' : listDir;
    projectAbsPath = projectAbsPath === VOID ? '' : projectAbsPath;
    projectDir = projectDir === VOID ? '' : projectDir;
    readFile = readFile === VOID ? '' : readFile;
    readFileContents = readFileContents === VOID ? '' : readFileContents;
    request = request === VOID ? new NetRequest() : request;
    response = response === VOID ? new NetResponse() : response;
    url = url === VOID ? '' : url;
    var tmp_1;
    if (writeFile === VOID) {
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp_1 = [];
    } else {
      tmp_1 = writeFile;
    }
    writeFile = tmp_1;
    recentField = recentField === VOID ? '' : recentField;
    this.arguments = arguments_0;
    this.browserDir = browserDir;
    this.defaultHTTPPort = defaultHTTPPort;
    this.didLaunch = didLaunch;
    this.didWriteFile = didWriteFile;
    this.dirFiles = dirFiles;
    this.httpPort = httpPort;
    this.inputDir = inputDir;
    this.inputHTTPPort = inputHTTPPort;
    this.listDir = listDir;
    this.projectAbsPath = projectAbsPath;
    this.projectDir = projectDir;
    this.readFile = readFile;
    this.readFileContents = readFileContents;
    this.request = request;
    this.response = response;
    this.url = url;
    this.writeFile = writeFile;
    this.recentField_1 = recentField;
  }
  protoOf(SrvContext).set_arguments_vjnq4p_k$ = function (_set____db54di) {
    this.arguments = _set____db54di;
  };
  protoOf(SrvContext).get_arguments_p5ddub_k$ = function () {
    return this.arguments;
  };
  protoOf(SrvContext).set_browserDir_siqaqg_k$ = function (_set____db54di) {
    this.browserDir = _set____db54di;
  };
  protoOf(SrvContext).get_browserDir_mvmuk2_k$ = function () {
    return this.browserDir;
  };
  protoOf(SrvContext).set_defaultHTTPPort_7jsy1n_k$ = function (_set____db54di) {
    this.defaultHTTPPort = _set____db54di;
  };
  protoOf(SrvContext).get_defaultHTTPPort_bfq1z5_k$ = function () {
    return this.defaultHTTPPort;
  };
  protoOf(SrvContext).set_didLaunch_x1bx8g_k$ = function (_set____db54di) {
    this.didLaunch = _set____db54di;
  };
  protoOf(SrvContext).get_didLaunch_bo02hj_k$ = function () {
    return this.didLaunch;
  };
  protoOf(SrvContext).set_didWriteFile_78khrw_k$ = function (_set____db54di) {
    this.didWriteFile = _set____db54di;
  };
  protoOf(SrvContext).get_didWriteFile_l4l8cb_k$ = function () {
    return this.didWriteFile;
  };
  protoOf(SrvContext).set_dirFiles_k800co_k$ = function (_set____db54di) {
    this.dirFiles = _set____db54di;
  };
  protoOf(SrvContext).get_dirFiles_s7e4wd_k$ = function () {
    return this.dirFiles;
  };
  protoOf(SrvContext).set_httpPort_3vgn6q_k$ = function (_set____db54di) {
    this.httpPort = _set____db54di;
  };
  protoOf(SrvContext).get_httpPort_y08mj2_k$ = function () {
    return this.httpPort;
  };
  protoOf(SrvContext).set_inputDir_cuogsq_k$ = function (_set____db54di) {
    this.inputDir = _set____db54di;
  };
  protoOf(SrvContext).get_inputDir_o0p80k_k$ = function () {
    return this.inputDir;
  };
  protoOf(SrvContext).set_inputHTTPPort_rd3ocu_k$ = function (_set____db54di) {
    this.inputHTTPPort = _set____db54di;
  };
  protoOf(SrvContext).get_inputHTTPPort_gy7xze_k$ = function () {
    return this.inputHTTPPort;
  };
  protoOf(SrvContext).set_listDir_s1krew_k$ = function (_set____db54di) {
    this.listDir = _set____db54di;
  };
  protoOf(SrvContext).get_listDir_49vjty_k$ = function () {
    return this.listDir;
  };
  protoOf(SrvContext).set_projectAbsPath_hfl7n3_k$ = function (_set____db54di) {
    this.projectAbsPath = _set____db54di;
  };
  protoOf(SrvContext).get_projectAbsPath_r6lu93_k$ = function () {
    return this.projectAbsPath;
  };
  protoOf(SrvContext).set_projectDir_nicdk7_k$ = function (_set____db54di) {
    this.projectDir = _set____db54di;
  };
  protoOf(SrvContext).get_projectDir_i9z4n1_k$ = function () {
    return this.projectDir;
  };
  protoOf(SrvContext).set_readFile_lgagpx_k$ = function (_set____db54di) {
    this.readFile = _set____db54di;
  };
  protoOf(SrvContext).get_readFile_oveznf_k$ = function () {
    return this.readFile;
  };
  protoOf(SrvContext).set_readFileContents_6a4sbl_k$ = function (_set____db54di) {
    this.readFileContents = _set____db54di;
  };
  protoOf(SrvContext).get_readFileContents_s4pvpn_k$ = function () {
    return this.readFileContents;
  };
  protoOf(SrvContext).set_request_qynj8c_k$ = function (_set____db54di) {
    this.request = _set____db54di;
  };
  protoOf(SrvContext).get_request_jdwg4m_k$ = function () {
    return this.request;
  };
  protoOf(SrvContext).set_response_p6qtoa_k$ = function (_set____db54di) {
    this.response = _set____db54di;
  };
  protoOf(SrvContext).get_response_xlk07e_k$ = function () {
    return this.response;
  };
  protoOf(SrvContext).set_url_caxf3c_k$ = function (_set____db54di) {
    this.url = _set____db54di;
  };
  protoOf(SrvContext).get_url_18iuii_k$ = function () {
    return this.url;
  };
  protoOf(SrvContext).set_writeFile_xdnmm4_k$ = function (_set____db54di) {
    this.writeFile = _set____db54di;
  };
  protoOf(SrvContext).get_writeFile_er0omm_k$ = function () {
    return this.writeFile;
  };
  protoOf(SrvContext).set_recentField_5jv888_k$ = function (_set____db54di) {
    this.recentField_1 = _set____db54di;
  };
  protoOf(SrvContext).get_recentField_jy0xl6_k$ = function () {
    return this.recentField_1;
  };
  protoOf(SrvContext).field = function (name) {
    switch (name) {
      case 'arguments':
        var tmp = this.arguments;
        return !(tmp == null) ? tmp : THROW_CCE();
      case 'browserDir':
        var tmp_0 = this.browserDir;
        return !(tmp_0 == null) ? tmp_0 : THROW_CCE();
      case 'defaultHTTPPort':
        var tmp_1 = this.defaultHTTPPort;
        return !(tmp_1 == null) ? tmp_1 : THROW_CCE();
      case 'didLaunch':
        var tmp_2 = this.didLaunch;
        return !(tmp_2 == null) ? tmp_2 : THROW_CCE();
      case 'didWriteFile':
        var tmp_3 = this.didWriteFile;
        return !(tmp_3 == null) ? tmp_3 : THROW_CCE();
      case 'dirFiles':
        var tmp_4 = this.dirFiles;
        return !(tmp_4 == null) ? tmp_4 : THROW_CCE();
      case 'httpPort':
        var tmp_5 = this.httpPort;
        return !(tmp_5 == null) ? tmp_5 : THROW_CCE();
      case 'inputDir':
        var tmp_6 = this.inputDir;
        return !(tmp_6 == null) ? tmp_6 : THROW_CCE();
      case 'inputHTTPPort':
        var tmp_7 = this.inputHTTPPort;
        return !(tmp_7 == null) ? tmp_7 : THROW_CCE();
      case 'listDir':
        var tmp_8 = this.listDir;
        return !(tmp_8 == null) ? tmp_8 : THROW_CCE();
      case 'projectAbsPath':
        var tmp_9 = this.projectAbsPath;
        return !(tmp_9 == null) ? tmp_9 : THROW_CCE();
      case 'projectDir':
        var tmp_10 = this.projectDir;
        return !(tmp_10 == null) ? tmp_10 : THROW_CCE();
      case 'readFile':
        var tmp_11 = this.readFile;
        return !(tmp_11 == null) ? tmp_11 : THROW_CCE();
      case 'readFileContents':
        var tmp_12 = this.readFileContents;
        return !(tmp_12 == null) ? tmp_12 : THROW_CCE();
      case 'request':
        var tmp_13 = this.request;
        return !(tmp_13 == null) ? tmp_13 : THROW_CCE();
      case 'response':
        var tmp_14 = this.response;
        return !(tmp_14 == null) ? tmp_14 : THROW_CCE();
      case 'url':
        var tmp_15 = this.url;
        return !(tmp_15 == null) ? tmp_15 : THROW_CCE();
      case 'writeFile':
        var tmp_16 = this.writeFile;
        return !(tmp_16 == null) ? tmp_16 : THROW_CCE();
    }
    return !('unknown-field-name' == null) ? 'unknown-field-name' : THROW_CCE();
  };
  protoOf(SrvContext).selfCopy = function () {
    return this.copy();
  };
  protoOf(SrvContext).setField = function (name, value) {
    switch (name) {
      case 'arguments':
        var tmp = this;
        tmp.arguments = (!(value == null) ? isArray(value) : false) ? value : THROW_CCE();
        break;
      case 'browserDir':
        var tmp_0 = this;
        tmp_0.browserDir = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'defaultHTTPPort':
        var tmp_1 = this;
        tmp_1.defaultHTTPPort = (!(value == null) ? typeof value === 'number' : false) ? value : THROW_CCE();
        break;
      case 'didLaunch':
        var tmp_2 = this;
        tmp_2.didLaunch = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'didWriteFile':
        var tmp_3 = this;
        tmp_3.didWriteFile = (!(value == null) ? typeof value === 'boolean' : false) ? value : THROW_CCE();
        break;
      case 'dirFiles':
        var tmp_4 = this;
        tmp_4.dirFiles = (!(value == null) ? isArray(value) : false) ? value : THROW_CCE();
        break;
      case 'httpPort':
        var tmp_5 = this;
        tmp_5.httpPort = (!(value == null) ? typeof value === 'number' : false) ? value : THROW_CCE();
        break;
      case 'inputDir':
        var tmp_6 = this;
        tmp_6.inputDir = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'inputHTTPPort':
        var tmp_7 = this;
        tmp_7.inputHTTPPort = (!(value == null) ? typeof value === 'number' : false) ? value : THROW_CCE();
        break;
      case 'listDir':
        var tmp_8 = this;
        tmp_8.listDir = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'projectAbsPath':
        var tmp_9 = this;
        tmp_9.projectAbsPath = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'projectDir':
        var tmp_10 = this;
        tmp_10.projectDir = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'readFile':
        var tmp_11 = this;
        tmp_11.readFile = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'readFileContents':
        var tmp_12 = this;
        tmp_12.readFileContents = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'request':
        var tmp_13 = this;
        tmp_13.request = value instanceof NetRequest ? value : THROW_CCE();
        break;
      case 'response':
        var tmp_14 = this;
        tmp_14.response = value instanceof NetResponse ? value : THROW_CCE();
        break;
      case 'url':
        var tmp_15 = this;
        tmp_15.url = (!(value == null) ? typeof value === 'string' : false) ? value : THROW_CCE();
        break;
      case 'writeFile':
        var tmp_16 = this;
        tmp_16.writeFile = (!(value == null) ? isArray(value) : false) ? value : THROW_CCE();
        break;
    }
  };
  protoOf(SrvContext).component1_7eebsc_k$ = function () {
    return this.arguments;
  };
  protoOf(SrvContext).component2_7eebsb_k$ = function () {
    return this.browserDir;
  };
  protoOf(SrvContext).component3_7eebsa_k$ = function () {
    return this.defaultHTTPPort;
  };
  protoOf(SrvContext).component4_7eebs9_k$ = function () {
    return this.didLaunch;
  };
  protoOf(SrvContext).component5_7eebs8_k$ = function () {
    return this.didWriteFile;
  };
  protoOf(SrvContext).component6_7eebs7_k$ = function () {
    return this.dirFiles;
  };
  protoOf(SrvContext).component7_7eebs6_k$ = function () {
    return this.httpPort;
  };
  protoOf(SrvContext).component8_7eebs5_k$ = function () {
    return this.inputDir;
  };
  protoOf(SrvContext).component9_7eebs4_k$ = function () {
    return this.inputHTTPPort;
  };
  protoOf(SrvContext).component10_gazzfo_k$ = function () {
    return this.listDir;
  };
  protoOf(SrvContext).component11_gazzfn_k$ = function () {
    return this.projectAbsPath;
  };
  protoOf(SrvContext).component12_gazzfm_k$ = function () {
    return this.projectDir;
  };
  protoOf(SrvContext).component13_gazzfl_k$ = function () {
    return this.readFile;
  };
  protoOf(SrvContext).component14_gazzfk_k$ = function () {
    return this.readFileContents;
  };
  protoOf(SrvContext).component15_gazzfj_k$ = function () {
    return this.request;
  };
  protoOf(SrvContext).component16_gazzfi_k$ = function () {
    return this.response;
  };
  protoOf(SrvContext).component17_gazzfh_k$ = function () {
    return this.url;
  };
  protoOf(SrvContext).component18_gazzfg_k$ = function () {
    return this.writeFile;
  };
  protoOf(SrvContext).component19_gazzff_k$ = function () {
    return this.recentField_1;
  };
  protoOf(SrvContext).copy_9v6n38_k$ = function (arguments_0, browserDir, defaultHTTPPort, didLaunch, didWriteFile, dirFiles, httpPort, inputDir, inputHTTPPort, listDir, projectAbsPath, projectDir, readFile, readFileContents, request, response, url, writeFile, recentField) {
    return new SrvContext(arguments_0, browserDir, defaultHTTPPort, didLaunch, didWriteFile, dirFiles, httpPort, inputDir, inputHTTPPort, listDir, projectAbsPath, projectDir, readFile, readFileContents, request, response, url, writeFile, recentField);
  };
  protoOf(SrvContext).copy = function (arguments_0, browserDir, defaultHTTPPort, didLaunch, didWriteFile, dirFiles, httpPort, inputDir, inputHTTPPort, listDir, projectAbsPath, projectDir, readFile, readFileContents, request, response, url, writeFile, recentField, $super) {
    arguments_0 = arguments_0 === VOID ? this.arguments : arguments_0;
    browserDir = browserDir === VOID ? this.browserDir : browserDir;
    defaultHTTPPort = defaultHTTPPort === VOID ? this.defaultHTTPPort : defaultHTTPPort;
    didLaunch = didLaunch === VOID ? this.didLaunch : didLaunch;
    didWriteFile = didWriteFile === VOID ? this.didWriteFile : didWriteFile;
    dirFiles = dirFiles === VOID ? this.dirFiles : dirFiles;
    httpPort = httpPort === VOID ? this.httpPort : httpPort;
    inputDir = inputDir === VOID ? this.inputDir : inputDir;
    inputHTTPPort = inputHTTPPort === VOID ? this.inputHTTPPort : inputHTTPPort;
    listDir = listDir === VOID ? this.listDir : listDir;
    projectAbsPath = projectAbsPath === VOID ? this.projectAbsPath : projectAbsPath;
    projectDir = projectDir === VOID ? this.projectDir : projectDir;
    readFile = readFile === VOID ? this.readFile : readFile;
    readFileContents = readFileContents === VOID ? this.readFileContents : readFileContents;
    request = request === VOID ? this.request : request;
    response = response === VOID ? this.response : response;
    url = url === VOID ? this.url : url;
    writeFile = writeFile === VOID ? this.writeFile : writeFile;
    recentField = recentField === VOID ? this.recentField_1 : recentField;
    return $super === VOID ? this.copy_9v6n38_k$(arguments_0, browserDir, defaultHTTPPort, didLaunch, didWriteFile, dirFiles, httpPort, inputDir, inputHTTPPort, listDir, projectAbsPath, projectDir, readFile, readFileContents, request, response, url, writeFile, recentField) : $super.copy_9v6n38_k$.call(this, arguments_0, browserDir, defaultHTTPPort, didLaunch, didWriteFile, dirFiles, httpPort, inputDir, inputHTTPPort, listDir, projectAbsPath, projectDir, readFile, readFileContents, request, response, url, writeFile, recentField);
  };
  protoOf(SrvContext).toString = function () {
    return 'SrvContext(arguments=' + toString(this.arguments) + ', browserDir=' + this.browserDir + ', defaultHTTPPort=' + this.defaultHTTPPort + ', didLaunch=' + this.didLaunch + ', didWriteFile=' + this.didWriteFile + ', dirFiles=' + toString(this.dirFiles) + ', httpPort=' + this.httpPort + ', inputDir=' + this.inputDir + ', inputHTTPPort=' + this.inputHTTPPort + ', listDir=' + this.listDir + ', projectAbsPath=' + this.projectAbsPath + ', projectDir=' + this.projectDir + ', readFile=' + this.readFile + ', readFileContents=' + this.readFileContents + ', request=' + this.request.toString() + ', response=' + this.response.toString() + ', url=' + this.url + ', writeFile=' + toString(this.writeFile) + ', recentField=' + this.recentField_1 + ')';
  };
  protoOf(SrvContext).hashCode = function () {
    var result = hashCode(this.arguments);
    result = imul(result, 31) + getStringHashCode(this.browserDir) | 0;
    result = imul(result, 31) + this.defaultHTTPPort | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didLaunch) | 0;
    result = imul(result, 31) + getBooleanHashCode(this.didWriteFile) | 0;
    result = imul(result, 31) + hashCode(this.dirFiles) | 0;
    result = imul(result, 31) + this.httpPort | 0;
    result = imul(result, 31) + getStringHashCode(this.inputDir) | 0;
    result = imul(result, 31) + this.inputHTTPPort | 0;
    result = imul(result, 31) + getStringHashCode(this.listDir) | 0;
    result = imul(result, 31) + getStringHashCode(this.projectAbsPath) | 0;
    result = imul(result, 31) + getStringHashCode(this.projectDir) | 0;
    result = imul(result, 31) + getStringHashCode(this.readFile) | 0;
    result = imul(result, 31) + getStringHashCode(this.readFileContents) | 0;
    result = imul(result, 31) + this.request.hashCode() | 0;
    result = imul(result, 31) + this.response.hashCode() | 0;
    result = imul(result, 31) + getStringHashCode(this.url) | 0;
    result = imul(result, 31) + hashCode(this.writeFile) | 0;
    result = imul(result, 31) + getStringHashCode(this.recentField_1) | 0;
    return result;
  };
  protoOf(SrvContext).equals = function (other) {
    if (this === other)
      return true;
    if (!(other instanceof SrvContext))
      return false;
    var tmp0_other_with_cast = other instanceof SrvContext ? other : THROW_CCE();
    if (!equals(this.arguments, tmp0_other_with_cast.arguments))
      return false;
    if (!(this.browserDir === tmp0_other_with_cast.browserDir))
      return false;
    if (!(this.defaultHTTPPort === tmp0_other_with_cast.defaultHTTPPort))
      return false;
    if (!(this.didLaunch === tmp0_other_with_cast.didLaunch))
      return false;
    if (!(this.didWriteFile === tmp0_other_with_cast.didWriteFile))
      return false;
    if (!equals(this.dirFiles, tmp0_other_with_cast.dirFiles))
      return false;
    if (!(this.httpPort === tmp0_other_with_cast.httpPort))
      return false;
    if (!(this.inputDir === tmp0_other_with_cast.inputDir))
      return false;
    if (!(this.inputHTTPPort === tmp0_other_with_cast.inputHTTPPort))
      return false;
    if (!(this.listDir === tmp0_other_with_cast.listDir))
      return false;
    if (!(this.projectAbsPath === tmp0_other_with_cast.projectAbsPath))
      return false;
    if (!(this.projectDir === tmp0_other_with_cast.projectDir))
      return false;
    if (!(this.readFile === tmp0_other_with_cast.readFile))
      return false;
    if (!(this.readFileContents === tmp0_other_with_cast.readFileContents))
      return false;
    if (!this.request.equals(tmp0_other_with_cast.request))
      return false;
    if (!this.response.equals(tmp0_other_with_cast.response))
      return false;
    if (!(this.url === tmp0_other_with_cast.url))
      return false;
    if (!equals(this.writeFile, tmp0_other_with_cast.writeFile))
      return false;
    if (!(this.recentField_1 === tmp0_other_with_cast.recentField_1))
      return false;
    return true;
  };
  function cliArgumentValue(args, argument) {
    var inductionVariable = 0;
    var last = args.length;
    while (inductionVariable < last) {
      var item = args[inductionVariable];
      inductionVariable = inductionVariable + 1 | 0;
      if (startsWith(item, argument)) {
        var prefix = argument + '=';
        var value = substring(item, prefix.length);
        return value;
      }
    }
    return '';
  }
  function debugShort(str) {
    if (str.length > 200) {
      return take(str, 100) + '\u2026';
    }
    return str;
  }
  function debugString(v) {
    if (typeof v === 'string') {
      return 'S(' + v.length + ')' + v;
    }
    if (isArray(v)) {
      var out = '';
      var inductionVariable = 0;
      var last = v.length;
      while (inductionVariable < last) {
        var item = v[inductionVariable];
        inductionVariable = inductionVariable + 1 | 0;
        // Inline function 'kotlin.text.isEmpty' call
        var this_0 = out;
        if (!(charSequenceLength(this_0) === 0)) {
          out = out + ',';
        }
        out = out + debugString(ensureNotNull(item));
      }
      return 'A(' + v.length + ')' + out;
    }
    if (isInterface(v, KtMap)) {
      var out_0 = '';
      // Inline function 'kotlin.collections.iterator' call
      var _iterator__ex2g4s = v.get_entries_p20ztl_k$().iterator_jk1svi_k$();
      while (_iterator__ex2g4s.hasNext_bitz1p_k$()) {
        var _destruct__k2r9zo = _iterator__ex2g4s.next_20eer_k$();
        // Inline function 'kotlin.collections.component1' call
        var key = _destruct__k2r9zo.get_key_18j28a_k$();
        // Inline function 'kotlin.collections.component2' call
        var value = _destruct__k2r9zo.get_value_j01efc_k$();
        // Inline function 'kotlin.text.isEmpty' call
        var this_1 = out_0;
        if (!(charSequenceLength(this_1) === 0)) {
          out_0 = out_0 + ',';
        }
        out_0 = out_0 + (debugString(ensureNotNull(key)) + ':' + debugString(ensureNotNull(value)));
      }
      return 'D(' + v.get_size_woubt6_k$() + ')' + out_0;
    }
    return toString(v);
  }
  function fileContentsToJSON(file, contents) {
    var contentsB64 = stringToBase64(contents);
    return '{"path":"' + file + '","contents":"' + contentsB64 + '"}';
  }
  function filesToJSON(files) {
    // Inline function 'kotlin.arrayOf' call
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    var lines = [];
    var inductionVariable = 0;
    var last = files.length;
    while (inductionVariable < last) {
      var file = files[inductionVariable];
      inductionVariable = inductionVariable + 1 | 0;
      var type = file.isFile ? 'file' : 'dir';
      var ln = '{"path":"' + file.path + '","type":"' + type + '"}';
      // Inline function 'kotlin.collections.plus' call
      // Inline function 'kotlin.js.asDynamic' call
      var tmp$ret$3 = lines;
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      var tmp$ret$6 = [ln];
      lines = tmp$ret$3.concat(tmp$ret$6);
    }
    var out = '[' + joinToString(lines, ',') + ']';
    return out;
  }
  function forKIntVArrayString(dict, cb) {
    var _iterator__ex2g4s = dict.get_keys_wop4xp_k$().iterator_jk1svi_k$();
    while (_iterator__ex2g4s.hasNext_bitz1p_k$()) {
      var key = _iterator__ex2g4s.next_20eer_k$();
      var items = ensureNotNull(dict.get_wei43m_k$(key));
      cb(key, items);
    }
  }
  function jsonToFileContents(json) {
    // Inline function 'kotlin.collections.mutableMapOf' call
    var d = LinkedHashMap_init_$Create$();
    var parts = split(json, ['"']);
    if (parts.get_size_woubt6_k$() === 9) {
      // Inline function 'kotlin.collections.set' call
      var value = parts.get_c1px32_k$(3);
      d.put_4fpzoq_k$('path', value);
      var contentsB64 = parts.get_c1px32_k$(7);
      var tmp2 = 'contents';
      // Inline function 'kotlin.collections.set' call
      var value_0 = decodeToString(Default_getInstance().decode$default_mne14l_k$(contentsB64));
      d.put_4fpzoq_k$(tmp2, value_0);
    }
    return d;
  }
  function jsonToFiles(raw) {
    // Inline function 'kotlin.arrayOf' call
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    var items = [];
    var lessBrackets = substring_0(raw, 2, raw.length - 1 | 0);
    var lines = split(lessBrackets, [',{']);
    var prefix = '"path":"';
    var suffix = '"}';
    var middle = '","type":"';
    var _iterator__ex2g4s = lines.iterator_jk1svi_k$();
    while (_iterator__ex2g4s.hasNext_bitz1p_k$()) {
      var ln = _iterator__ex2g4s.next_20eer_k$();
      var lessLn = substring_0(ln, prefix.length, ln.length - suffix.length | 0);
      var parts = split(lessLn, [middle]);
      var f = new FSFile(parts.get_c1px32_k$(1) === 'file', parts.get_c1px32_k$(0));
      // Inline function 'kotlin.collections.plus' call
      // Inline function 'kotlin.js.asDynamic' call
      var tmp$ret$3 = items;
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      var tmp$ret$6 = [f];
      items = tmp$ret$3.concat(tmp$ret$6);
    }
    return items;
  }
  function mdFiles(d) {
    // Inline function 'kotlin.collections.mutableMapOf' call
    var mds = LinkedHashMap_init_$Create$();
    var _iterator__ex2g4s = d.get_keys_wop4xp_k$().iterator_jk1svi_k$();
    while (_iterator__ex2g4s.hasNext_bitz1p_k$()) {
      var id = _iterator__ex2g4s.next_20eer_k$();
      var files = ensureNotNull(d.get_wei43m_k$(id));
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      var mdFiles = [];
      var inductionVariable = 0;
      var last = files.length;
      while (inductionVariable < last) {
        var f = files[inductionVariable];
        inductionVariable = inductionVariable + 1 | 0;
        if (endsWith(f.path, '.' + get_CONST_EXT_MD())) {
          var tmp0 = mdFiles;
          // Inline function 'kotlin.collections.plus' call
          // Inline function 'kotlin.js.asDynamic' call
          // Inline function 'kotlin.arrayOf' call
          // Inline function 'kotlin.js.unsafeCast' call
          // Inline function 'kotlin.js.asDynamic' call
          var tmp$ret$7 = [f.path];
          mdFiles = tmp0.concat(tmp$ret$7);
        }
      }
      // Inline function 'kotlin.collections.set' call
      var value = mdFiles;
      mds.put_4fpzoq_k$(id, value);
    }
    return mds;
  }
  function parseCfg(raw) {
    // Inline function 'kotlin.collections.mutableMapOf' call
    var d = LinkedHashMap_init_$Create$();
    var lines = split(raw, ['\n']);
    var _iterator__ex2g4s = lines.iterator_jk1svi_k$();
    while (_iterator__ex2g4s.hasNext_bitz1p_k$()) {
      var ln = _iterator__ex2g4s.next_20eer_k$();
      var parts = split(ln, [' = ']);
      if (parts.get_size_woubt6_k$() === 2) {
        var k = parts.get_c1px32_k$(0);
        var v = parts.get_c1px32_k$(1);
        // Inline function 'kotlin.collections.set' call
        d.put_4fpzoq_k$(k, v);
      }
    }
    return d;
  }
  function stringToBase64(txt) {
    var arr = encodeToByteArray(txt);
    return Default_getInstance().encode$default_h4nonq_k$(arr);
  }
  function get_SRV_API_ROOT() {
    return SRV_API_ROOT;
  }
  var SRV_API_ROOT;
  function get_SRV_ARGUMENT_BROWSER_DIR() {
    return SRV_ARGUMENT_BROWSER_DIR;
  }
  var SRV_ARGUMENT_BROWSER_DIR;
  function get_SRV_ARGUMENT_PROJECT_DIR() {
    return SRV_ARGUMENT_PROJECT_DIR;
  }
  var SRV_ARGUMENT_PROJECT_DIR;
  function get_SRV_DEFAULT_HTTP_PORT() {
    return SRV_DEFAULT_HTTP_PORT;
  }
  var SRV_DEFAULT_HTTP_PORT;
  function get_SRV_INDEX() {
    return SRV_INDEX;
  }
  var SRV_INDEX;
  function srvShouldListDir(c) {
    if (c.recentField === 'request' && c.request.method === get_CONST_POST() && c.request.url === get_CONST_API_LIST()) {
      c.set_listDir_s1krew_k$(c.projectAbsPath + '/' + c.request.body);
      c.set_recentField_5jv888_k$('listDir');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  function srvShouldOpenURL(c) {
    if (c.recentField === 'didLaunch') {
      c.set_url_caxf3c_k$('http://localhost:' + c.httpPort);
      c.set_recentField_5jv888_k$('url');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  function srvShouldReadFile(c) {
    if (c.recentField === 'request' && c.request.method === get_CONST_GET() && c.request.url === SRV_API_ROOT) {
      c.set_readFile_lgagpx_k$(c.browserDir + '/' + SRV_INDEX);
      c.set_recentField_5jv888_k$('readFile');
      return c;
    }
    if (c.recentField === 'request' && c.request.method === get_CONST_GET() && !(c.request.url === get_CONST_API_PATH())) {
      c.set_readFile_lgagpx_k$(c.browserDir + c.request.url);
      c.set_recentField_5jv888_k$('readFile');
      return c;
    }
    if (c.recentField === 'request' && c.request.method === get_CONST_POST() && c.request.url === get_CONST_API_READ()) {
      c.set_readFile_lgagpx_k$(c.projectAbsPath + '/' + c.request.body);
      c.set_recentField_5jv888_k$('readFile');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  function srvShouldResetBrowserDir(c) {
    if (c.recentField === 'didLaunch') {
      c.set_browserDir_siqaqg_k$(cliArgumentValue(c.arguments, SRV_ARGUMENT_BROWSER_DIR));
      c.set_recentField_5jv888_k$('browserDir');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  function srvShouldResetHTTPPort(c) {
    if (c.recentField === 'defaultHTTPPort') {
      c.set_httpPort_3vgn6q_k$(c.defaultHTTPPort);
      c.set_recentField_5jv888_k$('httpPort');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  function srvShouldResetProjectDir(c) {
    if (c.recentField === 'didLaunch') {
      c.set_projectDir_nicdk7_k$(cliArgumentValue(c.arguments, SRV_ARGUMENT_PROJECT_DIR));
      c.set_recentField_5jv888_k$('projectDir');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  function srvShouldResetResponse(c) {
    if (c.recentField === 'readFileContents') {
      c.set_response_p6qtoa_k$(new NetResponse(c.readFileContents, c.request));
      c.set_recentField_5jv888_k$('response');
      return c;
    }
    if (c.recentField === 'request' && c.request.method === get_CONST_GET() && c.request.url === get_CONST_API_PATH()) {
      c.set_response_p6qtoa_k$(new NetResponse(c.projectAbsPath, c.request));
      c.set_recentField_5jv888_k$('response');
      return c;
    }
    if (c.recentField === 'dirFiles') {
      var json = filesToJSON(c.dirFiles);
      c.set_response_p6qtoa_k$(new NetResponse(json, c.request));
      c.set_recentField_5jv888_k$('response');
      return c;
    }
    if (c.recentField === 'didWriteFile') {
      var reply = '';
      if (!c.didWriteFile) {
        reply = 'ERR write';
      }
      c.set_response_p6qtoa_k$(new NetResponse(reply, c.request));
      c.set_recentField_5jv888_k$('response');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  function srvShouldWriteFile(c) {
    if (c.recentField === 'request' && c.request.method === get_CONST_POST() && c.request.url === get_CONST_API_WRITE()) {
      var d = jsonToFileContents(c.request.body);
      var fullPath = c.projectAbsPath + '/' + ensureNotNull(d.get_wei43m_k$('path'));
      // Inline function 'kotlin.arrayOf' call
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      var tmp$ret$2 = [fullPath, ensureNotNull(d.get_wei43m_k$('contents'))];
      c.set_writeFile_xdnmm4_k$(tmp$ret$2);
      c.set_recentField_5jv888_k$('writeFile');
      return c;
    }
    c.set_recentField_5jv888_k$('none');
    return c;
  }
  //region block: post-declaration
  protoOf(AppContext).fieldAny = fieldAny;
  defineProp(protoOf(AppContext), 'recentField', function () {
    return this.get_recentField_jy0xl6_k$();
  }, function (value) {
    this.set_recentField_5jv888_k$(value);
  });
  protoOf(SrvContext).fieldAny = fieldAny;
  defineProp(protoOf(SrvContext), 'recentField', function () {
    return this.get_recentField_jy0xl6_k$();
  }, function (value) {
    this.set_recentField_5jv888_k$(value);
  });
  //endregion
  //region block: init
  CONST_API_LIST = '/list';
  CONST_API_PATH = '/path';
  CONST_API_READ = '/read';
  CONST_API_WRITE = '/write';
  CONST_EXT_MD = 'md';
  CONST_GET = 'GET';
  CONST_POST = 'POST';
  SRV_API_ROOT = '/';
  SRV_ARGUMENT_BROWSER_DIR = '--browserDir';
  SRV_ARGUMENT_PROJECT_DIR = '--projectDir';
  SRV_DEFAULT_HTTP_PORT = 8000;
  SRV_INDEX = 'index.html';
  //endregion
  //region block: exports
  function $jsExportAll$(_) {
    var $org = _.org || (_.org = {});
    var $org$opengamestudio = $org.opengamestudio || ($org.opengamestudio = {});
    var $org = _.org || (_.org = {});
    var $org$opengamestudio = $org.opengamestudio || ($org.opengamestudio = {});
    $org$opengamestudio.CLDController = CLDController;
    var $org = _.org || (_.org = {});
    var $org$opengamestudio = $org.opengamestudio || ($org.opengamestudio = {});
    defineProp($org$opengamestudio, 'CONST_API_LIST', get_CONST_API_LIST);
    defineProp($org$opengamestudio, 'CONST_API_PATH', get_CONST_API_PATH);
    defineProp($org$opengamestudio, 'CONST_API_READ', get_CONST_API_READ);
    defineProp($org$opengamestudio, 'CONST_API_WRITE', get_CONST_API_WRITE);
    defineProp($org$opengamestudio, 'CONST_EXT_MD', get_CONST_EXT_MD);
    defineProp($org$opengamestudio, 'CONST_GET', get_CONST_GET);
    defineProp($org$opengamestudio, 'CONST_POST', get_CONST_POST);
    var $org = _.org || (_.org = {});
    var $org$opengamestudio = $org.opengamestudio || ($org.opengamestudio = {});
    $org$opengamestudio.AppContext = AppContext;
    $org$opengamestudio.FSFile = FSFile;
    $org$opengamestudio.NetRequest = NetRequest;
    $org$opengamestudio.NetResponse = NetResponse;
    $org$opengamestudio.SrvContext = SrvContext;
    var $org = _.org || (_.org = {});
    var $org$opengamestudio = $org.opengamestudio || ($org.opengamestudio = {});
    $org$opengamestudio.cliArgumentValue = cliArgumentValue;
    $org$opengamestudio.debugShort = debugShort;
    $org$opengamestudio.debugString = debugString;
    $org$opengamestudio.fileContentsToJSON = fileContentsToJSON;
    $org$opengamestudio.filesToJSON = filesToJSON;
    $org$opengamestudio.forKIntVArrayString = forKIntVArrayString;
    $org$opengamestudio.jsonToFileContents = jsonToFileContents;
    $org$opengamestudio.jsonToFiles = jsonToFiles;
    $org$opengamestudio.mdFiles = mdFiles;
    $org$opengamestudio.parseCfg = parseCfg;
    $org$opengamestudio.stringToBase64 = stringToBase64;
    var $org = _.org || (_.org = {});
    var $org$opengamestudio = $org.opengamestudio || ($org.opengamestudio = {});
    defineProp($org$opengamestudio, 'SRV_API_ROOT', get_SRV_API_ROOT);
    defineProp($org$opengamestudio, 'SRV_ARGUMENT_BROWSER_DIR', get_SRV_ARGUMENT_BROWSER_DIR);
    defineProp($org$opengamestudio, 'SRV_ARGUMENT_PROJECT_DIR', get_SRV_ARGUMENT_PROJECT_DIR);
    defineProp($org$opengamestudio, 'SRV_DEFAULT_HTTP_PORT', get_SRV_DEFAULT_HTTP_PORT);
    defineProp($org$opengamestudio, 'SRV_INDEX', get_SRV_INDEX);
    $org$opengamestudio.srvShouldListDir = srvShouldListDir;
    $org$opengamestudio.srvShouldOpenURL = srvShouldOpenURL;
    $org$opengamestudio.srvShouldReadFile = srvShouldReadFile;
    $org$opengamestudio.srvShouldResetBrowserDir = srvShouldResetBrowserDir;
    $org$opengamestudio.srvShouldResetHTTPPort = srvShouldResetHTTPPort;
    $org$opengamestudio.srvShouldResetProjectDir = srvShouldResetProjectDir;
    $org$opengamestudio.srvShouldResetResponse = srvShouldResetResponse;
    $org$opengamestudio.srvShouldWriteFile = srvShouldWriteFile;
  }
  $jsExportAll$(_);
  kotlin_kotlin.$jsExportAll$(_);
  //endregion
  return _;
}));
