// pskov/ver-browser/kotlin-kotlin-stdlib.js
//region block: polyfills
(function () {
  if (typeof globalThis === 'object')
    return;
  Object.defineProperty(Object.prototype, '__magic__', {get: function () {
    return this;
  }, configurable: true});
  __magic__.globalThis = __magic__;
  delete Object.prototype.__magic__;
}());
if (typeof Math.imul === 'undefined') {
  Math.imul = function imul(a, b) {
    return (a & 4.29490176E9) * (b & 65535) + (a & 65535) * (b | 0) | 0;
  };
}
if (typeof ArrayBuffer.isView === 'undefined') {
  ArrayBuffer.isView = function (a) {
    return a != null && a.__proto__ != null && a.__proto__.__proto__ === Int8Array.prototype.__proto__;
  };
}
if (typeof Array.prototype.fill === 'undefined') {
  // Polyfill from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill#Polyfill
  Object.defineProperty(Array.prototype, 'fill', {value: function (value) {
    // Steps 1-2.
    if (this == null) {
      throw new TypeError('this is null or not defined');
    }
    var O = Object(this); // Steps 3-5.
    var len = O.length >>> 0; // Steps 6-7.
    var start = arguments[1];
    var relativeStart = start >> 0; // Step 8.
    var k = relativeStart < 0 ? Math.max(len + relativeStart, 0) : Math.min(relativeStart, len); // Steps 9-10.
    var end = arguments[2];
    var relativeEnd = end === undefined ? len : end >> 0; // Step 11.
    var finalValue = relativeEnd < 0 ? Math.max(len + relativeEnd, 0) : Math.min(relativeEnd, len); // Step 12.
    while (k < finalValue) {
      O[k] = value;
      k++;
    }
    ; // Step 13.
    return O;
  }});
}
[Int8Array, Int16Array, Uint16Array, Int32Array, Float32Array, Float64Array].forEach(function (TypedArray) {
  if (typeof TypedArray.prototype.fill === 'undefined') {
    Object.defineProperty(TypedArray.prototype, 'fill', {value: Array.prototype.fill});
  }
});
if (typeof Math.clz32 === 'undefined') {
  Math.clz32 = function (log, LN2) {
    return function (x) {
      var asUint = x >>> 0;
      if (asUint === 0) {
        return 32;
      }
      return 31 - (log(asUint) / LN2 | 0) | 0; // the "| 0" acts like math.floor
    };
  }(Math.log, Math.LN2);
}
if (typeof String.prototype.endsWith === 'undefined') {
  Object.defineProperty(String.prototype, 'endsWith', {value: function (searchString, position) {
    var subjectString = this.toString();
    if (position === undefined || position > subjectString.length) {
      position = subjectString.length;
    }
    position -= searchString.length;
    var lastIndex = subjectString.indexOf(searchString, position);
    return lastIndex !== -1 && lastIndex === position;
  }});
}
if (typeof String.prototype.startsWith === 'undefined') {
  Object.defineProperty(String.prototype, 'startsWith', {value: function (searchString, position) {
    position = position || 0;
    return this.lastIndexOf(searchString, position) === position;
  }});
}
//endregion
(function (factory) {
  if (typeof define === 'function' && define.amd)
    define(['exports'], factory);
  else if (typeof exports === 'object')
    factory(module.exports);
  else
    globalThis['kotlin-kotlin-stdlib'] = factory(typeof globalThis['kotlin-kotlin-stdlib'] === 'undefined' ? {} : globalThis['kotlin-kotlin-stdlib']);
}(function (_) {
  'use strict';
  //region block: imports
  var imul = Math.imul;
  var isView = ArrayBuffer.isView;
  var clz32 = Math.clz32;
  //endregion
  //region block: pre-declaration
  initMetadataForInterface(CharSequence, 'CharSequence');
  initMetadataForClass(Number_0, 'Number');
  initMetadataForClass(asIterable$$inlined$Iterable$1);
  initMetadataForClass(Char, 'Char');
  initMetadataForInterface(Collection, 'Collection');
  initMetadataForInterface(KtList, 'List', VOID, VOID, [Collection]);
  initMetadataForInterface(Entry, 'Entry');
  initMetadataForCompanion(Companion);
  function asJsReadonlyMapView() {
    return createJsReadonlyMapViewFrom(this);
  }
  initMetadataForInterface(KtMap, 'Map');
  initMetadataForInterface(KtSet, 'Set', VOID, VOID, [Collection]);
  initMetadataForCompanion(Companion_0);
  initMetadataForClass(Enum, 'Enum');
  initMetadataForCompanion(Companion_1);
  initMetadataForClass(Long, 'Long', VOID, Number_0);
  initMetadataForInterface(FunctionAdapter, 'FunctionAdapter');
  initMetadataForClass(JsMapView, 'JsMapView', JsMapView, Map);
  initMetadataForInterface(Comparator, 'Comparator');
  initMetadataForObject(Unit, 'Unit');
  initMetadataForClass(AbstractCollection, 'AbstractCollection', VOID, VOID, [Collection]);
  initMetadataForClass(AbstractMutableCollection, 'AbstractMutableCollection', VOID, AbstractCollection, [AbstractCollection, Collection]);
  initMetadataForClass(IteratorImpl, 'IteratorImpl');
  initMetadataForClass(AbstractMutableList, 'AbstractMutableList', VOID, AbstractMutableCollection, [AbstractMutableCollection, KtList, Collection]);
  initMetadataForClass(AbstractMap, 'AbstractMap', VOID, VOID, [KtMap]);
  initMetadataForClass(AbstractMutableMap, 'AbstractMutableMap', VOID, AbstractMap, [AbstractMap, KtMap]);
  initMetadataForClass(AbstractMutableSet, 'AbstractMutableSet', VOID, AbstractMutableCollection, [AbstractMutableCollection, Collection, KtSet]);
  initMetadataForCompanion(Companion_2);
  initMetadataForClass(ArrayList, 'ArrayList', ArrayList_init_$Create$, AbstractMutableList, [AbstractMutableList, KtList, Collection]);
  initMetadataForClass(HashMap, 'HashMap', HashMap_init_$Create$, AbstractMutableMap, [AbstractMutableMap, KtMap]);
  initMetadataForClass(HashMapKeys, 'HashMapKeys', VOID, AbstractMutableSet, [Collection, KtSet, AbstractMutableSet]);
  initMetadataForClass(HashMapValues, 'HashMapValues', VOID, AbstractMutableCollection, [Collection, AbstractMutableCollection]);
  initMetadataForClass(HashMapEntrySetBase, 'HashMapEntrySetBase', VOID, AbstractMutableSet, [Collection, KtSet, AbstractMutableSet]);
  initMetadataForClass(HashMapEntrySet, 'HashMapEntrySet', VOID, HashMapEntrySetBase);
  initMetadataForClass(HashMapKeysDefault$iterator$1);
  initMetadataForClass(HashMapKeysDefault, 'HashMapKeysDefault', VOID, AbstractMutableSet);
  initMetadataForClass(HashMapValuesDefault$iterator$1);
  initMetadataForClass(HashMapValuesDefault, 'HashMapValuesDefault', VOID, AbstractMutableCollection);
  initMetadataForCompanion(Companion_3);
  initMetadataForClass(Itr, 'Itr');
  initMetadataForClass(KeysItr, 'KeysItr', VOID, Itr);
  initMetadataForClass(ValuesItr, 'ValuesItr', VOID, Itr);
  initMetadataForClass(EntriesItr, 'EntriesItr', VOID, Itr);
  initMetadataForClass(EntryRef, 'EntryRef', VOID, VOID, [Entry]);
  function containsAllEntries(m) {
    var tmp$ret$0;
    $l$block_0: {
      // Inline function 'kotlin.collections.all' call
      var tmp;
      if (isInterface(m, Collection)) {
        tmp = m.m();
      } else {
        tmp = false;
      }
      if (tmp) {
        tmp$ret$0 = true;
        break $l$block_0;
      }
      var _iterator__ex2g4s = m.e();
      while (_iterator__ex2g4s.f()) {
        var element = _iterator__ex2g4s.g();
        // Inline function 'kotlin.js.unsafeCast' call
        // Inline function 'kotlin.js.asDynamic' call
        var entry = element;
        var tmp_0;
        if (!(entry == null) ? isInterface(entry, Entry) : false) {
          tmp_0 = this.v5(entry);
        } else {
          tmp_0 = false;
        }
        if (!tmp_0) {
          tmp$ret$0 = false;
          break $l$block_0;
        }
      }
      tmp$ret$0 = true;
    }
    return tmp$ret$0;
  }
  initMetadataForInterface(InternalMap, 'InternalMap');
  initMetadataForClass(InternalHashMap, 'InternalHashMap', InternalHashMap_init_$Create$, VOID, [InternalMap]);
  initMetadataForObject(EmptyHolder, 'EmptyHolder');
  initMetadataForClass(LinkedHashMap, 'LinkedHashMap', LinkedHashMap_init_$Create$, HashMap, [HashMap, KtMap]);
  initMetadataForClass(BaseOutput, 'BaseOutput');
  initMetadataForClass(NodeJsOutput, 'NodeJsOutput', VOID, BaseOutput);
  initMetadataForClass(BufferedOutput, 'BufferedOutput', BufferedOutput, BaseOutput);
  initMetadataForClass(BufferedOutputToConsoleLog, 'BufferedOutputToConsoleLog', BufferedOutputToConsoleLog, BufferedOutput);
  initMetadataForClass(Exception, 'Exception', Exception_init_$Create$, Error);
  initMetadataForClass(RuntimeException, 'RuntimeException', RuntimeException_init_$Create$, Exception);
  initMetadataForClass(IllegalArgumentException, 'IllegalArgumentException', IllegalArgumentException_init_$Create$, RuntimeException);
  initMetadataForClass(IllegalStateException, 'IllegalStateException', IllegalStateException_init_$Create$, RuntimeException);
  initMetadataForClass(UnsupportedOperationException, 'UnsupportedOperationException', UnsupportedOperationException_init_$Create$, RuntimeException);
  initMetadataForClass(NoSuchElementException, 'NoSuchElementException', NoSuchElementException_init_$Create$, RuntimeException);
  initMetadataForClass(IndexOutOfBoundsException, 'IndexOutOfBoundsException', IndexOutOfBoundsException_init_$Create$, RuntimeException);
  initMetadataForClass(ConcurrentModificationException, 'ConcurrentModificationException', ConcurrentModificationException_init_$Create$, RuntimeException);
  initMetadataForClass(NullPointerException, 'NullPointerException', NullPointerException_init_$Create$, RuntimeException);
  initMetadataForClass(ClassCastException, 'ClassCastException', ClassCastException_init_$Create$, RuntimeException);
  initMetadataForClass(CharacterCodingException, 'CharacterCodingException', CharacterCodingException_init_$Create$, Exception);
  initMetadataForClass(StringBuilder, 'StringBuilder', StringBuilder_init_$Create$_0, VOID, [CharSequence]);
  initMetadataForClass(sam$kotlin_Comparator$0, 'sam$kotlin_Comparator$0', VOID, VOID, [Comparator, FunctionAdapter]);
  initMetadataForCompanion(Companion_4);
  initMetadataForClass(AbstractMap$keys$1$iterator$1);
  initMetadataForClass(AbstractMap$values$1$iterator$1);
  initMetadataForCompanion(Companion_5);
  initMetadataForClass(AbstractSet, 'AbstractSet', VOID, AbstractCollection, [AbstractCollection, KtSet]);
  initMetadataForClass(AbstractMap$keys$1, VOID, VOID, AbstractSet);
  initMetadataForClass(AbstractMap$values$1, VOID, VOID, AbstractCollection);
  initMetadataForCompanion(Companion_6);
  initMetadataForObject(EmptyList, 'EmptyList', VOID, VOID, [KtList]);
  initMetadataForObject(EmptyIterator, 'EmptyIterator');
  initMetadataForObject(EmptyMap, 'EmptyMap', VOID, VOID, [KtMap]);
  initMetadataForClass(IntIterator, 'IntIterator');
  initMetadataForObject(EmptySet, 'EmptySet', VOID, VOID, [KtSet]);
  initMetadataForClass(PaddingOption, 'PaddingOption', VOID, Enum);
  initMetadataForClass(Base64, 'Base64');
  initMetadataForObject(Default, 'Default', VOID, Base64);
  initMetadataForCompanion(Companion_7);
  initMetadataForClass(IntProgression, 'IntProgression');
  initMetadataForClass(IntRange, 'IntRange', VOID, IntProgression);
  initMetadataForClass(IntProgressionIterator, 'IntProgressionIterator', VOID, IntIterator);
  initMetadataForCompanion(Companion_8);
  initMetadataForClass(DelimitedRangesSequence$iterator$1);
  initMetadataForClass(DelimitedRangesSequence, 'DelimitedRangesSequence');
  initMetadataForClass(Pair, 'Pair');
  //endregion
  function CharSequence() {
  }
  function Number_0() {
  }
  function joinToString(_this__u8e3s4, separator, prefix, postfix, limit, truncated, transform) {
    separator = separator === VOID ? ', ' : separator;
    prefix = prefix === VOID ? '' : prefix;
    postfix = postfix === VOID ? '' : postfix;
    limit = limit === VOID ? -1 : limit;
    truncated = truncated === VOID ? '...' : truncated;
    transform = transform === VOID ? null : transform;
    return joinTo(_this__u8e3s4, StringBuilder_init_$Create$_0(), separator, prefix, postfix, limit, truncated, transform).toString();
  }
  function joinTo(_this__u8e3s4, buffer, separator, prefix, postfix, limit, truncated, transform) {
    separator = separator === VOID ? ', ' : separator;
    prefix = prefix === VOID ? '' : prefix;
    postfix = postfix === VOID ? '' : postfix;
    limit = limit === VOID ? -1 : limit;
    truncated = truncated === VOID ? '...' : truncated;
    transform = transform === VOID ? null : transform;
    buffer.d(prefix);
    var count = 0;
    var inductionVariable = 0;
    var last = _this__u8e3s4.length;
    $l$loop: while (inductionVariable < last) {
      var element = _this__u8e3s4[inductionVariable];
      inductionVariable = inductionVariable + 1 | 0;
      count = count + 1 | 0;
      if (count > 1) {
        buffer.d(separator);
      }
      if (limit < 0 || count <= limit) {
        appendElement(buffer, element, transform);
      } else
        break $l$loop;
    }
    if (limit >= 0 && count > limit) {
      buffer.d(truncated);
    }
    buffer.d(postfix);
    return buffer;
  }
  function indexOf(_this__u8e3s4, element) {
    if (element == null) {
      var inductionVariable = 0;
      var last = _this__u8e3s4.length - 1 | 0;
      if (inductionVariable <= last)
        do {
          var index = inductionVariable;
          inductionVariable = inductionVariable + 1 | 0;
          if (_this__u8e3s4[index] == null) {
            return index;
          }
        }
         while (inductionVariable <= last);
    } else {
      var inductionVariable_0 = 0;
      var last_0 = _this__u8e3s4.length - 1 | 0;
      if (inductionVariable_0 <= last_0)
        do {
          var index_0 = inductionVariable_0;
          inductionVariable_0 = inductionVariable_0 + 1 | 0;
          if (equals(element, _this__u8e3s4[index_0])) {
            return index_0;
          }
        }
         while (inductionVariable_0 <= last_0);
    }
    return -1;
  }
  function joinToString_0(_this__u8e3s4, separator, prefix, postfix, limit, truncated, transform) {
    separator = separator === VOID ? ', ' : separator;
    prefix = prefix === VOID ? '' : prefix;
    postfix = postfix === VOID ? '' : postfix;
    limit = limit === VOID ? -1 : limit;
    truncated = truncated === VOID ? '...' : truncated;
    transform = transform === VOID ? null : transform;
    return joinTo_0(_this__u8e3s4, StringBuilder_init_$Create$_0(), separator, prefix, postfix, limit, truncated, transform).toString();
  }
  function joinTo_0(_this__u8e3s4, buffer, separator, prefix, postfix, limit, truncated, transform) {
    separator = separator === VOID ? ', ' : separator;
    prefix = prefix === VOID ? '' : prefix;
    postfix = postfix === VOID ? '' : postfix;
    limit = limit === VOID ? -1 : limit;
    truncated = truncated === VOID ? '...' : truncated;
    transform = transform === VOID ? null : transform;
    buffer.d(prefix);
    var count = 0;
    var _iterator__ex2g4s = _this__u8e3s4.e();
    $l$loop: while (_iterator__ex2g4s.f()) {
      var element = _iterator__ex2g4s.g();
      count = count + 1 | 0;
      if (count > 1) {
        buffer.d(separator);
      }
      if (limit < 0 || count <= limit) {
        appendElement(buffer, element, transform);
      } else
        break $l$loop;
    }
    if (limit >= 0 && count > limit) {
      buffer.d(truncated);
    }
    buffer.d(postfix);
    return buffer;
  }
  function single(_this__u8e3s4) {
    if (isInterface(_this__u8e3s4, KtList))
      return single_0(_this__u8e3s4);
    else {
      var iterator = _this__u8e3s4.e();
      if (!iterator.f())
        throw NoSuchElementException_init_$Create$_0('Collection is empty.');
      var single = iterator.g();
      if (iterator.f())
        throw IllegalArgumentException_init_$Create$_0('Collection has more than one element.');
      return single;
    }
  }
  function single_0(_this__u8e3s4) {
    var tmp;
    switch (_this__u8e3s4.h()) {
      case 0:
        throw NoSuchElementException_init_$Create$_0('List is empty.');
      case 1:
        tmp = _this__u8e3s4.i(0);
        break;
      default:
        throw IllegalArgumentException_init_$Create$_0('List has more than one element.');
    }
    return tmp;
  }
  function until(_this__u8e3s4, to) {
    if (to <= -2147483648)
      return Companion_getInstance_7().j_1;
    return numberRangeToNumber(_this__u8e3s4, to - 1 | 0);
  }
  function coerceAtMost(_this__u8e3s4, maximumValue) {
    return _this__u8e3s4 > maximumValue ? maximumValue : _this__u8e3s4;
  }
  function coerceIn(_this__u8e3s4, minimumValue, maximumValue) {
    if (minimumValue > maximumValue)
      throw IllegalArgumentException_init_$Create$_0('Cannot coerce value to an empty range: maximum ' + maximumValue + ' is less than minimum ' + minimumValue + '.');
    if (_this__u8e3s4 < minimumValue)
      return minimumValue;
    if (_this__u8e3s4 > maximumValue)
      return maximumValue;
    return _this__u8e3s4;
  }
  function coerceAtLeast(_this__u8e3s4, minimumValue) {
    return _this__u8e3s4 < minimumValue ? minimumValue : _this__u8e3s4;
  }
  function downTo(_this__u8e3s4, to) {
    return Companion_instance_8.k(_this__u8e3s4, to, -1);
  }
  function asIterable(_this__u8e3s4) {
    // Inline function 'kotlin.collections.Iterable' call
    return new asIterable$$inlined$Iterable$1(_this__u8e3s4);
  }
  function asIterable$$inlined$Iterable$1($this_asIterable) {
    this.l_1 = $this_asIterable;
  }
  protoOf(asIterable$$inlined$Iterable$1).e = function () {
    return this.l_1.e();
  };
  function take(_this__u8e3s4, n) {
    // Inline function 'kotlin.require' call
    if (!(n >= 0)) {
      var message = 'Requested character count ' + n + ' is less than zero.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    return substring(_this__u8e3s4, 0, coerceAtMost(n, _this__u8e3s4.length));
  }
  function _Char___init__impl__6a9atx(value) {
    return value;
  }
  function _get_value__a43j40($this) {
    return $this;
  }
  function _Char___init__impl__6a9atx_0(code) {
    // Inline function 'kotlin.UShort.toInt' call
    var tmp$ret$0 = _UShort___get_data__impl__g0245(code) & 65535;
    return _Char___init__impl__6a9atx(tmp$ret$0);
  }
  function Char__compareTo_impl_ypi4mb($this, other) {
    return _get_value__a43j40($this) - _get_value__a43j40(other) | 0;
  }
  function Char__toInt_impl_vasixd($this) {
    return _get_value__a43j40($this);
  }
  function toString($this) {
    // Inline function 'kotlin.js.unsafeCast' call
    return String.fromCharCode(_get_value__a43j40($this));
  }
  function Char() {
  }
  function KtList() {
  }
  function Collection() {
  }
  function Entry() {
  }
  protoOf(Companion).fromJsMap = function (map) {
    return createMapFrom(map);
  };
  function Companion() {
  }
  var Companion_instance;
  function Companion_getInstance() {
    return Companion_instance;
  }
  function KtMap() {
  }
  function KtSet() {
  }
  function Companion_0() {
  }
  var Companion_instance_0;
  function Companion_getInstance_0() {
    return Companion_instance_0;
  }
  function Enum(name, ordinal) {
    this.x_1 = name;
    this.y_1 = ordinal;
  }
  protoOf(Enum).z = function (other) {
    return compareTo(this.y_1, other.y_1);
  };
  protoOf(Enum).a1 = function (other) {
    return this.z(other instanceof Enum ? other : THROW_CCE());
  };
  protoOf(Enum).equals = function (other) {
    return this === other;
  };
  protoOf(Enum).hashCode = function () {
    return identityHashCode(this);
  };
  protoOf(Enum).toString = function () {
    return this.x_1;
  };
  function toString_0(_this__u8e3s4) {
    var tmp1_elvis_lhs = _this__u8e3s4 == null ? null : toString_1(_this__u8e3s4);
    return tmp1_elvis_lhs == null ? 'null' : tmp1_elvis_lhs;
  }
  function Companion_1() {
    Companion_instance_1 = this;
    this.b1_1 = new Long(0, -2147483648);
    this.c1_1 = new Long(-1, 2147483647);
    this.d1_1 = 8;
    this.e1_1 = 64;
  }
  var Companion_instance_1;
  function Companion_getInstance_1() {
    if (Companion_instance_1 == null)
      new Companion_1();
    return Companion_instance_1;
  }
  function Long(low, high) {
    Companion_getInstance_1();
    Number_0.call(this);
    this.f1_1 = low;
    this.g1_1 = high;
  }
  protoOf(Long).h1 = function (other) {
    return compare(this, other);
  };
  protoOf(Long).a1 = function (other) {
    return this.h1(other instanceof Long ? other : THROW_CCE());
  };
  protoOf(Long).i1 = function (other) {
    return add(this, other);
  };
  protoOf(Long).j1 = function (other) {
    return multiply(this, other);
  };
  protoOf(Long).k1 = function (other) {
    return divide(this, other);
  };
  protoOf(Long).l1 = function () {
    return this.m1().i1(new Long(1, 0));
  };
  protoOf(Long).m1 = function () {
    return new Long(~this.f1_1, ~this.g1_1);
  };
  protoOf(Long).n1 = function () {
    return this.f1_1;
  };
  protoOf(Long).o1 = function () {
    return toNumber(this);
  };
  protoOf(Long).toString = function () {
    return toStringImpl(this, 10);
  };
  protoOf(Long).equals = function (other) {
    var tmp;
    if (other instanceof Long) {
      tmp = equalsLong(this, other);
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(Long).hashCode = function () {
    return hashCode_0(this);
  };
  protoOf(Long).valueOf = function () {
    return this.o1();
  };
  function implement(interfaces) {
    var maxSize = 1;
    var masks = [];
    var inductionVariable = 0;
    var last = interfaces.length;
    while (inductionVariable < last) {
      var i = interfaces[inductionVariable];
      inductionVariable = inductionVariable + 1 | 0;
      var currentSize = maxSize;
      var tmp0_elvis_lhs = i.prototype.$imask$;
      var imask = tmp0_elvis_lhs == null ? i.$imask$ : tmp0_elvis_lhs;
      if (!(imask == null)) {
        masks.push(imask);
        currentSize = imask.length;
      }
      var iid = i.$metadata$.iid;
      var tmp;
      if (iid == null) {
        tmp = null;
      } else {
        // Inline function 'kotlin.let' call
        tmp = bitMaskWith(iid);
      }
      var iidImask = tmp;
      if (!(iidImask == null)) {
        masks.push(iidImask);
        currentSize = Math.max(currentSize, iidImask.length);
      }
      if (currentSize > maxSize) {
        maxSize = currentSize;
      }
    }
    return compositeBitMask(maxSize, masks);
  }
  function bitMaskWith(activeBit) {
    var numberIndex = activeBit >> 5;
    var intArray = new Int32Array(numberIndex + 1 | 0);
    var positionInNumber = activeBit & 31;
    var numberWithSettledBit = 1 << positionInNumber;
    intArray[numberIndex] = intArray[numberIndex] | numberWithSettledBit;
    return intArray;
  }
  function compositeBitMask(capacity, masks) {
    var tmp = 0;
    var tmp_0 = new Int32Array(capacity);
    while (tmp < capacity) {
      var tmp_1 = tmp;
      var result = 0;
      var inductionVariable = 0;
      var last = masks.length;
      while (inductionVariable < last) {
        var mask = masks[inductionVariable];
        inductionVariable = inductionVariable + 1 | 0;
        if (tmp_1 < mask.length) {
          result = result | mask[tmp_1];
        }
      }
      tmp_0[tmp_1] = result;
      tmp = tmp + 1 | 0;
    }
    return tmp_0;
  }
  function isBitSet(_this__u8e3s4, possibleActiveBit) {
    var numberIndex = possibleActiveBit >> 5;
    if (numberIndex > _this__u8e3s4.length)
      return false;
    var positionInNumber = possibleActiveBit & 31;
    var numberWithSettledBit = 1 << positionInNumber;
    return !((_this__u8e3s4[numberIndex] & numberWithSettledBit) === 0);
  }
  function FunctionAdapter() {
  }
  function get_buf() {
    _init_properties_bitUtils_kt__nfcg4k();
    return buf;
  }
  var buf;
  function get_bufFloat64() {
    _init_properties_bitUtils_kt__nfcg4k();
    return bufFloat64;
  }
  var bufFloat64;
  var bufFloat32;
  function get_bufInt32() {
    _init_properties_bitUtils_kt__nfcg4k();
    return bufInt32;
  }
  var bufInt32;
  function get_lowIndex() {
    _init_properties_bitUtils_kt__nfcg4k();
    return lowIndex;
  }
  var lowIndex;
  function get_highIndex() {
    _init_properties_bitUtils_kt__nfcg4k();
    return highIndex;
  }
  var highIndex;
  function getNumberHashCode(obj) {
    _init_properties_bitUtils_kt__nfcg4k();
    // Inline function 'kotlin.js.jsBitwiseOr' call
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    if ((obj | 0) === obj) {
      return numberToInt(obj);
    }
    get_bufFloat64()[0] = obj;
    return imul(get_bufInt32()[get_highIndex()], 31) + get_bufInt32()[get_lowIndex()] | 0;
  }
  var properties_initialized_bitUtils_kt_i2bo3e;
  function _init_properties_bitUtils_kt__nfcg4k() {
    if (!properties_initialized_bitUtils_kt_i2bo3e) {
      properties_initialized_bitUtils_kt_i2bo3e = true;
      buf = new ArrayBuffer(8);
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      bufFloat64 = new Float64Array(get_buf());
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      bufFloat32 = new Float32Array(get_buf());
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      bufInt32 = new Int32Array(get_buf());
      // Inline function 'kotlin.run' call
      get_bufFloat64()[0] = -1.0;
      lowIndex = !(get_bufInt32()[0] === 0) ? 1 : 0;
      highIndex = 1 - get_lowIndex() | 0;
    }
  }
  function charSequenceGet(a, index) {
    var tmp;
    if (isString(a)) {
      tmp = charCodeAt(a, index);
    } else {
      tmp = a.b(index);
    }
    return tmp;
  }
  function isString(a) {
    return typeof a === 'string';
  }
  function charCodeAt(_this__u8e3s4, index) {
    // Inline function 'kotlin.js.asDynamic' call
    return _this__u8e3s4.charCodeAt(index);
  }
  function charSequenceLength(a) {
    var tmp;
    if (isString(a)) {
      // Inline function 'kotlin.js.asDynamic' call
      // Inline function 'kotlin.js.unsafeCast' call
      tmp = a.length;
    } else {
      tmp = a.a();
    }
    return tmp;
  }
  function charSequenceSubSequence(a, startIndex, endIndex) {
    var tmp;
    if (isString(a)) {
      tmp = substring(a, startIndex, endIndex);
    } else {
      tmp = a.c(startIndex, endIndex);
    }
    return tmp;
  }
  function arrayToString(array) {
    return joinToString(array, ', ', '[', ']', VOID, VOID, arrayToString$lambda);
  }
  function arrayToString$lambda(it) {
    return toString_1(it);
  }
  function UNSUPPORTED_OPERATION() {
    throw UnsupportedOperationException_init_$Create$();
  }
  function createJsReadonlyMapViewFrom(map) {
    var tmp = createJsReadonlyMapViewFrom$lambda(map);
    var tmp_0 = createJsReadonlyMapViewFrom$lambda_0(map);
    var tmp_1 = createJsReadonlyMapViewFrom$lambda_1(map);
    // Inline function 'kotlin.js.asDynamic' call
    var tmp_2 = UNSUPPORTED_OPERATION$ref();
    // Inline function 'kotlin.js.asDynamic' call
    var tmp_3 = UNSUPPORTED_OPERATION$ref_0();
    // Inline function 'kotlin.js.asDynamic' call
    var tmp_4 = UNSUPPORTED_OPERATION$ref_1();
    var tmp_5 = createJsReadonlyMapViewFrom$lambda_2(map);
    var tmp_6 = createJsReadonlyMapViewFrom$lambda_3(map);
    var tmp_7 = createJsReadonlyMapViewFrom$lambda_4(map);
    return createJsMapViewWith(tmp, tmp_0, tmp_1, tmp_2, tmp_3, tmp_4, tmp_5, tmp_6, tmp_7, createJsReadonlyMapViewFrom$lambda_5);
  }
  function createJsMapViewWith(mapSize, mapGet, mapContains, mapPut, mapRemove, mapClear, keysIterator, valuesIterator, entriesIterator, forEach) {
    // Inline function 'kotlin.also' call
    var this_0 = objectCreate(protoOf(JsMapView));
    this_0[Symbol.iterator] = entriesIterator;
    defineProp(this_0, 'size', mapSize, VOID);
    var mapView = this_0;
    return Object.assign(mapView, {get: mapGet, set: function (key, value) {
      mapPut(key, value);
      return this;
    }, 'delete': mapRemove, clear: mapClear, has: mapContains, keys: keysIterator, values: valuesIterator, entries: entriesIterator, forEach: function (cb, thisArg) {
      forEach(cb, mapView, thisArg);
    }});
  }
  function createJsIteratorFrom(iterator, transform) {
    var tmp;
    if (transform === VOID) {
      tmp = createJsIteratorFrom$lambda;
    } else {
      tmp = transform;
    }
    transform = tmp;
    var iteratorNext = createJsIteratorFrom$lambda_0(iterator);
    var iteratorHasNext = createJsIteratorFrom$lambda_1(iterator);
    var jsIterator = {next: function () {
      var result = {done: !iteratorHasNext()};
      if (!result.done)
        result.value = transform(iteratorNext());
      return result;
    }};
    jsIterator[Symbol.iterator] = function () {
      return this;
    };
    return jsIterator;
  }
  function forEach(cb, collection, thisArg) {
    thisArg = thisArg === VOID ? undefined : thisArg;
    var iterator = collection.entries();
    var result = iterator.next();
    while (!result.done) {
      var value = result.value;
      // Inline function 'kotlin.js.asDynamic' call
      cb.call(thisArg, value[1], value[0], collection);
      result = iterator.next();
    }
  }
  function JsMapView() {
    Map.call(this);
  }
  function createMapFrom(map) {
    // Inline function 'kotlin.collections.buildMapInternal' call
    // Inline function 'kotlin.apply' call
    var this_0 = LinkedHashMap_init_$Create$();
    forEach(createMapFrom$lambda(this_0), map);
    return this_0.w1();
  }
  function createJsReadonlyMapViewFrom$lambda($map) {
    return function () {
      return $map.h();
    };
  }
  function createJsReadonlyMapViewFrom$lambda_0($map) {
    return function (k) {
      return $map.t(k);
    };
  }
  function createJsReadonlyMapViewFrom$lambda_1($map) {
    return function (k) {
      return $map.r(k);
    };
  }
  function UNSUPPORTED_OPERATION$ref() {
    var l = function () {
      UNSUPPORTED_OPERATION();
      return Unit_instance;
    };
    l.callableName = 'UNSUPPORTED_OPERATION';
    return l;
  }
  function UNSUPPORTED_OPERATION$ref_0() {
    var l = function () {
      UNSUPPORTED_OPERATION();
      return Unit_instance;
    };
    l.callableName = 'UNSUPPORTED_OPERATION';
    return l;
  }
  function UNSUPPORTED_OPERATION$ref_1() {
    var l = function () {
      UNSUPPORTED_OPERATION();
      return Unit_instance;
    };
    l.callableName = 'UNSUPPORTED_OPERATION';
    return l;
  }
  function createJsReadonlyMapViewFrom$lambda_2($map) {
    return function () {
      return createJsIteratorFrom($map.u().e());
    };
  }
  function createJsReadonlyMapViewFrom$lambda_3($map) {
    return function () {
      return createJsIteratorFrom($map.v().e());
    };
  }
  function createJsReadonlyMapViewFrom$lambda$lambda(it) {
    // Inline function 'kotlin.arrayOf' call
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    return [it.p(), it.q()];
  }
  function createJsReadonlyMapViewFrom$lambda_4($map) {
    return function () {
      var tmp = $map.w().e();
      return createJsIteratorFrom(tmp, createJsReadonlyMapViewFrom$lambda$lambda);
    };
  }
  function createJsReadonlyMapViewFrom$lambda_5(callback, map, thisArg) {
    forEach(callback, map, thisArg);
    return Unit_instance;
  }
  function createJsIteratorFrom$lambda(it) {
    return it;
  }
  function createJsIteratorFrom$lambda_0($iterator) {
    return function () {
      return $iterator.g();
    };
  }
  function createJsIteratorFrom$lambda_1($iterator) {
    return function () {
      return $iterator.f();
    };
  }
  function createMapFrom$lambda($$this$buildMapInternal) {
    return function (value, key, _unused_var__etf5q3) {
      $$this$buildMapInternal.x1(key, value);
      return Unit_instance;
    };
  }
  function compareTo(a, b) {
    var tmp;
    switch (typeof a) {
      case 'number':
        var tmp_0;
        if (typeof b === 'number') {
          tmp_0 = doubleCompareTo(a, b);
        } else {
          if (b instanceof Long) {
            tmp_0 = doubleCompareTo(a, b.o1());
          } else {
            tmp_0 = primitiveCompareTo(a, b);
          }
        }

        tmp = tmp_0;
        break;
      case 'string':
      case 'boolean':
        tmp = primitiveCompareTo(a, b);
        break;
      default:
        tmp = compareToDoNotIntrinsicify(a, b);
        break;
    }
    return tmp;
  }
  function doubleCompareTo(a, b) {
    var tmp;
    if (a < b) {
      tmp = -1;
    } else if (a > b) {
      tmp = 1;
    } else if (a === b) {
      var tmp_0;
      if (a !== 0) {
        tmp_0 = 0;
      } else {
        // Inline function 'kotlin.js.asDynamic' call
        var ia = 1 / a;
        var tmp_1;
        // Inline function 'kotlin.js.asDynamic' call
        if (ia === 1 / b) {
          tmp_1 = 0;
        } else {
          if (ia < 0) {
            tmp_1 = -1;
          } else {
            tmp_1 = 1;
          }
        }
        tmp_0 = tmp_1;
      }
      tmp = tmp_0;
    } else if (a !== a) {
      tmp = b !== b ? 0 : 1;
    } else {
      tmp = -1;
    }
    return tmp;
  }
  function primitiveCompareTo(a, b) {
    return a < b ? -1 : a > b ? 1 : 0;
  }
  function compareToDoNotIntrinsicify(a, b) {
    return a.a1(b);
  }
  function identityHashCode(obj) {
    return getObjectHashCode(obj);
  }
  function getObjectHashCode(obj) {
    // Inline function 'kotlin.js.jsIn' call
    if (!('kotlinHashCodeValue$' in obj)) {
      var hash = calculateRandomHash();
      var descriptor = new Object();
      descriptor.value = hash;
      descriptor.enumerable = false;
      Object.defineProperty(obj, 'kotlinHashCodeValue$', descriptor);
    }
    // Inline function 'kotlin.js.unsafeCast' call
    return obj['kotlinHashCodeValue$'];
  }
  function calculateRandomHash() {
    // Inline function 'kotlin.js.jsBitwiseOr' call
    return Math.random() * 4.294967296E9 | 0;
  }
  function objectCreate(proto) {
    proto = proto === VOID ? null : proto;
    return Object.create(proto);
  }
  function defineProp(obj, name, getter, setter) {
    return Object.defineProperty(obj, name, {configurable: true, get: getter, set: setter});
  }
  function toString_1(o) {
    var tmp;
    if (o == null) {
      tmp = 'null';
    } else if (isArrayish(o)) {
      tmp = '[...]';
    } else if (!(typeof o.toString === 'function')) {
      tmp = anyToString(o);
    } else {
      // Inline function 'kotlin.js.unsafeCast' call
      tmp = o.toString();
    }
    return tmp;
  }
  function anyToString(o) {
    return Object.prototype.toString.call(o);
  }
  function hashCode(obj) {
    if (obj == null)
      return 0;
    var typeOf = typeof obj;
    var tmp;
    switch (typeOf) {
      case 'object':
        tmp = 'function' === typeof obj.hashCode ? obj.hashCode() : getObjectHashCode(obj);
        break;
      case 'function':
        tmp = getObjectHashCode(obj);
        break;
      case 'number':
        tmp = getNumberHashCode(obj);
        break;
      case 'boolean':
        // Inline function 'kotlin.js.unsafeCast' call

        tmp = getBooleanHashCode(obj);
        break;
      case 'string':
        tmp = getStringHashCode(String(obj));
        break;
      case 'bigint':
        tmp = getBigIntHashCode(obj);
        break;
      case 'symbol':
        tmp = getSymbolHashCode(obj);
        break;
      default:
        tmp = function () {
          throw new Error('Unexpected typeof `' + typeOf + '`');
        }();
        break;
    }
    return tmp;
  }
  function getBooleanHashCode(value) {
    return value ? 1231 : 1237;
  }
  function getStringHashCode(str) {
    var hash = 0;
    var length = str.length;
    var inductionVariable = 0;
    var last = length - 1 | 0;
    if (inductionVariable <= last)
      do {
        var i = inductionVariable;
        inductionVariable = inductionVariable + 1 | 0;
        // Inline function 'kotlin.js.asDynamic' call
        var code = str.charCodeAt(i);
        hash = imul(hash, 31) + code | 0;
      }
       while (!(i === last));
    return hash;
  }
  function getBigIntHashCode(value) {
    var shiftNumber = BigInt(32);
    var MASK = BigInt(4.294967295E9);
    var bigNumber = value < 0 ? -value : value;
    var hashCode = 0;
    var signum = value < 0 ? -1 : 1;
    while (bigNumber != 0) {
      // Inline function 'kotlin.js.unsafeCast' call
      var chunk = Number(bigNumber & MASK);
      hashCode = imul(31, hashCode) + chunk | 0;
      bigNumber = bigNumber >> shiftNumber;
    }
    return imul(hashCode, signum);
  }
  function getSymbolHashCode(value) {
    var hashCodeMap = symbolIsSharable(value) ? getSymbolMap() : getSymbolWeakMap();
    var cachedHashCode = hashCodeMap.get(value);
    if (cachedHashCode !== VOID)
      return cachedHashCode;
    var hash = calculateRandomHash();
    hashCodeMap.set(value, hash);
    return hash;
  }
  function symbolIsSharable(symbol) {
    return Symbol.keyFor(symbol) != VOID;
  }
  function getSymbolMap() {
    if (symbolMap === VOID) {
      symbolMap = new Map();
    }
    return symbolMap;
  }
  function getSymbolWeakMap() {
    if (symbolWeakMap === VOID) {
      symbolWeakMap = new WeakMap();
    }
    return symbolWeakMap;
  }
  var symbolMap;
  var symbolWeakMap;
  function equals(obj1, obj2) {
    if (obj1 == null) {
      return obj2 == null;
    }
    if (obj2 == null) {
      return false;
    }
    if (typeof obj1 === 'object' && typeof obj1.equals === 'function') {
      return obj1.equals(obj2);
    }
    if (obj1 !== obj1) {
      return obj2 !== obj2;
    }
    if (typeof obj1 === 'number' && typeof obj2 === 'number') {
      var tmp;
      if (obj1 === obj2) {
        var tmp_0;
        if (obj1 !== 0) {
          tmp_0 = true;
        } else {
          // Inline function 'kotlin.js.asDynamic' call
          var tmp_1 = 1 / obj1;
          // Inline function 'kotlin.js.asDynamic' call
          tmp_0 = tmp_1 === 1 / obj2;
        }
        tmp = tmp_0;
      } else {
        tmp = false;
      }
      return tmp;
    }
    return obj1 === obj2;
  }
  function unboxIntrinsic(x) {
    var message = 'Should be lowered';
    throw IllegalStateException_init_$Create$_0(toString_1(message));
  }
  function captureStack(instance, constructorFunction) {
    if (Error.captureStackTrace != null) {
      Error.captureStackTrace(instance, constructorFunction);
    } else {
      // Inline function 'kotlin.js.asDynamic' call
      instance.stack = (new Error()).stack;
    }
  }
  function protoOf(constructor) {
    return constructor.prototype;
  }
  function defineMessage(message, cause) {
    var tmp;
    if (isUndefined(message)) {
      var tmp_0;
      if (isUndefined(cause)) {
        tmp_0 = message;
      } else {
        var tmp1_elvis_lhs = cause == null ? null : cause.toString();
        tmp_0 = tmp1_elvis_lhs == null ? VOID : tmp1_elvis_lhs;
      }
      tmp = tmp_0;
    } else {
      tmp = message == null ? VOID : message;
    }
    return tmp;
  }
  function isUndefined(value) {
    return value === VOID;
  }
  function extendThrowable(this_, message, cause) {
    defineFieldOnInstance(this_, 'message', defineMessage(message, cause));
    defineFieldOnInstance(this_, 'cause', cause);
    defineFieldOnInstance(this_, 'name', Object.getPrototypeOf(this_).constructor.name);
  }
  function defineFieldOnInstance(this_, name, value) {
    Object.defineProperty(this_, name, {configurable: true, writable: true, value: value});
  }
  function ensureNotNull(v) {
    var tmp;
    if (v == null) {
      THROW_NPE();
    } else {
      tmp = v;
    }
    return tmp;
  }
  function THROW_NPE() {
    throw NullPointerException_init_$Create$();
  }
  function THROW_CCE() {
    throw ClassCastException_init_$Create$();
  }
  function get_ZERO() {
    _init_properties_longJs_kt__elc2w5();
    return ZERO;
  }
  var ZERO;
  function get_ONE() {
    _init_properties_longJs_kt__elc2w5();
    return ONE;
  }
  var ONE;
  function get_NEG_ONE() {
    _init_properties_longJs_kt__elc2w5();
    return NEG_ONE;
  }
  var NEG_ONE;
  function get_MAX_VALUE() {
    _init_properties_longJs_kt__elc2w5();
    return MAX_VALUE;
  }
  var MAX_VALUE;
  function get_MIN_VALUE() {
    _init_properties_longJs_kt__elc2w5();
    return MIN_VALUE;
  }
  var MIN_VALUE;
  function get_TWO_PWR_24_() {
    _init_properties_longJs_kt__elc2w5();
    return TWO_PWR_24_;
  }
  var TWO_PWR_24_;
  function compare(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    if (equalsLong(_this__u8e3s4, other)) {
      return 0;
    }
    var thisNeg = isNegative(_this__u8e3s4);
    var otherNeg = isNegative(other);
    return thisNeg && !otherNeg ? -1 : !thisNeg && otherNeg ? 1 : isNegative(subtract(_this__u8e3s4, other)) ? -1 : 1;
  }
  function add(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    var a48 = _this__u8e3s4.g1_1 >>> 16 | 0;
    var a32 = _this__u8e3s4.g1_1 & 65535;
    var a16 = _this__u8e3s4.f1_1 >>> 16 | 0;
    var a00 = _this__u8e3s4.f1_1 & 65535;
    var b48 = other.g1_1 >>> 16 | 0;
    var b32 = other.g1_1 & 65535;
    var b16 = other.f1_1 >>> 16 | 0;
    var b00 = other.f1_1 & 65535;
    var c48 = 0;
    var c32 = 0;
    var c16 = 0;
    var c00 = 0;
    c00 = c00 + (a00 + b00 | 0) | 0;
    c16 = c16 + (c00 >>> 16 | 0) | 0;
    c00 = c00 & 65535;
    c16 = c16 + (a16 + b16 | 0) | 0;
    c32 = c32 + (c16 >>> 16 | 0) | 0;
    c16 = c16 & 65535;
    c32 = c32 + (a32 + b32 | 0) | 0;
    c48 = c48 + (c32 >>> 16 | 0) | 0;
    c32 = c32 & 65535;
    c48 = c48 + (a48 + b48 | 0) | 0;
    c48 = c48 & 65535;
    return new Long(c16 << 16 | c00, c48 << 16 | c32);
  }
  function subtract(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    return add(_this__u8e3s4, other.l1());
  }
  function multiply(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    if (isZero(_this__u8e3s4)) {
      return get_ZERO();
    } else if (isZero(other)) {
      return get_ZERO();
    }
    if (equalsLong(_this__u8e3s4, get_MIN_VALUE())) {
      return isOdd(other) ? get_MIN_VALUE() : get_ZERO();
    } else if (equalsLong(other, get_MIN_VALUE())) {
      return isOdd(_this__u8e3s4) ? get_MIN_VALUE() : get_ZERO();
    }
    if (isNegative(_this__u8e3s4)) {
      var tmp;
      if (isNegative(other)) {
        tmp = multiply(negate(_this__u8e3s4), negate(other));
      } else {
        tmp = negate(multiply(negate(_this__u8e3s4), other));
      }
      return tmp;
    } else if (isNegative(other)) {
      return negate(multiply(_this__u8e3s4, negate(other)));
    }
    if (lessThan(_this__u8e3s4, get_TWO_PWR_24_()) && lessThan(other, get_TWO_PWR_24_())) {
      return fromNumber(toNumber(_this__u8e3s4) * toNumber(other));
    }
    var a48 = _this__u8e3s4.g1_1 >>> 16 | 0;
    var a32 = _this__u8e3s4.g1_1 & 65535;
    var a16 = _this__u8e3s4.f1_1 >>> 16 | 0;
    var a00 = _this__u8e3s4.f1_1 & 65535;
    var b48 = other.g1_1 >>> 16 | 0;
    var b32 = other.g1_1 & 65535;
    var b16 = other.f1_1 >>> 16 | 0;
    var b00 = other.f1_1 & 65535;
    var c48 = 0;
    var c32 = 0;
    var c16 = 0;
    var c00 = 0;
    c00 = c00 + imul(a00, b00) | 0;
    c16 = c16 + (c00 >>> 16 | 0) | 0;
    c00 = c00 & 65535;
    c16 = c16 + imul(a16, b00) | 0;
    c32 = c32 + (c16 >>> 16 | 0) | 0;
    c16 = c16 & 65535;
    c16 = c16 + imul(a00, b16) | 0;
    c32 = c32 + (c16 >>> 16 | 0) | 0;
    c16 = c16 & 65535;
    c32 = c32 + imul(a32, b00) | 0;
    c48 = c48 + (c32 >>> 16 | 0) | 0;
    c32 = c32 & 65535;
    c32 = c32 + imul(a16, b16) | 0;
    c48 = c48 + (c32 >>> 16 | 0) | 0;
    c32 = c32 & 65535;
    c32 = c32 + imul(a00, b32) | 0;
    c48 = c48 + (c32 >>> 16 | 0) | 0;
    c32 = c32 & 65535;
    c48 = c48 + (((imul(a48, b00) + imul(a32, b16) | 0) + imul(a16, b32) | 0) + imul(a00, b48) | 0) | 0;
    c48 = c48 & 65535;
    return new Long(c16 << 16 | c00, c48 << 16 | c32);
  }
  function divide(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    if (isZero(other)) {
      throw Exception_init_$Create$_0('division by zero');
    } else if (isZero(_this__u8e3s4)) {
      return get_ZERO();
    }
    if (equalsLong(_this__u8e3s4, get_MIN_VALUE())) {
      if (equalsLong(other, get_ONE()) || equalsLong(other, get_NEG_ONE())) {
        return get_MIN_VALUE();
      } else if (equalsLong(other, get_MIN_VALUE())) {
        return get_ONE();
      } else {
        var halfThis = shiftRight(_this__u8e3s4, 1);
        var approx = shiftLeft(halfThis.k1(other), 1);
        if (equalsLong(approx, get_ZERO())) {
          return isNegative(other) ? get_ONE() : get_NEG_ONE();
        } else {
          var rem = subtract(_this__u8e3s4, multiply(other, approx));
          return add(approx, rem.k1(other));
        }
      }
    } else if (equalsLong(other, get_MIN_VALUE())) {
      return get_ZERO();
    }
    if (isNegative(_this__u8e3s4)) {
      var tmp;
      if (isNegative(other)) {
        tmp = negate(_this__u8e3s4).k1(negate(other));
      } else {
        tmp = negate(negate(_this__u8e3s4).k1(other));
      }
      return tmp;
    } else if (isNegative(other)) {
      return negate(_this__u8e3s4.k1(negate(other)));
    }
    var res = get_ZERO();
    var rem_0 = _this__u8e3s4;
    while (greaterThanOrEqual(rem_0, other)) {
      var approxDouble = toNumber(rem_0) / toNumber(other);
      var approx2 = Math.max(1.0, Math.floor(approxDouble));
      var log2 = Math.ceil(Math.log(approx2) / Math.LN2);
      var delta = log2 <= 48 ? 1.0 : Math.pow(2.0, log2 - 48);
      var approxRes = fromNumber(approx2);
      var approxRem = multiply(approxRes, other);
      while (isNegative(approxRem) || greaterThan(approxRem, rem_0)) {
        approx2 = approx2 - delta;
        approxRes = fromNumber(approx2);
        approxRem = multiply(approxRes, other);
      }
      if (isZero(approxRes)) {
        approxRes = get_ONE();
      }
      res = add(res, approxRes);
      rem_0 = subtract(rem_0, approxRem);
    }
    return res;
  }
  function shiftLeft(_this__u8e3s4, numBits) {
    _init_properties_longJs_kt__elc2w5();
    var numBits_0 = numBits & 63;
    if (numBits_0 === 0) {
      return _this__u8e3s4;
    } else {
      if (numBits_0 < 32) {
        return new Long(_this__u8e3s4.f1_1 << numBits_0, _this__u8e3s4.g1_1 << numBits_0 | (_this__u8e3s4.f1_1 >>> (32 - numBits_0 | 0) | 0));
      } else {
        return new Long(0, _this__u8e3s4.f1_1 << (numBits_0 - 32 | 0));
      }
    }
  }
  function shiftRight(_this__u8e3s4, numBits) {
    _init_properties_longJs_kt__elc2w5();
    var numBits_0 = numBits & 63;
    if (numBits_0 === 0) {
      return _this__u8e3s4;
    } else {
      if (numBits_0 < 32) {
        return new Long(_this__u8e3s4.f1_1 >>> numBits_0 | 0 | _this__u8e3s4.g1_1 << (32 - numBits_0 | 0), _this__u8e3s4.g1_1 >> numBits_0);
      } else {
        return new Long(_this__u8e3s4.g1_1 >> (numBits_0 - 32 | 0), _this__u8e3s4.g1_1 >= 0 ? 0 : -1);
      }
    }
  }
  function toNumber(_this__u8e3s4) {
    _init_properties_longJs_kt__elc2w5();
    return _this__u8e3s4.g1_1 * 4.294967296E9 + getLowBitsUnsigned(_this__u8e3s4);
  }
  function toStringImpl(_this__u8e3s4, radix) {
    _init_properties_longJs_kt__elc2w5();
    if (radix < 2 || 36 < radix) {
      throw Exception_init_$Create$_0('radix out of range: ' + radix);
    }
    if (isZero(_this__u8e3s4)) {
      return '0';
    }
    if (isNegative(_this__u8e3s4)) {
      if (equalsLong(_this__u8e3s4, get_MIN_VALUE())) {
        var radixLong = fromInt(radix);
        var div = _this__u8e3s4.k1(radixLong);
        var rem = subtract(multiply(div, radixLong), _this__u8e3s4).n1();
        var tmp = toStringImpl(div, radix);
        // Inline function 'kotlin.js.asDynamic' call
        // Inline function 'kotlin.js.unsafeCast' call
        return tmp + rem.toString(radix);
      } else {
        return '-' + toStringImpl(negate(_this__u8e3s4), radix);
      }
    }
    var digitsPerTime = radix === 2 ? 31 : radix <= 10 ? 9 : radix <= 21 ? 7 : radix <= 35 ? 6 : 5;
    var radixToPower = fromNumber(Math.pow(radix, digitsPerTime));
    var rem_0 = _this__u8e3s4;
    var result = '';
    while (true) {
      var remDiv = rem_0.k1(radixToPower);
      var intval = subtract(rem_0, multiply(remDiv, radixToPower)).n1();
      // Inline function 'kotlin.js.asDynamic' call
      // Inline function 'kotlin.js.unsafeCast' call
      var digits = intval.toString(radix);
      rem_0 = remDiv;
      if (isZero(rem_0)) {
        return digits + result;
      } else {
        while (digits.length < digitsPerTime) {
          digits = '0' + digits;
        }
        result = digits + result;
      }
    }
  }
  function equalsLong(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    return _this__u8e3s4.g1_1 === other.g1_1 && _this__u8e3s4.f1_1 === other.f1_1;
  }
  function hashCode_0(l) {
    _init_properties_longJs_kt__elc2w5();
    return l.f1_1 ^ l.g1_1;
  }
  function fromInt(value) {
    _init_properties_longJs_kt__elc2w5();
    return new Long(value, value < 0 ? -1 : 0);
  }
  function isNegative(_this__u8e3s4) {
    _init_properties_longJs_kt__elc2w5();
    return _this__u8e3s4.g1_1 < 0;
  }
  function isZero(_this__u8e3s4) {
    _init_properties_longJs_kt__elc2w5();
    return _this__u8e3s4.g1_1 === 0 && _this__u8e3s4.f1_1 === 0;
  }
  function isOdd(_this__u8e3s4) {
    _init_properties_longJs_kt__elc2w5();
    return (_this__u8e3s4.f1_1 & 1) === 1;
  }
  function negate(_this__u8e3s4) {
    _init_properties_longJs_kt__elc2w5();
    return _this__u8e3s4.l1();
  }
  function lessThan(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    return compare(_this__u8e3s4, other) < 0;
  }
  function fromNumber(value) {
    _init_properties_longJs_kt__elc2w5();
    if (isNaN_0(value)) {
      return get_ZERO();
    } else if (value <= -9.223372036854776E18) {
      return get_MIN_VALUE();
    } else if (value + 1 >= 9.223372036854776E18) {
      return get_MAX_VALUE();
    } else if (value < 0) {
      return negate(fromNumber(-value));
    } else {
      var twoPwr32 = 4.294967296E9;
      // Inline function 'kotlin.js.jsBitwiseOr' call
      var tmp = value % twoPwr32 | 0;
      // Inline function 'kotlin.js.jsBitwiseOr' call
      var tmp$ret$1 = value / twoPwr32 | 0;
      return new Long(tmp, tmp$ret$1);
    }
  }
  function greaterThan(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    return compare(_this__u8e3s4, other) > 0;
  }
  function greaterThanOrEqual(_this__u8e3s4, other) {
    _init_properties_longJs_kt__elc2w5();
    return compare(_this__u8e3s4, other) >= 0;
  }
  function getLowBitsUnsigned(_this__u8e3s4) {
    _init_properties_longJs_kt__elc2w5();
    return _this__u8e3s4.f1_1 >= 0 ? _this__u8e3s4.f1_1 : 4.294967296E9 + _this__u8e3s4.f1_1;
  }
  var properties_initialized_longJs_kt_4syf89;
  function _init_properties_longJs_kt__elc2w5() {
    if (!properties_initialized_longJs_kt_4syf89) {
      properties_initialized_longJs_kt_4syf89 = true;
      ZERO = fromInt(0);
      ONE = fromInt(1);
      NEG_ONE = fromInt(-1);
      MAX_VALUE = new Long(-1, 2147483647);
      MIN_VALUE = new Long(0, -2147483648);
      TWO_PWR_24_ = fromInt(16777216);
    }
  }
  function createMetadata(kind, name, defaultConstructor, associatedObjectKey, associatedObjects, suspendArity) {
    var undef = VOID;
    var iid = kind === 'interface' ? generateInterfaceId() : VOID;
    return {kind: kind, simpleName: name, associatedObjectKey: associatedObjectKey, associatedObjects: associatedObjects, suspendArity: suspendArity, $kClass$: undef, defaultConstructor: defaultConstructor, iid: iid};
  }
  function generateInterfaceId() {
    if (globalInterfaceId === VOID) {
      globalInterfaceId = 0;
    }
    // Inline function 'kotlin.js.unsafeCast' call
    globalInterfaceId = globalInterfaceId + 1 | 0;
    // Inline function 'kotlin.js.unsafeCast' call
    return globalInterfaceId;
  }
  var globalInterfaceId;
  function initMetadataFor(kind, ctor, name, defaultConstructor, parent, interfaces, suspendArity, associatedObjectKey, associatedObjects) {
    if (!(parent == null)) {
      ctor.prototype = Object.create(parent.prototype);
      ctor.prototype.constructor = ctor;
    }
    var metadata = createMetadata(kind, name, defaultConstructor, associatedObjectKey, associatedObjects, suspendArity);
    ctor.$metadata$ = metadata;
    if (!(interfaces == null)) {
      var receiver = !equals(metadata.iid, VOID) ? ctor : ctor.prototype;
      receiver.$imask$ = implement(interfaces);
    }
  }
  function initMetadataForClass(ctor, name, defaultConstructor, parent, interfaces, suspendArity, associatedObjectKey, associatedObjects) {
    var kind = 'class';
    initMetadataFor(kind, ctor, name, defaultConstructor, parent, interfaces, suspendArity, associatedObjectKey, associatedObjects);
  }
  function initMetadataForObject(ctor, name, defaultConstructor, parent, interfaces, suspendArity, associatedObjectKey, associatedObjects) {
    var kind = 'object';
    initMetadataFor(kind, ctor, name, defaultConstructor, parent, interfaces, suspendArity, associatedObjectKey, associatedObjects);
  }
  function initMetadataForInterface(ctor, name, defaultConstructor, parent, interfaces, suspendArity, associatedObjectKey, associatedObjects) {
    var kind = 'interface';
    initMetadataFor(kind, ctor, name, defaultConstructor, parent, interfaces, suspendArity, associatedObjectKey, associatedObjects);
  }
  function initMetadataForLambda(ctor, parent, interfaces, suspendArity) {
    initMetadataForClass(ctor, 'Lambda', VOID, parent, interfaces, suspendArity, VOID, VOID);
  }
  function initMetadataForCoroutine(ctor, parent, interfaces, suspendArity) {
    initMetadataForClass(ctor, 'Coroutine', VOID, parent, interfaces, suspendArity, VOID, VOID);
  }
  function initMetadataForFunctionReference(ctor, parent, interfaces, suspendArity) {
    initMetadataForClass(ctor, 'FunctionReference', VOID, parent, interfaces, suspendArity, VOID, VOID);
  }
  function initMetadataForCompanion(ctor, parent, interfaces, suspendArity) {
    initMetadataForObject(ctor, 'Companion', VOID, parent, interfaces, suspendArity, VOID, VOID);
  }
  function toByte(a) {
    // Inline function 'kotlin.js.unsafeCast' call
    return a << 24 >> 24;
  }
  function numberToInt(a) {
    var tmp;
    if (a instanceof Long) {
      tmp = a.n1();
    } else {
      tmp = doubleToInt(a);
    }
    return tmp;
  }
  function doubleToInt(a) {
    var tmp;
    if (a > 2147483647) {
      tmp = 2147483647;
    } else if (a < -2147483648) {
      tmp = -2147483648;
    } else {
      // Inline function 'kotlin.js.jsBitwiseOr' call
      tmp = a | 0;
    }
    return tmp;
  }
  function toShort(a) {
    // Inline function 'kotlin.js.unsafeCast' call
    return a << 16 >> 16;
  }
  function numberToChar(a) {
    // Inline function 'kotlin.toUShort' call
    var this_0 = numberToInt(a);
    var tmp$ret$0 = _UShort___init__impl__jigrne(toShort(this_0));
    return _Char___init__impl__6a9atx_0(tmp$ret$0);
  }
  function toLong(a) {
    return fromInt(a);
  }
  function numberRangeToNumber(start, endInclusive) {
    return new IntRange(start, endInclusive);
  }
  function isArrayish(o) {
    return isJsArray(o) || isView(o);
  }
  function isJsArray(obj) {
    // Inline function 'kotlin.js.unsafeCast' call
    return Array.isArray(obj);
  }
  function isInterface(obj, iface) {
    return isInterfaceImpl(obj, iface.$metadata$.iid);
  }
  function isInterfaceImpl(obj, iface) {
    // Inline function 'kotlin.js.unsafeCast' call
    var tmp0_elvis_lhs = obj.$imask$;
    var tmp;
    if (tmp0_elvis_lhs == null) {
      return false;
    } else {
      tmp = tmp0_elvis_lhs;
    }
    var mask = tmp;
    return isBitSet(mask, iface);
  }
  function isArray(obj) {
    var tmp;
    if (isJsArray(obj)) {
      // Inline function 'kotlin.js.asDynamic' call
      tmp = !obj.$type$;
    } else {
      tmp = false;
    }
    return tmp;
  }
  function isCharSequence(value) {
    return typeof value === 'string' || isInterface(value, CharSequence);
  }
  function get_VOID() {
    _init_properties_void_kt__3zg9as();
    return VOID;
  }
  var VOID;
  var properties_initialized_void_kt_e4ret2;
  function _init_properties_void_kt__3zg9as() {
    if (!properties_initialized_void_kt_e4ret2) {
      properties_initialized_void_kt_e4ret2 = true;
      VOID = void 0;
    }
  }
  function fill(_this__u8e3s4, element, fromIndex, toIndex) {
    fromIndex = fromIndex === VOID ? 0 : fromIndex;
    toIndex = toIndex === VOID ? _this__u8e3s4.length : toIndex;
    Companion_instance_4.z1(fromIndex, toIndex, _this__u8e3s4.length);
    // Inline function 'kotlin.js.nativeFill' call
    // Inline function 'kotlin.js.asDynamic' call
    _this__u8e3s4.fill(element, fromIndex, toIndex);
  }
  function asList(_this__u8e3s4) {
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    return new ArrayList(_this__u8e3s4);
  }
  function copyOf(_this__u8e3s4, newSize) {
    // Inline function 'kotlin.require' call
    if (!(newSize >= 0)) {
      var message = 'Invalid new array size: ' + newSize + '.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    return fillFrom(_this__u8e3s4, new Int8Array(newSize));
  }
  function copyOf_0(_this__u8e3s4, newSize) {
    // Inline function 'kotlin.require' call
    if (!(newSize >= 0)) {
      var message = 'Invalid new array size: ' + newSize + '.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    return fillFrom(_this__u8e3s4, new Int32Array(newSize));
  }
  function copyOf_1(_this__u8e3s4, newSize) {
    // Inline function 'kotlin.require' call
    if (!(newSize >= 0)) {
      var message = 'Invalid new array size: ' + newSize + '.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    return arrayCopyResize(_this__u8e3s4, newSize, null);
  }
  function Comparator() {
  }
  function isNaN_0(_this__u8e3s4) {
    return !(_this__u8e3s4 === _this__u8e3s4);
  }
  function takeHighestOneBit(_this__u8e3s4) {
    var tmp;
    if (_this__u8e3s4 === 0) {
      tmp = 0;
    } else {
      // Inline function 'kotlin.countLeadingZeroBits' call
      tmp = 1 << (31 - clz32(_this__u8e3s4) | 0);
    }
    return tmp;
  }
  function Unit() {
  }
  protoOf(Unit).toString = function () {
    return 'kotlin.Unit';
  };
  var Unit_instance;
  function Unit_getInstance() {
    return Unit_instance;
  }
  function collectionToArray(collection) {
    return collectionToArrayCommonImpl(collection);
  }
  function listOf(element) {
    // Inline function 'kotlin.arrayOf' call
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    var tmp$ret$2 = [element];
    return new ArrayList(tmp$ret$2);
  }
  function copyToArray(collection) {
    var tmp;
    // Inline function 'kotlin.js.asDynamic' call
    if (collection.toArray !== undefined) {
      // Inline function 'kotlin.js.asDynamic' call
      // Inline function 'kotlin.js.unsafeCast' call
      tmp = collection.toArray();
    } else {
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp = collectionToArray(collection);
    }
    return tmp;
  }
  function AbstractMutableCollection() {
    AbstractCollection.call(this);
  }
  protoOf(AbstractMutableCollection).toJSON = function () {
    return this.toArray();
  };
  protoOf(AbstractMutableCollection).b2 = function () {
  };
  function IteratorImpl($outer) {
    this.e2_1 = $outer;
    this.c2_1 = 0;
    this.d2_1 = -1;
  }
  protoOf(IteratorImpl).f = function () {
    return this.c2_1 < this.e2_1.h();
  };
  protoOf(IteratorImpl).g = function () {
    if (!this.f())
      throw NoSuchElementException_init_$Create$();
    var tmp = this;
    var _unary__edvuaz = this.c2_1;
    this.c2_1 = _unary__edvuaz + 1 | 0;
    tmp.d2_1 = _unary__edvuaz;
    return this.e2_1.i(this.d2_1);
  };
  function AbstractMutableList() {
    AbstractMutableCollection.call(this);
    this.f2_1 = 0;
  }
  protoOf(AbstractMutableList).a2 = function (element) {
    this.b2();
    this.g2(this.h(), element);
    return true;
  };
  protoOf(AbstractMutableList).e = function () {
    return new IteratorImpl(this);
  };
  protoOf(AbstractMutableList).n = function (element) {
    return this.i2(element) >= 0;
  };
  protoOf(AbstractMutableList).i2 = function (element) {
    var tmp$ret$1;
    $l$block: {
      // Inline function 'kotlin.collections.indexOfFirst' call
      var index = 0;
      var _iterator__ex2g4s = this.e();
      while (_iterator__ex2g4s.f()) {
        var item = _iterator__ex2g4s.g();
        if (equals(item, element)) {
          tmp$ret$1 = index;
          break $l$block;
        }
        index = index + 1 | 0;
      }
      tmp$ret$1 = -1;
    }
    return tmp$ret$1;
  };
  protoOf(AbstractMutableList).equals = function (other) {
    if (other === this)
      return true;
    if (!(!(other == null) ? isInterface(other, KtList) : false))
      return false;
    return Companion_instance_4.j2(this, other);
  };
  protoOf(AbstractMutableList).hashCode = function () {
    return Companion_instance_4.k2(this);
  };
  function AbstractMutableMap() {
    AbstractMap.call(this);
    this.n2_1 = null;
    this.o2_1 = null;
  }
  protoOf(AbstractMutableMap).p2 = function () {
    return new HashMapKeysDefault(this);
  };
  protoOf(AbstractMutableMap).q2 = function () {
    return new HashMapValuesDefault(this);
  };
  protoOf(AbstractMutableMap).u = function () {
    var tmp0_elvis_lhs = this.n2_1;
    var tmp;
    if (tmp0_elvis_lhs == null) {
      // Inline function 'kotlin.also' call
      var this_0 = this.p2();
      this.n2_1 = this_0;
      tmp = this_0;
    } else {
      tmp = tmp0_elvis_lhs;
    }
    return tmp;
  };
  protoOf(AbstractMutableMap).v = function () {
    var tmp0_elvis_lhs = this.o2_1;
    var tmp;
    if (tmp0_elvis_lhs == null) {
      // Inline function 'kotlin.also' call
      var this_0 = this.q2();
      this.o2_1 = this_0;
      tmp = this_0;
    } else {
      tmp = tmp0_elvis_lhs;
    }
    return tmp;
  };
  function AbstractMutableSet() {
    AbstractMutableCollection.call(this);
  }
  protoOf(AbstractMutableSet).equals = function (other) {
    if (other === this)
      return true;
    if (!(!(other == null) ? isInterface(other, KtSet) : false))
      return false;
    return Companion_instance_6.u2(this, other);
  };
  protoOf(AbstractMutableSet).hashCode = function () {
    return Companion_instance_6.v2(this);
  };
  function arrayOfUninitializedElements(capacity) {
    // Inline function 'kotlin.require' call
    if (!(capacity >= 0)) {
      var message = 'capacity must be non-negative.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    // Inline function 'kotlin.arrayOfNulls' call
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    return Array(capacity);
  }
  function resetRange(_this__u8e3s4, fromIndex, toIndex) {
    // Inline function 'kotlin.js.nativeFill' call
    // Inline function 'kotlin.js.asDynamic' call
    _this__u8e3s4.fill(null, fromIndex, toIndex);
  }
  function copyOfUninitializedElements(_this__u8e3s4, newSize) {
    // Inline function 'kotlin.js.unsafeCast' call
    // Inline function 'kotlin.js.asDynamic' call
    return copyOf_1(_this__u8e3s4, newSize);
  }
  function Companion_2() {
    Companion_instance_2 = this;
    var tmp = this;
    // Inline function 'kotlin.also' call
    var this_0 = ArrayList_init_$Create$_0(0);
    this_0.y2_1 = true;
    tmp.z2_1 = this_0;
  }
  var Companion_instance_2;
  function Companion_getInstance_2() {
    if (Companion_instance_2 == null)
      new Companion_2();
    return Companion_instance_2;
  }
  function ArrayList_init_$Init$($this) {
    // Inline function 'kotlin.emptyArray' call
    var tmp$ret$0 = [];
    ArrayList.call($this, tmp$ret$0);
    return $this;
  }
  function ArrayList_init_$Create$() {
    return ArrayList_init_$Init$(objectCreate(protoOf(ArrayList)));
  }
  function ArrayList_init_$Init$_0(initialCapacity, $this) {
    // Inline function 'kotlin.emptyArray' call
    var tmp$ret$0 = [];
    ArrayList.call($this, tmp$ret$0);
    // Inline function 'kotlin.require' call
    if (!(initialCapacity >= 0)) {
      var message = 'Negative initial capacity: ' + initialCapacity;
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    return $this;
  }
  function ArrayList_init_$Create$_0(initialCapacity) {
    return ArrayList_init_$Init$_0(initialCapacity, objectCreate(protoOf(ArrayList)));
  }
  function rangeCheck($this, index) {
    // Inline function 'kotlin.apply' call
    Companion_instance_4.a3(index, $this.h());
    return index;
  }
  function insertionRangeCheck($this, index) {
    // Inline function 'kotlin.apply' call
    Companion_instance_4.b3(index, $this.h());
    return index;
  }
  function ArrayList(array) {
    Companion_getInstance_2();
    AbstractMutableList.call(this);
    this.x2_1 = array;
    this.y2_1 = false;
  }
  protoOf(ArrayList).h = function () {
    return this.x2_1.length;
  };
  protoOf(ArrayList).i = function (index) {
    var tmp = this.x2_1[rangeCheck(this, index)];
    return (tmp == null ? true : !(tmp == null)) ? tmp : THROW_CCE();
  };
  protoOf(ArrayList).a2 = function (element) {
    this.b2();
    // Inline function 'kotlin.js.asDynamic' call
    this.x2_1.push(element);
    this.f2_1 = this.f2_1 + 1 | 0;
    return true;
  };
  protoOf(ArrayList).g2 = function (index, element) {
    this.b2();
    // Inline function 'kotlin.js.asDynamic' call
    this.x2_1.splice(insertionRangeCheck(this, index), 0, element);
    this.f2_1 = this.f2_1 + 1 | 0;
  };
  protoOf(ArrayList).h2 = function (index) {
    this.b2();
    rangeCheck(this, index);
    this.f2_1 = this.f2_1 + 1 | 0;
    var tmp;
    if (index === get_lastIndex(this)) {
      // Inline function 'kotlin.js.asDynamic' call
      tmp = this.x2_1.pop();
    } else {
      // Inline function 'kotlin.js.asDynamic' call
      tmp = this.x2_1.splice(index, 1)[0];
    }
    return tmp;
  };
  protoOf(ArrayList).i2 = function (element) {
    return indexOf(this.x2_1, element);
  };
  protoOf(ArrayList).toString = function () {
    return arrayToString(this.x2_1);
  };
  protoOf(ArrayList).c3 = function () {
    return [].slice.call(this.x2_1);
  };
  protoOf(ArrayList).toArray = function () {
    return this.c3();
  };
  protoOf(ArrayList).b2 = function () {
    if (this.y2_1)
      throw UnsupportedOperationException_init_$Create$();
  };
  function HashMap_init_$Init$(internalMap, $this) {
    AbstractMutableMap.call($this);
    HashMap.call($this);
    $this.h3_1 = internalMap;
    return $this;
  }
  function HashMap_init_$Init$_0($this) {
    HashMap_init_$Init$(InternalHashMap_init_$Create$(), $this);
    return $this;
  }
  function HashMap_init_$Create$() {
    return HashMap_init_$Init$_0(objectCreate(protoOf(HashMap)));
  }
  function HashMap_init_$Init$_1(original, $this) {
    HashMap_init_$Init$(InternalHashMap_init_$Create$_1(original), $this);
    return $this;
  }
  protoOf(HashMap).r = function (key) {
    return this.h3_1.j3(key);
  };
  protoOf(HashMap).s = function (value) {
    return this.h3_1.s(value);
  };
  protoOf(HashMap).p2 = function () {
    return new HashMapKeys(this.h3_1);
  };
  protoOf(HashMap).q2 = function () {
    return new HashMapValues(this.h3_1);
  };
  protoOf(HashMap).w = function () {
    var tmp0_elvis_lhs = this.i3_1;
    var tmp;
    if (tmp0_elvis_lhs == null) {
      // Inline function 'kotlin.also' call
      var this_0 = new HashMapEntrySet(this.h3_1);
      this.i3_1 = this_0;
      tmp = this_0;
    } else {
      tmp = tmp0_elvis_lhs;
    }
    return tmp;
  };
  protoOf(HashMap).t = function (key) {
    return this.h3_1.t(key);
  };
  protoOf(HashMap).x1 = function (key, value) {
    return this.h3_1.x1(key, value);
  };
  protoOf(HashMap).h = function () {
    return this.h3_1.h();
  };
  function HashMap() {
    this.i3_1 = null;
  }
  function HashMapKeys(backing) {
    AbstractMutableSet.call(this);
    this.k3_1 = backing;
  }
  protoOf(HashMapKeys).h = function () {
    return this.k3_1.h();
  };
  protoOf(HashMapKeys).m = function () {
    return this.k3_1.h() === 0;
  };
  protoOf(HashMapKeys).n = function (element) {
    return this.k3_1.j3(element);
  };
  protoOf(HashMapKeys).a2 = function (element) {
    throw UnsupportedOperationException_init_$Create$();
  };
  protoOf(HashMapKeys).e = function () {
    return this.k3_1.l3();
  };
  function HashMapValues(backing) {
    AbstractMutableCollection.call(this);
    this.m3_1 = backing;
  }
  protoOf(HashMapValues).h = function () {
    return this.m3_1.h();
  };
  protoOf(HashMapValues).m = function () {
    return this.m3_1.h() === 0;
  };
  protoOf(HashMapValues).n3 = function (element) {
    return this.m3_1.s(element);
  };
  protoOf(HashMapValues).n = function (element) {
    if (!(element == null ? true : !(element == null)))
      return false;
    return this.n3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(HashMapValues).o3 = function (element) {
    throw UnsupportedOperationException_init_$Create$();
  };
  protoOf(HashMapValues).a2 = function (element) {
    return this.o3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(HashMapValues).e = function () {
    return this.m3_1.p3();
  };
  function HashMapEntrySet(backing) {
    HashMapEntrySetBase.call(this, backing);
  }
  protoOf(HashMapEntrySet).e = function () {
    return this.r3_1.s3();
  };
  function HashMapEntrySetBase(backing) {
    AbstractMutableSet.call(this);
    this.r3_1 = backing;
  }
  protoOf(HashMapEntrySetBase).h = function () {
    return this.r3_1.h();
  };
  protoOf(HashMapEntrySetBase).m = function () {
    return this.r3_1.h() === 0;
  };
  protoOf(HashMapEntrySetBase).t3 = function (element) {
    return this.r3_1.v3(element);
  };
  protoOf(HashMapEntrySetBase).n = function (element) {
    if (!(!(element == null) ? isInterface(element, Entry) : false))
      return false;
    return this.t3((!(element == null) ? isInterface(element, Entry) : false) ? element : THROW_CCE());
  };
  protoOf(HashMapEntrySetBase).u3 = function (element) {
    throw UnsupportedOperationException_init_$Create$();
  };
  protoOf(HashMapEntrySetBase).a2 = function (element) {
    return this.u3((!(element == null) ? isInterface(element, Entry) : false) ? element : THROW_CCE());
  };
  protoOf(HashMapEntrySetBase).o = function (elements) {
    return this.r3_1.w3(elements);
  };
  function HashMapKeysDefault$iterator$1($entryIterator) {
    this.x3_1 = $entryIterator;
  }
  protoOf(HashMapKeysDefault$iterator$1).f = function () {
    return this.x3_1.f();
  };
  protoOf(HashMapKeysDefault$iterator$1).g = function () {
    return this.x3_1.g().p();
  };
  function HashMapKeysDefault(backingMap) {
    AbstractMutableSet.call(this);
    this.y3_1 = backingMap;
  }
  protoOf(HashMapKeysDefault).z3 = function (element) {
    throw UnsupportedOperationException_init_$Create$_0('Add is not supported on keys');
  };
  protoOf(HashMapKeysDefault).a2 = function (element) {
    return this.z3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(HashMapKeysDefault).j3 = function (element) {
    return this.y3_1.r(element);
  };
  protoOf(HashMapKeysDefault).n = function (element) {
    if (!(element == null ? true : !(element == null)))
      return false;
    return this.j3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(HashMapKeysDefault).e = function () {
    var entryIterator = this.y3_1.w().e();
    return new HashMapKeysDefault$iterator$1(entryIterator);
  };
  protoOf(HashMapKeysDefault).h = function () {
    return this.y3_1.h();
  };
  function HashMapValuesDefault$iterator$1($entryIterator) {
    this.a4_1 = $entryIterator;
  }
  protoOf(HashMapValuesDefault$iterator$1).f = function () {
    return this.a4_1.f();
  };
  protoOf(HashMapValuesDefault$iterator$1).g = function () {
    return this.a4_1.g().q();
  };
  function HashMapValuesDefault(backingMap) {
    AbstractMutableCollection.call(this);
    this.b4_1 = backingMap;
  }
  protoOf(HashMapValuesDefault).o3 = function (element) {
    throw UnsupportedOperationException_init_$Create$_0('Add is not supported on values');
  };
  protoOf(HashMapValuesDefault).a2 = function (element) {
    return this.o3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(HashMapValuesDefault).n3 = function (element) {
    return this.b4_1.s(element);
  };
  protoOf(HashMapValuesDefault).n = function (element) {
    if (!(element == null ? true : !(element == null)))
      return false;
    return this.n3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(HashMapValuesDefault).e = function () {
    var entryIterator = this.b4_1.w().e();
    return new HashMapValuesDefault$iterator$1(entryIterator);
  };
  protoOf(HashMapValuesDefault).h = function () {
    return this.b4_1.h();
  };
  function computeHashSize($this, capacity) {
    return takeHighestOneBit(imul(coerceAtLeast(capacity, 1), 3));
  }
  function computeShift($this, hashSize) {
    // Inline function 'kotlin.countLeadingZeroBits' call
    return clz32(hashSize) + 1 | 0;
  }
  function checkForComodification($this) {
    if (!($this.m4_1.j4_1 === $this.o4_1))
      throw ConcurrentModificationException_init_$Create$_0('The backing map has been modified after this entry was obtained.');
  }
  function InternalHashMap_init_$Init$($this) {
    InternalHashMap_init_$Init$_0(8, $this);
    return $this;
  }
  function InternalHashMap_init_$Create$() {
    return InternalHashMap_init_$Init$(objectCreate(protoOf(InternalHashMap)));
  }
  function InternalHashMap_init_$Init$_0(initialCapacity, $this) {
    InternalHashMap.call($this, arrayOfUninitializedElements(initialCapacity), null, new Int32Array(initialCapacity), new Int32Array(computeHashSize(Companion_instance_3, initialCapacity)), 2, 0);
    return $this;
  }
  function InternalHashMap_init_$Create$_0(initialCapacity) {
    return InternalHashMap_init_$Init$_0(initialCapacity, objectCreate(protoOf(InternalHashMap)));
  }
  function InternalHashMap_init_$Init$_1(original, $this) {
    InternalHashMap_init_$Init$_0(original.h(), $this);
    $this.p4(original);
    return $this;
  }
  function InternalHashMap_init_$Create$_1(original) {
    return InternalHashMap_init_$Init$_1(original, objectCreate(protoOf(InternalHashMap)));
  }
  function _get_capacity__a9k9f3($this) {
    return $this.c4_1.length;
  }
  function _get_hashSize__tftcho($this) {
    return $this.f4_1.length;
  }
  function registerModification($this) {
    $this.j4_1 = $this.j4_1 + 1 | 0;
  }
  function ensureExtraCapacity($this, n) {
    if (shouldCompact($this, n)) {
      compact($this, true);
    } else {
      ensureCapacity($this, $this.h4_1 + n | 0);
    }
  }
  function shouldCompact($this, extraCapacity) {
    var spareCapacity = _get_capacity__a9k9f3($this) - $this.h4_1 | 0;
    var gaps = $this.h4_1 - $this.h() | 0;
    return spareCapacity < extraCapacity && (gaps + spareCapacity | 0) >= extraCapacity && gaps >= (_get_capacity__a9k9f3($this) / 4 | 0);
  }
  function ensureCapacity($this, minCapacity) {
    if (minCapacity < 0)
      throw RuntimeException_init_$Create$_0('too many elements');
    if (minCapacity > _get_capacity__a9k9f3($this)) {
      var newSize = Companion_instance_4.q4(_get_capacity__a9k9f3($this), minCapacity);
      $this.c4_1 = copyOfUninitializedElements($this.c4_1, newSize);
      var tmp = $this;
      var tmp0_safe_receiver = $this.d4_1;
      tmp.d4_1 = tmp0_safe_receiver == null ? null : copyOfUninitializedElements(tmp0_safe_receiver, newSize);
      $this.e4_1 = copyOf_0($this.e4_1, newSize);
      var newHashSize = computeHashSize(Companion_instance_3, newSize);
      if (newHashSize > _get_hashSize__tftcho($this)) {
        rehash($this, newHashSize);
      }
    }
  }
  function allocateValuesArray($this) {
    var curValuesArray = $this.d4_1;
    if (!(curValuesArray == null))
      return curValuesArray;
    var newValuesArray = arrayOfUninitializedElements(_get_capacity__a9k9f3($this));
    $this.d4_1 = newValuesArray;
    return newValuesArray;
  }
  function hash($this, key) {
    return key == null ? 0 : imul(hashCode(key), -1640531527) >>> $this.i4_1 | 0;
  }
  function compact($this, updateHashArray) {
    var i = 0;
    var j = 0;
    var valuesArray = $this.d4_1;
    while (i < $this.h4_1) {
      var hash = $this.e4_1[i];
      if (hash >= 0) {
        $this.c4_1[j] = $this.c4_1[i];
        if (!(valuesArray == null)) {
          valuesArray[j] = valuesArray[i];
        }
        if (updateHashArray) {
          $this.e4_1[j] = hash;
          $this.f4_1[hash] = j + 1 | 0;
        }
        j = j + 1 | 0;
      }
      i = i + 1 | 0;
    }
    resetRange($this.c4_1, j, $this.h4_1);
    if (valuesArray == null)
      null;
    else {
      resetRange(valuesArray, j, $this.h4_1);
    }
    $this.h4_1 = j;
  }
  function rehash($this, newHashSize) {
    registerModification($this);
    if ($this.h4_1 > $this.k4_1) {
      compact($this, false);
    }
    $this.f4_1 = new Int32Array(newHashSize);
    $this.i4_1 = computeShift(Companion_instance_3, newHashSize);
    var i = 0;
    while (i < $this.h4_1) {
      var _unary__edvuaz = i;
      i = _unary__edvuaz + 1 | 0;
      if (!putRehash($this, _unary__edvuaz)) {
        throw IllegalStateException_init_$Create$_0('This cannot happen with fixed magic multiplier and grow-only hash array. Have object hashCodes changed?');
      }
    }
  }
  function putRehash($this, i) {
    var hash_0 = hash($this, $this.c4_1[i]);
    var probesLeft = $this.g4_1;
    while (true) {
      var index = $this.f4_1[hash_0];
      if (index === 0) {
        $this.f4_1[hash_0] = i + 1 | 0;
        $this.e4_1[i] = hash_0;
        return true;
      }
      probesLeft = probesLeft - 1 | 0;
      if (probesLeft < 0)
        return false;
      var _unary__edvuaz = hash_0;
      hash_0 = _unary__edvuaz - 1 | 0;
      if (_unary__edvuaz === 0)
        hash_0 = _get_hashSize__tftcho($this) - 1 | 0;
    }
  }
  function findKey($this, key) {
    var hash_0 = hash($this, key);
    var probesLeft = $this.g4_1;
    while (true) {
      var index = $this.f4_1[hash_0];
      if (index === 0)
        return -1;
      if (index > 0 && equals($this.c4_1[index - 1 | 0], key))
        return index - 1 | 0;
      probesLeft = probesLeft - 1 | 0;
      if (probesLeft < 0)
        return -1;
      var _unary__edvuaz = hash_0;
      hash_0 = _unary__edvuaz - 1 | 0;
      if (_unary__edvuaz === 0)
        hash_0 = _get_hashSize__tftcho($this) - 1 | 0;
    }
  }
  function findValue($this, value) {
    var i = $this.h4_1;
    $l$loop: while (true) {
      i = i - 1 | 0;
      if (!(i >= 0)) {
        break $l$loop;
      }
      if ($this.e4_1[i] >= 0 && equals(ensureNotNull($this.d4_1)[i], value))
        return i;
    }
    return -1;
  }
  function addKey($this, key) {
    $this.r4();
    retry: while (true) {
      var hash_0 = hash($this, key);
      var tentativeMaxProbeDistance = coerceAtMost(imul($this.g4_1, 2), _get_hashSize__tftcho($this) / 2 | 0);
      var probeDistance = 0;
      while (true) {
        var index = $this.f4_1[hash_0];
        if (index <= 0) {
          if ($this.h4_1 >= _get_capacity__a9k9f3($this)) {
            ensureExtraCapacity($this, 1);
            continue retry;
          }
          var _unary__edvuaz = $this.h4_1;
          $this.h4_1 = _unary__edvuaz + 1 | 0;
          var putIndex = _unary__edvuaz;
          $this.c4_1[putIndex] = key;
          $this.e4_1[putIndex] = hash_0;
          $this.f4_1[hash_0] = putIndex + 1 | 0;
          $this.k4_1 = $this.k4_1 + 1 | 0;
          registerModification($this);
          if (probeDistance > $this.g4_1)
            $this.g4_1 = probeDistance;
          return putIndex;
        }
        if (equals($this.c4_1[index - 1 | 0], key)) {
          return -index | 0;
        }
        probeDistance = probeDistance + 1 | 0;
        if (probeDistance > tentativeMaxProbeDistance) {
          rehash($this, imul(_get_hashSize__tftcho($this), 2));
          continue retry;
        }
        var _unary__edvuaz_0 = hash_0;
        hash_0 = _unary__edvuaz_0 - 1 | 0;
        if (_unary__edvuaz_0 === 0)
          hash_0 = _get_hashSize__tftcho($this) - 1 | 0;
      }
    }
  }
  function contentEquals($this, other) {
    return $this.k4_1 === other.h() && $this.w3(other.w());
  }
  function putEntry($this, entry) {
    var index = addKey($this, entry.p());
    var valuesArray = allocateValuesArray($this);
    if (index >= 0) {
      valuesArray[index] = entry.q();
      return true;
    }
    var oldValue = valuesArray[(-index | 0) - 1 | 0];
    if (!equals(entry.q(), oldValue)) {
      valuesArray[(-index | 0) - 1 | 0] = entry.q();
      return true;
    }
    return false;
  }
  function putAllEntries($this, from) {
    if (from.m())
      return false;
    ensureExtraCapacity($this, from.h());
    var it = from.e();
    var updated = false;
    while (it.f()) {
      if (putEntry($this, it.g()))
        updated = true;
    }
    return updated;
  }
  function Companion_3() {
    this.s4_1 = -1640531527;
    this.t4_1 = 8;
    this.u4_1 = 2;
    this.v4_1 = -1;
  }
  var Companion_instance_3;
  function Companion_getInstance_3() {
    return Companion_instance_3;
  }
  function Itr(map) {
    this.w4_1 = map;
    this.x4_1 = 0;
    this.y4_1 = -1;
    this.z4_1 = this.w4_1.j4_1;
    this.a5();
  }
  protoOf(Itr).a5 = function () {
    while (this.x4_1 < this.w4_1.h4_1 && this.w4_1.e4_1[this.x4_1] < 0) {
      this.x4_1 = this.x4_1 + 1 | 0;
    }
  };
  protoOf(Itr).f = function () {
    return this.x4_1 < this.w4_1.h4_1;
  };
  protoOf(Itr).b5 = function () {
    if (!(this.w4_1.j4_1 === this.z4_1))
      throw ConcurrentModificationException_init_$Create$();
  };
  function KeysItr(map) {
    Itr.call(this, map);
  }
  protoOf(KeysItr).g = function () {
    this.b5();
    if (this.x4_1 >= this.w4_1.h4_1)
      throw NoSuchElementException_init_$Create$();
    var tmp = this;
    var _unary__edvuaz = this.x4_1;
    this.x4_1 = _unary__edvuaz + 1 | 0;
    tmp.y4_1 = _unary__edvuaz;
    var result = this.w4_1.c4_1[this.y4_1];
    this.a5();
    return result;
  };
  function ValuesItr(map) {
    Itr.call(this, map);
  }
  protoOf(ValuesItr).g = function () {
    this.b5();
    if (this.x4_1 >= this.w4_1.h4_1)
      throw NoSuchElementException_init_$Create$();
    var tmp = this;
    var _unary__edvuaz = this.x4_1;
    this.x4_1 = _unary__edvuaz + 1 | 0;
    tmp.y4_1 = _unary__edvuaz;
    var result = ensureNotNull(this.w4_1.d4_1)[this.y4_1];
    this.a5();
    return result;
  };
  function EntriesItr(map) {
    Itr.call(this, map);
  }
  protoOf(EntriesItr).g = function () {
    this.b5();
    if (this.x4_1 >= this.w4_1.h4_1)
      throw NoSuchElementException_init_$Create$();
    var tmp = this;
    var _unary__edvuaz = this.x4_1;
    this.x4_1 = _unary__edvuaz + 1 | 0;
    tmp.y4_1 = _unary__edvuaz;
    var result = new EntryRef(this.w4_1, this.y4_1);
    this.a5();
    return result;
  };
  protoOf(EntriesItr).o5 = function () {
    if (this.x4_1 >= this.w4_1.h4_1)
      throw NoSuchElementException_init_$Create$();
    var tmp = this;
    var _unary__edvuaz = this.x4_1;
    this.x4_1 = _unary__edvuaz + 1 | 0;
    tmp.y4_1 = _unary__edvuaz;
    // Inline function 'kotlin.hashCode' call
    var tmp0_safe_receiver = this.w4_1.c4_1[this.y4_1];
    var tmp1_elvis_lhs = tmp0_safe_receiver == null ? null : hashCode(tmp0_safe_receiver);
    var tmp_0 = tmp1_elvis_lhs == null ? 0 : tmp1_elvis_lhs;
    // Inline function 'kotlin.hashCode' call
    var tmp0_safe_receiver_0 = ensureNotNull(this.w4_1.d4_1)[this.y4_1];
    var tmp1_elvis_lhs_0 = tmp0_safe_receiver_0 == null ? null : hashCode(tmp0_safe_receiver_0);
    var result = tmp_0 ^ (tmp1_elvis_lhs_0 == null ? 0 : tmp1_elvis_lhs_0);
    this.a5();
    return result;
  };
  protoOf(EntriesItr).p5 = function (sb) {
    if (this.x4_1 >= this.w4_1.h4_1)
      throw NoSuchElementException_init_$Create$();
    var tmp = this;
    var _unary__edvuaz = this.x4_1;
    this.x4_1 = _unary__edvuaz + 1 | 0;
    tmp.y4_1 = _unary__edvuaz;
    var key = this.w4_1.c4_1[this.y4_1];
    if (equals(key, this.w4_1))
      sb.s5('(this Map)');
    else
      sb.r5(key);
    sb.t5(_Char___init__impl__6a9atx(61));
    var value = ensureNotNull(this.w4_1.d4_1)[this.y4_1];
    if (equals(value, this.w4_1))
      sb.s5('(this Map)');
    else
      sb.r5(value);
    this.a5();
  };
  function EntryRef(map, index) {
    this.m4_1 = map;
    this.n4_1 = index;
    this.o4_1 = this.m4_1.j4_1;
  }
  protoOf(EntryRef).p = function () {
    checkForComodification(this);
    return this.m4_1.c4_1[this.n4_1];
  };
  protoOf(EntryRef).q = function () {
    checkForComodification(this);
    return ensureNotNull(this.m4_1.d4_1)[this.n4_1];
  };
  protoOf(EntryRef).equals = function (other) {
    var tmp;
    var tmp_0;
    if (!(other == null) ? isInterface(other, Entry) : false) {
      tmp_0 = equals(other.p(), this.p());
    } else {
      tmp_0 = false;
    }
    if (tmp_0) {
      tmp = equals(other.q(), this.q());
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(EntryRef).hashCode = function () {
    // Inline function 'kotlin.hashCode' call
    var tmp0_safe_receiver = this.p();
    var tmp1_elvis_lhs = tmp0_safe_receiver == null ? null : hashCode(tmp0_safe_receiver);
    var tmp = tmp1_elvis_lhs == null ? 0 : tmp1_elvis_lhs;
    // Inline function 'kotlin.hashCode' call
    var tmp0_safe_receiver_0 = this.q();
    var tmp1_elvis_lhs_0 = tmp0_safe_receiver_0 == null ? null : hashCode(tmp0_safe_receiver_0);
    return tmp ^ (tmp1_elvis_lhs_0 == null ? 0 : tmp1_elvis_lhs_0);
  };
  protoOf(EntryRef).toString = function () {
    return toString_0(this.p()) + '=' + toString_0(this.q());
  };
  function InternalHashMap(keysArray, valuesArray, presenceArray, hashArray, maxProbeDistance, length) {
    this.c4_1 = keysArray;
    this.d4_1 = valuesArray;
    this.e4_1 = presenceArray;
    this.f4_1 = hashArray;
    this.g4_1 = maxProbeDistance;
    this.h4_1 = length;
    this.i4_1 = computeShift(Companion_instance_3, _get_hashSize__tftcho(this));
    this.j4_1 = 0;
    this.k4_1 = 0;
    this.l4_1 = false;
  }
  protoOf(InternalHashMap).h = function () {
    return this.k4_1;
  };
  protoOf(InternalHashMap).u5 = function () {
    this.r4();
    this.l4_1 = true;
  };
  protoOf(InternalHashMap).s = function (value) {
    return findValue(this, value) >= 0;
  };
  protoOf(InternalHashMap).t = function (key) {
    var index = findKey(this, key);
    if (index < 0)
      return null;
    return ensureNotNull(this.d4_1)[index];
  };
  protoOf(InternalHashMap).j3 = function (key) {
    return findKey(this, key) >= 0;
  };
  protoOf(InternalHashMap).x1 = function (key, value) {
    var index = addKey(this, key);
    var valuesArray = allocateValuesArray(this);
    if (index < 0) {
      var oldValue = valuesArray[(-index | 0) - 1 | 0];
      valuesArray[(-index | 0) - 1 | 0] = value;
      return oldValue;
    } else {
      valuesArray[index] = value;
      return null;
    }
  };
  protoOf(InternalHashMap).p4 = function (from) {
    this.r4();
    putAllEntries(this, from.w());
  };
  protoOf(InternalHashMap).equals = function (other) {
    var tmp;
    if (other === this) {
      tmp = true;
    } else {
      var tmp_0;
      if (!(other == null) ? isInterface(other, KtMap) : false) {
        tmp_0 = contentEquals(this, other);
      } else {
        tmp_0 = false;
      }
      tmp = tmp_0;
    }
    return tmp;
  };
  protoOf(InternalHashMap).hashCode = function () {
    var result = 0;
    var it = this.s3();
    while (it.f()) {
      result = result + it.o5() | 0;
    }
    return result;
  };
  protoOf(InternalHashMap).toString = function () {
    var sb = StringBuilder_init_$Create$(2 + imul(this.k4_1, 3) | 0);
    sb.s5('{');
    var i = 0;
    var it = this.s3();
    while (it.f()) {
      if (i > 0) {
        sb.s5(', ');
      }
      it.p5(sb);
      i = i + 1 | 0;
    }
    sb.s5('}');
    return sb.toString();
  };
  protoOf(InternalHashMap).r4 = function () {
    if (this.l4_1)
      throw UnsupportedOperationException_init_$Create$();
  };
  protoOf(InternalHashMap).v3 = function (entry) {
    var index = findKey(this, entry.p());
    if (index < 0)
      return false;
    return equals(ensureNotNull(this.d4_1)[index], entry.q());
  };
  protoOf(InternalHashMap).v5 = function (entry) {
    return this.v3(isInterface(entry, Entry) ? entry : THROW_CCE());
  };
  protoOf(InternalHashMap).l3 = function () {
    return new KeysItr(this);
  };
  protoOf(InternalHashMap).p3 = function () {
    return new ValuesItr(this);
  };
  protoOf(InternalHashMap).s3 = function () {
    return new EntriesItr(this);
  };
  function InternalMap() {
  }
  function LinkedHashMap_init_$Init$($this) {
    HashMap_init_$Init$_0($this);
    LinkedHashMap.call($this);
    return $this;
  }
  function LinkedHashMap_init_$Create$() {
    return LinkedHashMap_init_$Init$(objectCreate(protoOf(LinkedHashMap)));
  }
  function LinkedHashMap_init_$Init$_0(original, $this) {
    HashMap_init_$Init$_1(original, $this);
    LinkedHashMap.call($this);
    return $this;
  }
  function LinkedHashMap_init_$Create$_0(original) {
    return LinkedHashMap_init_$Init$_0(original, objectCreate(protoOf(LinkedHashMap)));
  }
  function LinkedHashMap_init_$Init$_1(internalMap, $this) {
    HashMap_init_$Init$(internalMap, $this);
    LinkedHashMap.call($this);
    return $this;
  }
  function LinkedHashMap_init_$Create$_1(internalMap) {
    return LinkedHashMap_init_$Init$_1(internalMap, objectCreate(protoOf(LinkedHashMap)));
  }
  function EmptyHolder() {
    EmptyHolder_instance = this;
    var tmp = this;
    // Inline function 'kotlin.also' call
    var this_0 = InternalHashMap_init_$Create$_0(0);
    this_0.u5();
    tmp.w5_1 = LinkedHashMap_init_$Create$_1(this_0);
  }
  var EmptyHolder_instance;
  function EmptyHolder_getInstance() {
    if (EmptyHolder_instance == null)
      new EmptyHolder();
    return EmptyHolder_instance;
  }
  protoOf(LinkedHashMap).w1 = function () {
    this.h3_1.u5();
    var tmp;
    if (this.h() > 0) {
      tmp = this;
    } else {
      // Inline function 'kotlin.js.unsafeCast' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp = EmptyHolder_getInstance().w5_1;
    }
    return tmp;
  };
  function LinkedHashMap() {
  }
  function get_output() {
    _init_properties_console_kt__rfg7jv();
    return output;
  }
  var output;
  function BaseOutput() {
  }
  protoOf(BaseOutput).x5 = function () {
    this.y5('\n');
  };
  protoOf(BaseOutput).z5 = function (message) {
    this.y5(message);
    this.x5();
  };
  function NodeJsOutput(outputStream) {
    BaseOutput.call(this);
    this.a6_1 = outputStream;
  }
  protoOf(NodeJsOutput).y5 = function (message) {
    // Inline function 'kotlin.io.String' call
    var tmp1_elvis_lhs = message == null ? null : toString_1(message);
    var messageString = tmp1_elvis_lhs == null ? 'null' : tmp1_elvis_lhs;
    this.a6_1.write(messageString);
  };
  function BufferedOutputToConsoleLog() {
    BufferedOutput.call(this);
  }
  protoOf(BufferedOutputToConsoleLog).y5 = function (message) {
    // Inline function 'kotlin.io.String' call
    var tmp1_elvis_lhs = message == null ? null : toString_1(message);
    var s = tmp1_elvis_lhs == null ? 'null' : tmp1_elvis_lhs;
    // Inline function 'kotlin.text.nativeLastIndexOf' call
    // Inline function 'kotlin.js.asDynamic' call
    var i = s.lastIndexOf('\n', 0);
    if (i >= 0) {
      this.c6_1 = this.c6_1 + substring(s, 0, i);
      this.d6();
      s = substring_0(s, i + 1 | 0);
    }
    this.c6_1 = this.c6_1 + s;
  };
  protoOf(BufferedOutputToConsoleLog).d6 = function () {
    console.log(this.c6_1);
    this.c6_1 = '';
  };
  function BufferedOutput() {
    BaseOutput.call(this);
    this.c6_1 = '';
  }
  protoOf(BufferedOutput).y5 = function (message) {
    var tmp = this;
    var tmp_0 = this.c6_1;
    // Inline function 'kotlin.io.String' call
    var tmp1_elvis_lhs = message == null ? null : toString_1(message);
    tmp.c6_1 = tmp_0 + (tmp1_elvis_lhs == null ? 'null' : tmp1_elvis_lhs);
  };
  function println(message) {
    _init_properties_console_kt__rfg7jv();
    get_output().z5(message);
  }
  var properties_initialized_console_kt_gll9dl;
  function _init_properties_console_kt__rfg7jv() {
    if (!properties_initialized_console_kt_gll9dl) {
      properties_initialized_console_kt_gll9dl = true;
      // Inline function 'kotlin.run' call
      var isNode = typeof process !== 'undefined' && process.versions && !!process.versions.node;
      output = isNode ? new NodeJsOutput(process.stdout) : new BufferedOutputToConsoleLog();
    }
  }
  function Exception_init_$Init$($this) {
    extendThrowable($this);
    Exception.call($this);
    return $this;
  }
  function Exception_init_$Create$() {
    var tmp = Exception_init_$Init$(objectCreate(protoOf(Exception)));
    captureStack(tmp, Exception_init_$Create$);
    return tmp;
  }
  function Exception_init_$Init$_0(message, $this) {
    extendThrowable($this, message);
    Exception.call($this);
    return $this;
  }
  function Exception_init_$Create$_0(message) {
    var tmp = Exception_init_$Init$_0(message, objectCreate(protoOf(Exception)));
    captureStack(tmp, Exception_init_$Create$_0);
    return tmp;
  }
  function Exception() {
    captureStack(this, Exception);
  }
  function IllegalArgumentException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    IllegalArgumentException.call($this);
    return $this;
  }
  function IllegalArgumentException_init_$Create$() {
    var tmp = IllegalArgumentException_init_$Init$(objectCreate(protoOf(IllegalArgumentException)));
    captureStack(tmp, IllegalArgumentException_init_$Create$);
    return tmp;
  }
  function IllegalArgumentException_init_$Init$_0(message, $this) {
    RuntimeException_init_$Init$_0(message, $this);
    IllegalArgumentException.call($this);
    return $this;
  }
  function IllegalArgumentException_init_$Create$_0(message) {
    var tmp = IllegalArgumentException_init_$Init$_0(message, objectCreate(protoOf(IllegalArgumentException)));
    captureStack(tmp, IllegalArgumentException_init_$Create$_0);
    return tmp;
  }
  function IllegalArgumentException() {
    captureStack(this, IllegalArgumentException);
  }
  function IllegalStateException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    IllegalStateException.call($this);
    return $this;
  }
  function IllegalStateException_init_$Create$() {
    var tmp = IllegalStateException_init_$Init$(objectCreate(protoOf(IllegalStateException)));
    captureStack(tmp, IllegalStateException_init_$Create$);
    return tmp;
  }
  function IllegalStateException_init_$Init$_0(message, $this) {
    RuntimeException_init_$Init$_0(message, $this);
    IllegalStateException.call($this);
    return $this;
  }
  function IllegalStateException_init_$Create$_0(message) {
    var tmp = IllegalStateException_init_$Init$_0(message, objectCreate(protoOf(IllegalStateException)));
    captureStack(tmp, IllegalStateException_init_$Create$_0);
    return tmp;
  }
  function IllegalStateException() {
    captureStack(this, IllegalStateException);
  }
  function UnsupportedOperationException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    UnsupportedOperationException.call($this);
    return $this;
  }
  function UnsupportedOperationException_init_$Create$() {
    var tmp = UnsupportedOperationException_init_$Init$(objectCreate(protoOf(UnsupportedOperationException)));
    captureStack(tmp, UnsupportedOperationException_init_$Create$);
    return tmp;
  }
  function UnsupportedOperationException_init_$Init$_0(message, $this) {
    RuntimeException_init_$Init$_0(message, $this);
    UnsupportedOperationException.call($this);
    return $this;
  }
  function UnsupportedOperationException_init_$Create$_0(message) {
    var tmp = UnsupportedOperationException_init_$Init$_0(message, objectCreate(protoOf(UnsupportedOperationException)));
    captureStack(tmp, UnsupportedOperationException_init_$Create$_0);
    return tmp;
  }
  function UnsupportedOperationException() {
    captureStack(this, UnsupportedOperationException);
  }
  function RuntimeException_init_$Init$($this) {
    Exception_init_$Init$($this);
    RuntimeException.call($this);
    return $this;
  }
  function RuntimeException_init_$Create$() {
    var tmp = RuntimeException_init_$Init$(objectCreate(protoOf(RuntimeException)));
    captureStack(tmp, RuntimeException_init_$Create$);
    return tmp;
  }
  function RuntimeException_init_$Init$_0(message, $this) {
    Exception_init_$Init$_0(message, $this);
    RuntimeException.call($this);
    return $this;
  }
  function RuntimeException_init_$Create$_0(message) {
    var tmp = RuntimeException_init_$Init$_0(message, objectCreate(protoOf(RuntimeException)));
    captureStack(tmp, RuntimeException_init_$Create$_0);
    return tmp;
  }
  function RuntimeException() {
    captureStack(this, RuntimeException);
  }
  function NoSuchElementException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    NoSuchElementException.call($this);
    return $this;
  }
  function NoSuchElementException_init_$Create$() {
    var tmp = NoSuchElementException_init_$Init$(objectCreate(protoOf(NoSuchElementException)));
    captureStack(tmp, NoSuchElementException_init_$Create$);
    return tmp;
  }
  function NoSuchElementException_init_$Init$_0(message, $this) {
    RuntimeException_init_$Init$_0(message, $this);
    NoSuchElementException.call($this);
    return $this;
  }
  function NoSuchElementException_init_$Create$_0(message) {
    var tmp = NoSuchElementException_init_$Init$_0(message, objectCreate(protoOf(NoSuchElementException)));
    captureStack(tmp, NoSuchElementException_init_$Create$_0);
    return tmp;
  }
  function NoSuchElementException() {
    captureStack(this, NoSuchElementException);
  }
  function IndexOutOfBoundsException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    IndexOutOfBoundsException.call($this);
    return $this;
  }
  function IndexOutOfBoundsException_init_$Create$() {
    var tmp = IndexOutOfBoundsException_init_$Init$(objectCreate(protoOf(IndexOutOfBoundsException)));
    captureStack(tmp, IndexOutOfBoundsException_init_$Create$);
    return tmp;
  }
  function IndexOutOfBoundsException_init_$Init$_0(message, $this) {
    RuntimeException_init_$Init$_0(message, $this);
    IndexOutOfBoundsException.call($this);
    return $this;
  }
  function IndexOutOfBoundsException_init_$Create$_0(message) {
    var tmp = IndexOutOfBoundsException_init_$Init$_0(message, objectCreate(protoOf(IndexOutOfBoundsException)));
    captureStack(tmp, IndexOutOfBoundsException_init_$Create$_0);
    return tmp;
  }
  function IndexOutOfBoundsException() {
    captureStack(this, IndexOutOfBoundsException);
  }
  function ConcurrentModificationException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    ConcurrentModificationException.call($this);
    return $this;
  }
  function ConcurrentModificationException_init_$Create$() {
    var tmp = ConcurrentModificationException_init_$Init$(objectCreate(protoOf(ConcurrentModificationException)));
    captureStack(tmp, ConcurrentModificationException_init_$Create$);
    return tmp;
  }
  function ConcurrentModificationException_init_$Init$_0(message, $this) {
    RuntimeException_init_$Init$_0(message, $this);
    ConcurrentModificationException.call($this);
    return $this;
  }
  function ConcurrentModificationException_init_$Create$_0(message) {
    var tmp = ConcurrentModificationException_init_$Init$_0(message, objectCreate(protoOf(ConcurrentModificationException)));
    captureStack(tmp, ConcurrentModificationException_init_$Create$_0);
    return tmp;
  }
  function ConcurrentModificationException() {
    captureStack(this, ConcurrentModificationException);
  }
  function NullPointerException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    NullPointerException.call($this);
    return $this;
  }
  function NullPointerException_init_$Create$() {
    var tmp = NullPointerException_init_$Init$(objectCreate(protoOf(NullPointerException)));
    captureStack(tmp, NullPointerException_init_$Create$);
    return tmp;
  }
  function NullPointerException() {
    captureStack(this, NullPointerException);
  }
  function ClassCastException_init_$Init$($this) {
    RuntimeException_init_$Init$($this);
    ClassCastException.call($this);
    return $this;
  }
  function ClassCastException_init_$Create$() {
    var tmp = ClassCastException_init_$Init$(objectCreate(protoOf(ClassCastException)));
    captureStack(tmp, ClassCastException_init_$Create$);
    return tmp;
  }
  function ClassCastException() {
    captureStack(this, ClassCastException);
  }
  function fillFrom(src, dst) {
    var srcLen = src.length;
    var dstLen = dst.length;
    var index = 0;
    // Inline function 'kotlin.js.unsafeCast' call
    var arr = dst;
    while (index < srcLen && index < dstLen) {
      var tmp = index;
      var _unary__edvuaz = index;
      index = _unary__edvuaz + 1 | 0;
      arr[tmp] = src[_unary__edvuaz];
    }
    return dst;
  }
  function arrayCopyResize(source, newSize, defaultValue) {
    // Inline function 'kotlin.js.unsafeCast' call
    var result = source.slice(0, newSize);
    // Inline function 'kotlin.copyArrayType' call
    if (source.$type$ !== undefined) {
      result.$type$ = source.$type$;
    }
    var index = source.length;
    if (newSize > index) {
      // Inline function 'kotlin.js.asDynamic' call
      result.length = newSize;
      while (index < newSize) {
        var _unary__edvuaz = index;
        index = _unary__edvuaz + 1 | 0;
        result[_unary__edvuaz] = defaultValue;
      }
    }
    return result;
  }
  function CharacterCodingException_init_$Init$($this) {
    CharacterCodingException.call($this, null);
    return $this;
  }
  function CharacterCodingException_init_$Create$() {
    var tmp = CharacterCodingException_init_$Init$(objectCreate(protoOf(CharacterCodingException)));
    captureStack(tmp, CharacterCodingException_init_$Create$);
    return tmp;
  }
  function CharacterCodingException(message) {
    Exception_init_$Init$_0(message, this);
    captureStack(this, CharacterCodingException);
  }
  function StringBuilder_init_$Init$(capacity, $this) {
    StringBuilder_init_$Init$_0($this);
    return $this;
  }
  function StringBuilder_init_$Create$(capacity) {
    return StringBuilder_init_$Init$(capacity, objectCreate(protoOf(StringBuilder)));
  }
  function StringBuilder_init_$Init$_0($this) {
    StringBuilder.call($this, '');
    return $this;
  }
  function StringBuilder_init_$Create$_0() {
    return StringBuilder_init_$Init$_0(objectCreate(protoOf(StringBuilder)));
  }
  function StringBuilder(content) {
    this.q5_1 = content;
  }
  protoOf(StringBuilder).a = function () {
    // Inline function 'kotlin.js.asDynamic' call
    return this.q5_1.length;
  };
  protoOf(StringBuilder).b = function (index) {
    // Inline function 'kotlin.text.getOrElse' call
    var this_0 = this.q5_1;
    var tmp;
    if (0 <= index ? index <= (charSequenceLength(this_0) - 1 | 0) : false) {
      tmp = charSequenceGet(this_0, index);
    } else {
      throw IndexOutOfBoundsException_init_$Create$_0('index: ' + index + ', length: ' + this.a() + '}');
    }
    return tmp;
  };
  protoOf(StringBuilder).c = function (startIndex, endIndex) {
    return substring(this.q5_1, startIndex, endIndex);
  };
  protoOf(StringBuilder).t5 = function (value) {
    this.q5_1 = this.q5_1 + toString(value);
    return this;
  };
  protoOf(StringBuilder).d = function (value) {
    this.q5_1 = this.q5_1 + toString_0(value);
    return this;
  };
  protoOf(StringBuilder).r5 = function (value) {
    this.q5_1 = this.q5_1 + toString_0(value);
    return this;
  };
  protoOf(StringBuilder).s5 = function (value) {
    var tmp = this;
    var tmp_0 = this.q5_1;
    tmp.q5_1 = tmp_0 + (value == null ? 'null' : value);
    return this;
  };
  protoOf(StringBuilder).toString = function () {
    return this.q5_1;
  };
  function uppercaseChar(_this__u8e3s4) {
    // Inline function 'kotlin.text.uppercase' call
    // Inline function 'kotlin.js.asDynamic' call
    // Inline function 'kotlin.js.unsafeCast' call
    var uppercase = toString(_this__u8e3s4).toUpperCase();
    return uppercase.length > 1 ? _this__u8e3s4 : charCodeAt(uppercase, 0);
  }
  function checkRadix(radix) {
    if (!(2 <= radix ? radix <= 36 : false)) {
      throw IllegalArgumentException_init_$Create$_0('radix ' + radix + ' was not in valid range 2..36');
    }
    return radix;
  }
  function toString_2(_this__u8e3s4, radix) {
    // Inline function 'kotlin.js.asDynamic' call
    return _this__u8e3s4.toString(checkRadix(radix));
  }
  var STRING_CASE_INSENSITIVE_ORDER;
  function substring(_this__u8e3s4, startIndex, endIndex) {
    _init_properties_stringJs_kt__bg7zye();
    // Inline function 'kotlin.js.asDynamic' call
    return _this__u8e3s4.substring(startIndex, endIndex);
  }
  function substring_0(_this__u8e3s4, startIndex) {
    _init_properties_stringJs_kt__bg7zye();
    // Inline function 'kotlin.js.asDynamic' call
    return _this__u8e3s4.substring(startIndex);
  }
  function compareTo_0(_this__u8e3s4, other, ignoreCase) {
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    _init_properties_stringJs_kt__bg7zye();
    if (ignoreCase) {
      var n1 = _this__u8e3s4.length;
      var n2 = other.length;
      // Inline function 'kotlin.comparisons.minOf' call
      var min = Math.min(n1, n2);
      if (min === 0)
        return n1 - n2 | 0;
      var inductionVariable = 0;
      if (inductionVariable < min)
        do {
          var index = inductionVariable;
          inductionVariable = inductionVariable + 1 | 0;
          var thisChar = charCodeAt(_this__u8e3s4, index);
          var otherChar = charCodeAt(other, index);
          if (!(thisChar === otherChar)) {
            thisChar = uppercaseChar(thisChar);
            otherChar = uppercaseChar(otherChar);
            if (!(thisChar === otherChar)) {
              // Inline function 'kotlin.text.lowercaseChar' call
              // Inline function 'kotlin.text.lowercase' call
              var this_0 = thisChar;
              // Inline function 'kotlin.js.asDynamic' call
              // Inline function 'kotlin.js.unsafeCast' call
              var tmp$ret$3 = toString(this_0).toLowerCase();
              thisChar = charCodeAt(tmp$ret$3, 0);
              // Inline function 'kotlin.text.lowercaseChar' call
              // Inline function 'kotlin.text.lowercase' call
              var this_1 = otherChar;
              // Inline function 'kotlin.js.asDynamic' call
              // Inline function 'kotlin.js.unsafeCast' call
              var tmp$ret$7 = toString(this_1).toLowerCase();
              otherChar = charCodeAt(tmp$ret$7, 0);
              if (!(thisChar === otherChar)) {
                return Char__compareTo_impl_ypi4mb(thisChar, otherChar);
              }
            }
          }
        }
         while (inductionVariable < min);
      return n1 - n2 | 0;
    } else {
      return compareTo(_this__u8e3s4, other);
    }
  }
  function decodeToString(_this__u8e3s4) {
    _init_properties_stringJs_kt__bg7zye();
    return decodeUtf8(_this__u8e3s4, 0, _this__u8e3s4.length, false);
  }
  function encodeToByteArray(_this__u8e3s4) {
    _init_properties_stringJs_kt__bg7zye();
    return encodeUtf8(_this__u8e3s4, 0, _this__u8e3s4.length, false);
  }
  function sam$kotlin_Comparator$0(function_0) {
    this.g6_1 = function_0;
  }
  protoOf(sam$kotlin_Comparator$0).h6 = function (a, b) {
    return this.g6_1(a, b);
  };
  protoOf(sam$kotlin_Comparator$0).compare = function (a, b) {
    return this.h6(a, b);
  };
  protoOf(sam$kotlin_Comparator$0).p1 = function () {
    return this.g6_1;
  };
  protoOf(sam$kotlin_Comparator$0).equals = function (other) {
    var tmp;
    if (!(other == null) ? isInterface(other, Comparator) : false) {
      var tmp_0;
      if (!(other == null) ? isInterface(other, FunctionAdapter) : false) {
        tmp_0 = equals(this.p1(), other.p1());
      } else {
        tmp_0 = false;
      }
      tmp = tmp_0;
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(sam$kotlin_Comparator$0).hashCode = function () {
    return hashCode(this.p1());
  };
  function STRING_CASE_INSENSITIVE_ORDER$lambda(a, b) {
    _init_properties_stringJs_kt__bg7zye();
    return compareTo_0(a, b, true);
  }
  var properties_initialized_stringJs_kt_nta8o4;
  function _init_properties_stringJs_kt__bg7zye() {
    if (!properties_initialized_stringJs_kt_nta8o4) {
      properties_initialized_stringJs_kt_nta8o4 = true;
      var tmp = STRING_CASE_INSENSITIVE_ORDER$lambda;
      STRING_CASE_INSENSITIVE_ORDER = new sam$kotlin_Comparator$0(tmp);
    }
  }
  function startsWith(_this__u8e3s4, prefix, ignoreCase) {
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    if (!ignoreCase) {
      // Inline function 'kotlin.text.nativeStartsWith' call
      // Inline function 'kotlin.js.asDynamic' call
      return _this__u8e3s4.startsWith(prefix, 0);
    } else
      return regionMatches(_this__u8e3s4, 0, prefix, 0, prefix.length, ignoreCase);
  }
  function endsWith(_this__u8e3s4, suffix, ignoreCase) {
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    if (!ignoreCase) {
      // Inline function 'kotlin.text.nativeEndsWith' call
      // Inline function 'kotlin.js.asDynamic' call
      return _this__u8e3s4.endsWith(suffix);
    } else
      return regionMatches(_this__u8e3s4, _this__u8e3s4.length - suffix.length | 0, suffix, 0, suffix.length, ignoreCase);
  }
  function regionMatches(_this__u8e3s4, thisOffset, other, otherOffset, length, ignoreCase) {
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    return regionMatchesImpl(_this__u8e3s4, thisOffset, other, otherOffset, length, ignoreCase);
  }
  function get_REPLACEMENT_BYTE_SEQUENCE() {
    _init_properties_utf8Encoding_kt__9thjs4();
    return REPLACEMENT_BYTE_SEQUENCE;
  }
  var REPLACEMENT_BYTE_SEQUENCE;
  function decodeUtf8(bytes, startIndex, endIndex, throwOnMalformed) {
    _init_properties_utf8Encoding_kt__9thjs4();
    // Inline function 'kotlin.require' call
    // Inline function 'kotlin.require' call
    if (!(startIndex >= 0 && endIndex <= bytes.length && startIndex <= endIndex)) {
      var message = 'Failed requirement.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    var byteIndex = startIndex;
    var stringBuilder = StringBuilder_init_$Create$_0();
    while (byteIndex < endIndex) {
      var _unary__edvuaz = byteIndex;
      byteIndex = _unary__edvuaz + 1 | 0;
      var byte = bytes[_unary__edvuaz];
      if (byte >= 0)
        stringBuilder.t5(numberToChar(byte));
      else if (byte >> 5 === -2) {
        var code = codePointFrom2(bytes, byte, byteIndex, endIndex, throwOnMalformed);
        if (code <= 0) {
          stringBuilder.t5(_Char___init__impl__6a9atx(65533));
          byteIndex = byteIndex + (-code | 0) | 0;
        } else {
          stringBuilder.t5(numberToChar(code));
          byteIndex = byteIndex + 1 | 0;
        }
      } else if (byte >> 4 === -2) {
        var code_0 = codePointFrom3(bytes, byte, byteIndex, endIndex, throwOnMalformed);
        if (code_0 <= 0) {
          stringBuilder.t5(_Char___init__impl__6a9atx(65533));
          byteIndex = byteIndex + (-code_0 | 0) | 0;
        } else {
          stringBuilder.t5(numberToChar(code_0));
          byteIndex = byteIndex + 2 | 0;
        }
      } else if (byte >> 3 === -2) {
        var code_1 = codePointFrom4(bytes, byte, byteIndex, endIndex, throwOnMalformed);
        if (code_1 <= 0) {
          stringBuilder.t5(_Char___init__impl__6a9atx(65533));
          byteIndex = byteIndex + (-code_1 | 0) | 0;
        } else {
          var high = (code_1 - 65536 | 0) >> 10 | 55296;
          var low = code_1 & 1023 | 56320;
          stringBuilder.t5(numberToChar(high));
          stringBuilder.t5(numberToChar(low));
          byteIndex = byteIndex + 3 | 0;
        }
      } else {
        malformed(0, byteIndex, throwOnMalformed);
        stringBuilder.t5(_Char___init__impl__6a9atx(65533));
      }
    }
    return stringBuilder.toString();
  }
  function encodeUtf8(string, startIndex, endIndex, throwOnMalformed) {
    _init_properties_utf8Encoding_kt__9thjs4();
    // Inline function 'kotlin.require' call
    // Inline function 'kotlin.require' call
    if (!(startIndex >= 0 && endIndex <= string.length && startIndex <= endIndex)) {
      var message = 'Failed requirement.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    var bytes = new Int8Array(imul(endIndex - startIndex | 0, 3));
    var byteIndex = 0;
    var charIndex = startIndex;
    while (charIndex < endIndex) {
      var _unary__edvuaz = charIndex;
      charIndex = _unary__edvuaz + 1 | 0;
      // Inline function 'kotlin.code' call
      var this_0 = charCodeAt(string, _unary__edvuaz);
      var code = Char__toInt_impl_vasixd(this_0);
      if (code < 128) {
        var _unary__edvuaz_0 = byteIndex;
        byteIndex = _unary__edvuaz_0 + 1 | 0;
        bytes[_unary__edvuaz_0] = toByte(code);
      } else if (code < 2048) {
        var _unary__edvuaz_1 = byteIndex;
        byteIndex = _unary__edvuaz_1 + 1 | 0;
        bytes[_unary__edvuaz_1] = toByte(code >> 6 | 192);
        var _unary__edvuaz_2 = byteIndex;
        byteIndex = _unary__edvuaz_2 + 1 | 0;
        bytes[_unary__edvuaz_2] = toByte(code & 63 | 128);
      } else if (code < 55296 || code >= 57344) {
        var _unary__edvuaz_3 = byteIndex;
        byteIndex = _unary__edvuaz_3 + 1 | 0;
        bytes[_unary__edvuaz_3] = toByte(code >> 12 | 224);
        var _unary__edvuaz_4 = byteIndex;
        byteIndex = _unary__edvuaz_4 + 1 | 0;
        bytes[_unary__edvuaz_4] = toByte(code >> 6 & 63 | 128);
        var _unary__edvuaz_5 = byteIndex;
        byteIndex = _unary__edvuaz_5 + 1 | 0;
        bytes[_unary__edvuaz_5] = toByte(code & 63 | 128);
      } else {
        var codePoint = codePointFromSurrogate(string, code, charIndex, endIndex, throwOnMalformed);
        if (codePoint <= 0) {
          var _unary__edvuaz_6 = byteIndex;
          byteIndex = _unary__edvuaz_6 + 1 | 0;
          bytes[_unary__edvuaz_6] = get_REPLACEMENT_BYTE_SEQUENCE()[0];
          var _unary__edvuaz_7 = byteIndex;
          byteIndex = _unary__edvuaz_7 + 1 | 0;
          bytes[_unary__edvuaz_7] = get_REPLACEMENT_BYTE_SEQUENCE()[1];
          var _unary__edvuaz_8 = byteIndex;
          byteIndex = _unary__edvuaz_8 + 1 | 0;
          bytes[_unary__edvuaz_8] = get_REPLACEMENT_BYTE_SEQUENCE()[2];
        } else {
          var _unary__edvuaz_9 = byteIndex;
          byteIndex = _unary__edvuaz_9 + 1 | 0;
          bytes[_unary__edvuaz_9] = toByte(codePoint >> 18 | 240);
          var _unary__edvuaz_10 = byteIndex;
          byteIndex = _unary__edvuaz_10 + 1 | 0;
          bytes[_unary__edvuaz_10] = toByte(codePoint >> 12 & 63 | 128);
          var _unary__edvuaz_11 = byteIndex;
          byteIndex = _unary__edvuaz_11 + 1 | 0;
          bytes[_unary__edvuaz_11] = toByte(codePoint >> 6 & 63 | 128);
          var _unary__edvuaz_12 = byteIndex;
          byteIndex = _unary__edvuaz_12 + 1 | 0;
          bytes[_unary__edvuaz_12] = toByte(codePoint & 63 | 128);
          charIndex = charIndex + 1 | 0;
        }
      }
    }
    return bytes.length === byteIndex ? bytes : copyOf(bytes, byteIndex);
  }
  function codePointFrom2(bytes, byte1, index, endIndex, throwOnMalformed) {
    _init_properties_utf8Encoding_kt__9thjs4();
    if ((byte1 & 30) === 0 || index >= endIndex) {
      return malformed(0, index, throwOnMalformed);
    }
    var byte2 = bytes[index];
    if (!((byte2 & 192) === 128)) {
      return malformed(0, index, throwOnMalformed);
    }
    return byte1 << 6 ^ byte2 ^ 3968;
  }
  function codePointFrom3(bytes, byte1, index, endIndex, throwOnMalformed) {
    _init_properties_utf8Encoding_kt__9thjs4();
    if (index >= endIndex) {
      return malformed(0, index, throwOnMalformed);
    }
    var byte2 = bytes[index];
    if ((byte1 & 15) === 0) {
      if (!((byte2 & 224) === 160)) {
        return malformed(0, index, throwOnMalformed);
      }
    } else if ((byte1 & 15) === 13) {
      if (!((byte2 & 224) === 128)) {
        return malformed(0, index, throwOnMalformed);
      }
    } else if (!((byte2 & 192) === 128)) {
      return malformed(0, index, throwOnMalformed);
    }
    if ((index + 1 | 0) === endIndex) {
      return malformed(1, index, throwOnMalformed);
    }
    var byte3 = bytes[index + 1 | 0];
    if (!((byte3 & 192) === 128)) {
      return malformed(1, index, throwOnMalformed);
    }
    return byte1 << 12 ^ byte2 << 6 ^ byte3 ^ -123008;
  }
  function codePointFrom4(bytes, byte1, index, endIndex, throwOnMalformed) {
    _init_properties_utf8Encoding_kt__9thjs4();
    if (index >= endIndex) {
      return malformed(0, index, throwOnMalformed);
    }
    var byte2 = bytes[index];
    if ((byte1 & 15) === 0) {
      if ((byte2 & 240) <= 128) {
        return malformed(0, index, throwOnMalformed);
      }
    } else if ((byte1 & 15) === 4) {
      if (!((byte2 & 240) === 128)) {
        return malformed(0, index, throwOnMalformed);
      }
    } else if ((byte1 & 15) > 4) {
      return malformed(0, index, throwOnMalformed);
    }
    if (!((byte2 & 192) === 128)) {
      return malformed(0, index, throwOnMalformed);
    }
    if ((index + 1 | 0) === endIndex) {
      return malformed(1, index, throwOnMalformed);
    }
    var byte3 = bytes[index + 1 | 0];
    if (!((byte3 & 192) === 128)) {
      return malformed(1, index, throwOnMalformed);
    }
    if ((index + 2 | 0) === endIndex) {
      return malformed(2, index, throwOnMalformed);
    }
    var byte4 = bytes[index + 2 | 0];
    if (!((byte4 & 192) === 128)) {
      return malformed(2, index, throwOnMalformed);
    }
    return byte1 << 18 ^ byte2 << 12 ^ byte3 << 6 ^ byte4 ^ 3678080;
  }
  function malformed(size, index, throwOnMalformed) {
    _init_properties_utf8Encoding_kt__9thjs4();
    if (throwOnMalformed)
      throw new CharacterCodingException('Malformed sequence starting at ' + (index - 1 | 0));
    return -size | 0;
  }
  function codePointFromSurrogate(string, high, index, endIndex, throwOnMalformed) {
    _init_properties_utf8Encoding_kt__9thjs4();
    if (!(55296 <= high ? high <= 56319 : false) || index >= endIndex) {
      return malformed(0, index, throwOnMalformed);
    }
    // Inline function 'kotlin.code' call
    var this_0 = charCodeAt(string, index);
    var low = Char__toInt_impl_vasixd(this_0);
    if (!(56320 <= low ? low <= 57343 : false)) {
      return malformed(0, index, throwOnMalformed);
    }
    return 65536 + ((high & 1023) << 10) | 0 | low & 1023;
  }
  var properties_initialized_utf8Encoding_kt_eee1vq;
  function _init_properties_utf8Encoding_kt__9thjs4() {
    if (!properties_initialized_utf8Encoding_kt_eee1vq) {
      properties_initialized_utf8Encoding_kt_eee1vq = true;
      // Inline function 'kotlin.byteArrayOf' call
      REPLACEMENT_BYTE_SEQUENCE = new Int8Array([-17, -65, -67]);
    }
  }
  function AbstractCollection$toString$lambda(this$0) {
    return function (it) {
      return it === this$0 ? '(this Collection)' : toString_0(it);
    };
  }
  function AbstractCollection() {
  }
  protoOf(AbstractCollection).n = function (element) {
    var tmp$ret$0;
    $l$block_0: {
      // Inline function 'kotlin.collections.any' call
      var tmp;
      if (isInterface(this, Collection)) {
        tmp = this.m();
      } else {
        tmp = false;
      }
      if (tmp) {
        tmp$ret$0 = false;
        break $l$block_0;
      }
      var _iterator__ex2g4s = this.e();
      while (_iterator__ex2g4s.f()) {
        var element_0 = _iterator__ex2g4s.g();
        if (equals(element_0, element)) {
          tmp$ret$0 = true;
          break $l$block_0;
        }
      }
      tmp$ret$0 = false;
    }
    return tmp$ret$0;
  };
  protoOf(AbstractCollection).o = function (elements) {
    var tmp$ret$0;
    $l$block_0: {
      // Inline function 'kotlin.collections.all' call
      var tmp;
      if (isInterface(elements, Collection)) {
        tmp = elements.m();
      } else {
        tmp = false;
      }
      if (tmp) {
        tmp$ret$0 = true;
        break $l$block_0;
      }
      var _iterator__ex2g4s = elements.e();
      while (_iterator__ex2g4s.f()) {
        var element = _iterator__ex2g4s.g();
        if (!this.n(element)) {
          tmp$ret$0 = false;
          break $l$block_0;
        }
      }
      tmp$ret$0 = true;
    }
    return tmp$ret$0;
  };
  protoOf(AbstractCollection).m = function () {
    return this.h() === 0;
  };
  protoOf(AbstractCollection).toString = function () {
    return joinToString_0(this, ', ', '[', ']', VOID, VOID, AbstractCollection$toString$lambda(this));
  };
  protoOf(AbstractCollection).toArray = function () {
    return collectionToArray(this);
  };
  function Companion_4() {
    this.y1_1 = 2147483639;
  }
  protoOf(Companion_4).a3 = function (index, size) {
    if (index < 0 || index >= size) {
      throw IndexOutOfBoundsException_init_$Create$_0('index: ' + index + ', size: ' + size);
    }
  };
  protoOf(Companion_4).b3 = function (index, size) {
    if (index < 0 || index > size) {
      throw IndexOutOfBoundsException_init_$Create$_0('index: ' + index + ', size: ' + size);
    }
  };
  protoOf(Companion_4).z1 = function (fromIndex, toIndex, size) {
    if (fromIndex < 0 || toIndex > size) {
      throw IndexOutOfBoundsException_init_$Create$_0('fromIndex: ' + fromIndex + ', toIndex: ' + toIndex + ', size: ' + size);
    }
    if (fromIndex > toIndex) {
      throw IllegalArgumentException_init_$Create$_0('fromIndex: ' + fromIndex + ' > toIndex: ' + toIndex);
    }
  };
  protoOf(Companion_4).i6 = function (startIndex, endIndex, size) {
    if (startIndex < 0 || endIndex > size) {
      throw IndexOutOfBoundsException_init_$Create$_0('startIndex: ' + startIndex + ', endIndex: ' + endIndex + ', size: ' + size);
    }
    if (startIndex > endIndex) {
      throw IllegalArgumentException_init_$Create$_0('startIndex: ' + startIndex + ' > endIndex: ' + endIndex);
    }
  };
  protoOf(Companion_4).q4 = function (oldCapacity, minCapacity) {
    var newCapacity = oldCapacity + (oldCapacity >> 1) | 0;
    if ((newCapacity - minCapacity | 0) < 0)
      newCapacity = minCapacity;
    if ((newCapacity - 2147483639 | 0) > 0)
      newCapacity = minCapacity > 2147483639 ? 2147483647 : 2147483639;
    return newCapacity;
  };
  protoOf(Companion_4).k2 = function (c) {
    var hashCode_0 = 1;
    var _iterator__ex2g4s = c.e();
    while (_iterator__ex2g4s.f()) {
      var e = _iterator__ex2g4s.g();
      var tmp = imul(31, hashCode_0);
      var tmp1_elvis_lhs = e == null ? null : hashCode(e);
      hashCode_0 = tmp + (tmp1_elvis_lhs == null ? 0 : tmp1_elvis_lhs) | 0;
    }
    return hashCode_0;
  };
  protoOf(Companion_4).j2 = function (c, other) {
    if (!(c.h() === other.h()))
      return false;
    var otherIterator = other.e();
    var _iterator__ex2g4s = c.e();
    while (_iterator__ex2g4s.f()) {
      var elem = _iterator__ex2g4s.g();
      var elemOther = otherIterator.g();
      if (!equals(elem, elemOther)) {
        return false;
      }
    }
    return true;
  };
  var Companion_instance_4;
  function Companion_getInstance_4() {
    return Companion_instance_4;
  }
  function AbstractMap$keys$1$iterator$1($entryIterator) {
    this.j6_1 = $entryIterator;
  }
  protoOf(AbstractMap$keys$1$iterator$1).f = function () {
    return this.j6_1.f();
  };
  protoOf(AbstractMap$keys$1$iterator$1).g = function () {
    return this.j6_1.g().p();
  };
  function AbstractMap$values$1$iterator$1($entryIterator) {
    this.k6_1 = $entryIterator;
  }
  protoOf(AbstractMap$values$1$iterator$1).f = function () {
    return this.k6_1.f();
  };
  protoOf(AbstractMap$values$1$iterator$1).g = function () {
    return this.k6_1.g().q();
  };
  function toString_3($this, entry) {
    return toString_4($this, entry.p()) + '=' + toString_4($this, entry.q());
  }
  function toString_4($this, o) {
    return o === $this ? '(this Map)' : toString_0(o);
  }
  function implFindEntry($this, key) {
    var tmp0 = $this.w();
    var tmp$ret$1;
    $l$block: {
      // Inline function 'kotlin.collections.firstOrNull' call
      var _iterator__ex2g4s = tmp0.e();
      while (_iterator__ex2g4s.f()) {
        var element = _iterator__ex2g4s.g();
        if (equals(element.p(), key)) {
          tmp$ret$1 = element;
          break $l$block;
        }
      }
      tmp$ret$1 = null;
    }
    return tmp$ret$1;
  }
  function Companion_5() {
  }
  var Companion_instance_5;
  function Companion_getInstance_5() {
    return Companion_instance_5;
  }
  function AbstractMap$keys$1(this$0) {
    this.l6_1 = this$0;
    AbstractSet.call(this);
  }
  protoOf(AbstractMap$keys$1).j3 = function (element) {
    return this.l6_1.r(element);
  };
  protoOf(AbstractMap$keys$1).n = function (element) {
    if (!(element == null ? true : !(element == null)))
      return false;
    return this.j3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(AbstractMap$keys$1).e = function () {
    var entryIterator = this.l6_1.w().e();
    return new AbstractMap$keys$1$iterator$1(entryIterator);
  };
  protoOf(AbstractMap$keys$1).h = function () {
    return this.l6_1.h();
  };
  function AbstractMap$toString$lambda(this$0) {
    return function (it) {
      return toString_3(this$0, it);
    };
  }
  function AbstractMap$values$1(this$0) {
    this.m6_1 = this$0;
    AbstractCollection.call(this);
  }
  protoOf(AbstractMap$values$1).n3 = function (element) {
    return this.m6_1.s(element);
  };
  protoOf(AbstractMap$values$1).n = function (element) {
    if (!(element == null ? true : !(element == null)))
      return false;
    return this.n3((element == null ? true : !(element == null)) ? element : THROW_CCE());
  };
  protoOf(AbstractMap$values$1).e = function () {
    var entryIterator = this.m6_1.w().e();
    return new AbstractMap$values$1$iterator$1(entryIterator);
  };
  protoOf(AbstractMap$values$1).h = function () {
    return this.m6_1.h();
  };
  function AbstractMap() {
    this.r2_1 = null;
    this.s2_1 = null;
  }
  protoOf(AbstractMap).r = function (key) {
    return !(implFindEntry(this, key) == null);
  };
  protoOf(AbstractMap).s = function (value) {
    var tmp0 = this.w();
    var tmp$ret$0;
    $l$block_0: {
      // Inline function 'kotlin.collections.any' call
      var tmp;
      if (isInterface(tmp0, Collection)) {
        tmp = tmp0.m();
      } else {
        tmp = false;
      }
      if (tmp) {
        tmp$ret$0 = false;
        break $l$block_0;
      }
      var _iterator__ex2g4s = tmp0.e();
      while (_iterator__ex2g4s.f()) {
        var element = _iterator__ex2g4s.g();
        if (equals(element.q(), value)) {
          tmp$ret$0 = true;
          break $l$block_0;
        }
      }
      tmp$ret$0 = false;
    }
    return tmp$ret$0;
  };
  protoOf(AbstractMap).t2 = function (entry) {
    if (!(!(entry == null) ? isInterface(entry, Entry) : false))
      return false;
    var key = entry.p();
    var value = entry.q();
    // Inline function 'kotlin.collections.get' call
    var ourValue = (isInterface(this, KtMap) ? this : THROW_CCE()).t(key);
    if (!equals(value, ourValue)) {
      return false;
    }
    var tmp;
    if (ourValue == null) {
      // Inline function 'kotlin.collections.containsKey' call
      tmp = !(isInterface(this, KtMap) ? this : THROW_CCE()).r(key);
    } else {
      tmp = false;
    }
    if (tmp) {
      return false;
    }
    return true;
  };
  protoOf(AbstractMap).equals = function (other) {
    if (other === this)
      return true;
    if (!(!(other == null) ? isInterface(other, KtMap) : false))
      return false;
    if (!(this.h() === other.h()))
      return false;
    var tmp0 = other.w();
    var tmp$ret$0;
    $l$block_0: {
      // Inline function 'kotlin.collections.all' call
      var tmp;
      if (isInterface(tmp0, Collection)) {
        tmp = tmp0.m();
      } else {
        tmp = false;
      }
      if (tmp) {
        tmp$ret$0 = true;
        break $l$block_0;
      }
      var _iterator__ex2g4s = tmp0.e();
      while (_iterator__ex2g4s.f()) {
        var element = _iterator__ex2g4s.g();
        if (!this.t2(element)) {
          tmp$ret$0 = false;
          break $l$block_0;
        }
      }
      tmp$ret$0 = true;
    }
    return tmp$ret$0;
  };
  protoOf(AbstractMap).t = function (key) {
    var tmp0_safe_receiver = implFindEntry(this, key);
    return tmp0_safe_receiver == null ? null : tmp0_safe_receiver.q();
  };
  protoOf(AbstractMap).hashCode = function () {
    return hashCode(this.w());
  };
  protoOf(AbstractMap).m = function () {
    return this.h() === 0;
  };
  protoOf(AbstractMap).h = function () {
    return this.w().h();
  };
  protoOf(AbstractMap).u = function () {
    if (this.r2_1 == null) {
      var tmp = this;
      tmp.r2_1 = new AbstractMap$keys$1(this);
    }
    return ensureNotNull(this.r2_1);
  };
  protoOf(AbstractMap).toString = function () {
    var tmp = this.w();
    return joinToString_0(tmp, ', ', '{', '}', VOID, VOID, AbstractMap$toString$lambda(this));
  };
  protoOf(AbstractMap).v = function () {
    if (this.s2_1 == null) {
      var tmp = this;
      tmp.s2_1 = new AbstractMap$values$1(this);
    }
    return ensureNotNull(this.s2_1);
  };
  function Companion_6() {
  }
  protoOf(Companion_6).v2 = function (c) {
    var hashCode_0 = 0;
    var _iterator__ex2g4s = c.e();
    while (_iterator__ex2g4s.f()) {
      var element = _iterator__ex2g4s.g();
      var tmp = hashCode_0;
      var tmp1_elvis_lhs = element == null ? null : hashCode(element);
      hashCode_0 = tmp + (tmp1_elvis_lhs == null ? 0 : tmp1_elvis_lhs) | 0;
    }
    return hashCode_0;
  };
  protoOf(Companion_6).u2 = function (c, other) {
    if (!(c.h() === other.h()))
      return false;
    return c.o(other);
  };
  var Companion_instance_6;
  function Companion_getInstance_6() {
    return Companion_instance_6;
  }
  function AbstractSet() {
    AbstractCollection.call(this);
  }
  protoOf(AbstractSet).equals = function (other) {
    if (other === this)
      return true;
    if (!(!(other == null) ? isInterface(other, KtSet) : false))
      return false;
    return Companion_instance_6.u2(this, other);
  };
  protoOf(AbstractSet).hashCode = function () {
    return Companion_instance_6.v2(this);
  };
  function collectionToArrayCommonImpl(collection) {
    if (collection.m()) {
      // Inline function 'kotlin.emptyArray' call
      return [];
    }
    // Inline function 'kotlin.arrayOfNulls' call
    var size = collection.h();
    var destination = Array(size);
    var iterator = collection.e();
    var index = 0;
    while (iterator.f()) {
      var _unary__edvuaz = index;
      index = _unary__edvuaz + 1 | 0;
      destination[_unary__edvuaz] = iterator.g();
    }
    return destination;
  }
  function emptyList() {
    return EmptyList_getInstance();
  }
  function EmptyList() {
    EmptyList_instance = this;
    this.n6_1 = new Long(-1478467534, -1720727600);
  }
  protoOf(EmptyList).equals = function (other) {
    var tmp;
    if (!(other == null) ? isInterface(other, KtList) : false) {
      tmp = other.m();
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(EmptyList).hashCode = function () {
    return 1;
  };
  protoOf(EmptyList).toString = function () {
    return '[]';
  };
  protoOf(EmptyList).h = function () {
    return 0;
  };
  protoOf(EmptyList).m = function () {
    return true;
  };
  protoOf(EmptyList).i = function (index) {
    throw IndexOutOfBoundsException_init_$Create$_0("Empty list doesn't contain element at index " + index + '.');
  };
  protoOf(EmptyList).e = function () {
    return EmptyIterator_instance;
  };
  var EmptyList_instance;
  function EmptyList_getInstance() {
    if (EmptyList_instance == null)
      new EmptyList();
    return EmptyList_instance;
  }
  function EmptyIterator() {
  }
  protoOf(EmptyIterator).f = function () {
    return false;
  };
  protoOf(EmptyIterator).g = function () {
    throw NoSuchElementException_init_$Create$();
  };
  var EmptyIterator_instance;
  function EmptyIterator_getInstance() {
    return EmptyIterator_instance;
  }
  function get_lastIndex(_this__u8e3s4) {
    return _this__u8e3s4.h() - 1 | 0;
  }
  function collectionSizeOrDefault(_this__u8e3s4, default_0) {
    var tmp;
    if (isInterface(_this__u8e3s4, Collection)) {
      tmp = _this__u8e3s4.h();
    } else {
      tmp = default_0;
    }
    return tmp;
  }
  function toMutableMap(_this__u8e3s4) {
    return LinkedHashMap_init_$Create$_0(_this__u8e3s4);
  }
  function emptyMap() {
    var tmp = EmptyMap_getInstance();
    return isInterface(tmp, KtMap) ? tmp : THROW_CCE();
  }
  function EmptyMap() {
    EmptyMap_instance = this;
    this.o6_1 = new Long(-888910638, 1920087921);
  }
  protoOf(EmptyMap).equals = function (other) {
    var tmp;
    if (!(other == null) ? isInterface(other, KtMap) : false) {
      tmp = other.m();
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(EmptyMap).hashCode = function () {
    return 0;
  };
  protoOf(EmptyMap).toString = function () {
    return '{}';
  };
  protoOf(EmptyMap).h = function () {
    return 0;
  };
  protoOf(EmptyMap).m = function () {
    return true;
  };
  protoOf(EmptyMap).p6 = function (key) {
    return false;
  };
  protoOf(EmptyMap).r = function (key) {
    if (!(key == null ? true : !(key == null)))
      return false;
    return this.p6((key == null ? true : !(key == null)) ? key : THROW_CCE());
  };
  protoOf(EmptyMap).q6 = function (value) {
    return false;
  };
  protoOf(EmptyMap).s = function (value) {
    if (!false)
      return false;
    var tmp;
    if (false) {
      tmp = value;
    } else {
      tmp = THROW_CCE();
    }
    return this.q6(tmp);
  };
  protoOf(EmptyMap).r6 = function (key) {
    return null;
  };
  protoOf(EmptyMap).t = function (key) {
    if (!(key == null ? true : !(key == null)))
      return null;
    return this.r6((key == null ? true : !(key == null)) ? key : THROW_CCE());
  };
  protoOf(EmptyMap).w = function () {
    return EmptySet_getInstance();
  };
  protoOf(EmptyMap).u = function () {
    return EmptySet_getInstance();
  };
  protoOf(EmptyMap).v = function () {
    return EmptyList_getInstance();
  };
  var EmptyMap_instance;
  function EmptyMap_getInstance() {
    if (EmptyMap_instance == null)
      new EmptyMap();
    return EmptyMap_instance;
  }
  function removeFirst(_this__u8e3s4) {
    var tmp;
    if (_this__u8e3s4.m()) {
      throw NoSuchElementException_init_$Create$_0('List is empty.');
    } else {
      tmp = _this__u8e3s4.h2(0);
    }
    return tmp;
  }
  function IntIterator() {
  }
  protoOf(IntIterator).g = function () {
    return this.s6();
  };
  function EmptySet() {
    EmptySet_instance = this;
    this.t6_1 = new Long(1993859828, 793161749);
  }
  protoOf(EmptySet).equals = function (other) {
    var tmp;
    if (!(other == null) ? isInterface(other, KtSet) : false) {
      tmp = other.m();
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(EmptySet).hashCode = function () {
    return 0;
  };
  protoOf(EmptySet).toString = function () {
    return '[]';
  };
  protoOf(EmptySet).h = function () {
    return 0;
  };
  protoOf(EmptySet).m = function () {
    return true;
  };
  protoOf(EmptySet).u6 = function (elements) {
    return elements.m();
  };
  protoOf(EmptySet).o = function (elements) {
    return this.u6(elements);
  };
  protoOf(EmptySet).e = function () {
    return EmptyIterator_instance;
  };
  var EmptySet_instance;
  function EmptySet_getInstance() {
    if (EmptySet_instance == null)
      new EmptySet();
    return EmptySet_instance;
  }
  function getProgressionLastElement(start, end, step) {
    var tmp;
    if (step > 0) {
      tmp = start >= end ? end : end - differenceModulo(end, start, step) | 0;
    } else if (step < 0) {
      tmp = start <= end ? end : end + differenceModulo(start, end, -step | 0) | 0;
    } else {
      throw IllegalArgumentException_init_$Create$_0('Step is zero.');
    }
    return tmp;
  }
  function differenceModulo(a, b, c) {
    return mod(mod(a, c) - mod(b, c) | 0, c);
  }
  function mod(a, b) {
    var mod = a % b | 0;
    return mod >= 0 ? mod : mod + b | 0;
  }
  function get_base64EncodeMap() {
    _init_properties_Base64_kt__ymmsz3();
    return base64EncodeMap;
  }
  var base64EncodeMap;
  function get_base64DecodeMap() {
    _init_properties_Base64_kt__ymmsz3();
    return base64DecodeMap;
  }
  var base64DecodeMap;
  function get_base64UrlEncodeMap() {
    _init_properties_Base64_kt__ymmsz3();
    return base64UrlEncodeMap;
  }
  var base64UrlEncodeMap;
  function get_base64UrlDecodeMap() {
    _init_properties_Base64_kt__ymmsz3();
    return base64UrlDecodeMap;
  }
  var base64UrlDecodeMap;
  var PaddingOption_PRESENT_instance;
  var PaddingOption_ABSENT_instance;
  var PaddingOption_PRESENT_OPTIONAL_instance;
  var PaddingOption_ABSENT_OPTIONAL_instance;
  var PaddingOption_entriesInitialized;
  function PaddingOption_initEntries() {
    if (PaddingOption_entriesInitialized)
      return Unit_instance;
    PaddingOption_entriesInitialized = true;
    PaddingOption_PRESENT_instance = new PaddingOption('PRESENT', 0);
    PaddingOption_ABSENT_instance = new PaddingOption('ABSENT', 1);
    PaddingOption_PRESENT_OPTIONAL_instance = new PaddingOption('PRESENT_OPTIONAL', 2);
    PaddingOption_ABSENT_OPTIONAL_instance = new PaddingOption('ABSENT_OPTIONAL', 3);
  }
  function PaddingOption(name, ordinal) {
    Enum.call(this, name, ordinal);
  }
  function shouldPadOnEncode($this) {
    return $this.y6_1.equals(PaddingOption_PRESENT_getInstance()) || $this.y6_1.equals(PaddingOption_PRESENT_OPTIONAL_getInstance());
  }
  function decodeImpl($this, source, destination, destinationOffset, startIndex, endIndex) {
    var decodeMap = $this.v6_1 ? get_base64UrlDecodeMap() : get_base64DecodeMap();
    var payload = 0;
    var byteStart = -8;
    var sourceIndex = startIndex;
    var destinationIndex = destinationOffset;
    var hasPadding = false;
    $l$loop_1: while (sourceIndex < endIndex) {
      if (byteStart === -8 && (sourceIndex + 3 | 0) < endIndex) {
        var _unary__edvuaz = sourceIndex;
        sourceIndex = _unary__edvuaz + 1 | 0;
        var symbol1 = decodeMap[source[_unary__edvuaz] & 255];
        var _unary__edvuaz_0 = sourceIndex;
        sourceIndex = _unary__edvuaz_0 + 1 | 0;
        var symbol2 = decodeMap[source[_unary__edvuaz_0] & 255];
        var _unary__edvuaz_1 = sourceIndex;
        sourceIndex = _unary__edvuaz_1 + 1 | 0;
        var symbol3 = decodeMap[source[_unary__edvuaz_1] & 255];
        var _unary__edvuaz_2 = sourceIndex;
        sourceIndex = _unary__edvuaz_2 + 1 | 0;
        var symbol4 = decodeMap[source[_unary__edvuaz_2] & 255];
        var bits = symbol1 << 18 | symbol2 << 12 | symbol3 << 6 | symbol4;
        if (bits >= 0) {
          var _unary__edvuaz_3 = destinationIndex;
          destinationIndex = _unary__edvuaz_3 + 1 | 0;
          destination[_unary__edvuaz_3] = toByte(bits >> 16);
          var _unary__edvuaz_4 = destinationIndex;
          destinationIndex = _unary__edvuaz_4 + 1 | 0;
          destination[_unary__edvuaz_4] = toByte(bits >> 8);
          var _unary__edvuaz_5 = destinationIndex;
          destinationIndex = _unary__edvuaz_5 + 1 | 0;
          destination[_unary__edvuaz_5] = toByte(bits);
          continue $l$loop_1;
        }
        sourceIndex = sourceIndex - 4 | 0;
      }
      var symbol = source[sourceIndex] & 255;
      var symbolBits = decodeMap[symbol];
      if (symbolBits < 0) {
        if (symbolBits === -2) {
          hasPadding = true;
          sourceIndex = handlePaddingSymbol($this, source, sourceIndex, endIndex, byteStart);
          break $l$loop_1;
        } else if ($this.w6_1) {
          sourceIndex = sourceIndex + 1 | 0;
          continue $l$loop_1;
        } else {
          throw IllegalArgumentException_init_$Create$_0("Invalid symbol '" + toString(numberToChar(symbol)) + "'(" + toString_2(symbol, 8) + ') at index ' + sourceIndex);
        }
      } else {
        sourceIndex = sourceIndex + 1 | 0;
      }
      payload = payload << 6 | symbolBits;
      byteStart = byteStart + 6 | 0;
      if (byteStart >= 0) {
        var _unary__edvuaz_6 = destinationIndex;
        destinationIndex = _unary__edvuaz_6 + 1 | 0;
        destination[_unary__edvuaz_6] = toByte(payload >>> byteStart | 0);
        payload = payload & ((1 << byteStart) - 1 | 0);
        byteStart = byteStart - 8 | 0;
      }
    }
    if (byteStart === -2) {
      throw IllegalArgumentException_init_$Create$_0('The last unit of input does not have enough bits');
    }
    if (!(byteStart === -8) && !hasPadding && $this.y6_1.equals(PaddingOption_PRESENT_getInstance())) {
      throw IllegalArgumentException_init_$Create$_0('The padding option is set to PRESENT, but the input is not properly padded');
    }
    if (!(payload === 0)) {
      throw IllegalArgumentException_init_$Create$_0('The pad bits must be zeros');
    }
    sourceIndex = skipIllegalSymbolsIfMime($this, source, sourceIndex, endIndex);
    if (sourceIndex < endIndex) {
      var symbol_0 = source[sourceIndex] & 255;
      throw IllegalArgumentException_init_$Create$_0("Symbol '" + toString(numberToChar(symbol_0)) + "'(" + toString_2(symbol_0, 8) + ') at index ' + (sourceIndex - 1 | 0) + ' is prohibited after the pad character');
    }
    return destinationIndex - destinationOffset | 0;
  }
  function handlePaddingSymbol($this, source, padIndex, endIndex, byteStart) {
    var tmp;
    switch (byteStart) {
      case -8:
        throw IllegalArgumentException_init_$Create$_0('Redundant pad character at index ' + padIndex);
      case -2:
        tmp = padIndex + 1 | 0;
        break;
      case -4:
        checkPaddingIsAllowed($this, padIndex);
        var secondPadIndex = skipIllegalSymbolsIfMime($this, source, padIndex + 1 | 0, endIndex);
        if (secondPadIndex === endIndex || !(source[secondPadIndex] === 61)) {
          throw IllegalArgumentException_init_$Create$_0('Missing one pad character at index ' + secondPadIndex);
        }

        tmp = secondPadIndex + 1 | 0;
        break;
      case -6:
        checkPaddingIsAllowed($this, padIndex);
        tmp = padIndex + 1 | 0;
        break;
      default:
        var message = 'Unreachable';
        throw IllegalStateException_init_$Create$_0(toString_1(message));
    }
    return tmp;
  }
  function checkPaddingIsAllowed($this, padIndex) {
    if ($this.y6_1.equals(PaddingOption_ABSENT_getInstance())) {
      throw IllegalArgumentException_init_$Create$_0('The padding option is set to ABSENT, but the input has a pad character at index ' + padIndex);
    }
  }
  function skipIllegalSymbolsIfMime($this, source, startIndex, endIndex) {
    if (!$this.w6_1) {
      return startIndex;
    }
    var sourceIndex = startIndex;
    while (sourceIndex < endIndex) {
      var symbol = source[sourceIndex] & 255;
      if (!(get_base64DecodeMap()[symbol] === -1)) {
        return sourceIndex;
      }
      sourceIndex = sourceIndex + 1 | 0;
    }
    return sourceIndex;
  }
  function checkDestinationBounds($this, destinationSize, destinationOffset, capacityNeeded) {
    if (destinationOffset < 0 || destinationOffset > destinationSize) {
      throw IndexOutOfBoundsException_init_$Create$_0('destination offset: ' + destinationOffset + ', destination size: ' + destinationSize);
    }
    var destinationEndIndex = destinationOffset + capacityNeeded | 0;
    if (destinationEndIndex < 0 || destinationEndIndex > destinationSize) {
      throw IndexOutOfBoundsException_init_$Create$_0('The destination array does not have enough capacity, ' + ('destination offset: ' + destinationOffset + ', destination size: ' + destinationSize + ', capacity needed: ' + capacityNeeded));
    }
  }
  function Default() {
    Default_instance = this;
    Base64.call(this, false, false, -1, PaddingOption_PRESENT_getInstance());
    this.f7_1 = 8;
    this.g7_1 = 6;
    this.h7_1 = 3;
    this.i7_1 = 4;
    this.j7_1 = 61;
    this.k7_1 = 76;
    this.l7_1 = 64;
    var tmp = this;
    // Inline function 'kotlin.code' call
    var this_0 = _Char___init__impl__6a9atx(13);
    var tmp$ret$0 = Char__toInt_impl_vasixd(this_0);
    var tmp_0 = toByte(tmp$ret$0);
    // Inline function 'kotlin.code' call
    var this_1 = _Char___init__impl__6a9atx(10);
    var tmp$ret$1 = Char__toInt_impl_vasixd(this_1);
    // Inline function 'kotlin.byteArrayOf' call
    tmp.m7_1 = new Int8Array([tmp_0, toByte(tmp$ret$1)]);
    this.n7_1 = new Base64(true, false, -1, PaddingOption_PRESENT_getInstance());
    this.o7_1 = new Base64(false, true, 76, PaddingOption_PRESENT_getInstance());
    this.p7_1 = new Base64(false, true, 64, PaddingOption_PRESENT_getInstance());
  }
  var Default_instance;
  function Default_getInstance() {
    if (Default_instance == null)
      new Default();
    return Default_instance;
  }
  function PaddingOption_PRESENT_getInstance() {
    PaddingOption_initEntries();
    return PaddingOption_PRESENT_instance;
  }
  function PaddingOption_ABSENT_getInstance() {
    PaddingOption_initEntries();
    return PaddingOption_ABSENT_instance;
  }
  function PaddingOption_PRESENT_OPTIONAL_getInstance() {
    PaddingOption_initEntries();
    return PaddingOption_PRESENT_OPTIONAL_instance;
  }
  function Base64(isUrlSafe, isMimeScheme, mimeLineLength, paddingOption) {
    Default_getInstance();
    this.v6_1 = isUrlSafe;
    this.w6_1 = isMimeScheme;
    this.x6_1 = mimeLineLength;
    this.y6_1 = paddingOption;
    // Inline function 'kotlin.require' call
    // Inline function 'kotlin.require' call
    if (!(!this.v6_1 || !this.w6_1)) {
      var message = 'Failed requirement.';
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    this.z6_1 = this.x6_1 / 4 | 0;
  }
  protoOf(Base64).q7 = function (source, startIndex, endIndex) {
    // Inline function 'kotlin.io.encoding.platformEncodeToString' call
    var byteResult = this.w7(source, startIndex, endIndex);
    return this.b8(byteResult);
  };
  protoOf(Base64).r7 = function (source, startIndex, endIndex, $super) {
    startIndex = startIndex === VOID ? 0 : startIndex;
    endIndex = endIndex === VOID ? source.length : endIndex;
    return $super === VOID ? this.q7(source, startIndex, endIndex) : $super.q7.call(this, source, startIndex, endIndex);
  };
  protoOf(Base64).s7 = function (source, startIndex, endIndex) {
    this.c8(source.length, startIndex, endIndex);
    var decodeSize = this.z7(source, startIndex, endIndex);
    var destination = new Int8Array(decodeSize);
    var bytesWritten = decodeImpl(this, source, destination, 0, startIndex, endIndex);
    // Inline function 'kotlin.check' call
    if (!(bytesWritten === destination.length)) {
      throw IllegalStateException_init_$Create$_0('Check failed.');
    }
    return destination;
  };
  protoOf(Base64).t7 = function (source, startIndex, endIndex, $super) {
    startIndex = startIndex === VOID ? 0 : startIndex;
    endIndex = endIndex === VOID ? source.length : endIndex;
    return $super === VOID ? this.s7(source, startIndex, endIndex) : $super.s7.call(this, source, startIndex, endIndex);
  };
  protoOf(Base64).u7 = function (source, startIndex, endIndex) {
    // Inline function 'kotlin.io.encoding.platformCharsToBytes' call
    var byteSource = this.a8(source, startIndex, endIndex);
    return this.t7(byteSource);
  };
  protoOf(Base64).v7 = function (source, startIndex, endIndex, $super) {
    startIndex = startIndex === VOID ? 0 : startIndex;
    endIndex = endIndex === VOID ? charSequenceLength(source) : endIndex;
    return $super === VOID ? this.u7(source, startIndex, endIndex) : $super.u7.call(this, source, startIndex, endIndex);
  };
  protoOf(Base64).w7 = function (source, startIndex, endIndex) {
    this.c8(source.length, startIndex, endIndex);
    var encodeSize = this.y7(endIndex - startIndex | 0);
    var destination = new Int8Array(encodeSize);
    this.x7(source, destination, 0, startIndex, endIndex);
    return destination;
  };
  protoOf(Base64).x7 = function (source, destination, destinationOffset, startIndex, endIndex) {
    this.c8(source.length, startIndex, endIndex);
    checkDestinationBounds(this, destination.length, destinationOffset, this.y7(endIndex - startIndex | 0));
    var encodeMap = this.v6_1 ? get_base64UrlEncodeMap() : get_base64EncodeMap();
    var sourceIndex = startIndex;
    var destinationIndex = destinationOffset;
    var groupsPerLine = this.w6_1 ? this.z6_1 : 2147483647;
    while ((sourceIndex + 2 | 0) < endIndex) {
      // Inline function 'kotlin.comparisons.minOf' call
      var a = (endIndex - sourceIndex | 0) / 3 | 0;
      var groups = Math.min(a, groupsPerLine);
      var inductionVariable = 0;
      if (inductionVariable < groups)
        do {
          var i = inductionVariable;
          inductionVariable = inductionVariable + 1 | 0;
          var _unary__edvuaz = sourceIndex;
          sourceIndex = _unary__edvuaz + 1 | 0;
          var byte1 = source[_unary__edvuaz] & 255;
          var _unary__edvuaz_0 = sourceIndex;
          sourceIndex = _unary__edvuaz_0 + 1 | 0;
          var byte2 = source[_unary__edvuaz_0] & 255;
          var _unary__edvuaz_1 = sourceIndex;
          sourceIndex = _unary__edvuaz_1 + 1 | 0;
          var byte3 = source[_unary__edvuaz_1] & 255;
          var bits = byte1 << 16 | byte2 << 8 | byte3;
          var _unary__edvuaz_2 = destinationIndex;
          destinationIndex = _unary__edvuaz_2 + 1 | 0;
          destination[_unary__edvuaz_2] = encodeMap[bits >>> 18 | 0];
          var _unary__edvuaz_3 = destinationIndex;
          destinationIndex = _unary__edvuaz_3 + 1 | 0;
          destination[_unary__edvuaz_3] = encodeMap[(bits >>> 12 | 0) & 63];
          var _unary__edvuaz_4 = destinationIndex;
          destinationIndex = _unary__edvuaz_4 + 1 | 0;
          destination[_unary__edvuaz_4] = encodeMap[(bits >>> 6 | 0) & 63];
          var _unary__edvuaz_5 = destinationIndex;
          destinationIndex = _unary__edvuaz_5 + 1 | 0;
          destination[_unary__edvuaz_5] = encodeMap[bits & 63];
        }
         while (inductionVariable < groups);
      if (groups === groupsPerLine && !(sourceIndex === endIndex)) {
        var _unary__edvuaz_6 = destinationIndex;
        destinationIndex = _unary__edvuaz_6 + 1 | 0;
        destination[_unary__edvuaz_6] = Default_getInstance().m7_1[0];
        var _unary__edvuaz_7 = destinationIndex;
        destinationIndex = _unary__edvuaz_7 + 1 | 0;
        destination[_unary__edvuaz_7] = Default_getInstance().m7_1[1];
      }
    }
    var tmp0_subject = endIndex - sourceIndex | 0;
    if (tmp0_subject === 1) {
      var _unary__edvuaz_8 = sourceIndex;
      sourceIndex = _unary__edvuaz_8 + 1 | 0;
      var byte1_0 = source[_unary__edvuaz_8] & 255;
      var bits_0 = byte1_0 << 4;
      var _unary__edvuaz_9 = destinationIndex;
      destinationIndex = _unary__edvuaz_9 + 1 | 0;
      destination[_unary__edvuaz_9] = encodeMap[bits_0 >>> 6 | 0];
      var _unary__edvuaz_10 = destinationIndex;
      destinationIndex = _unary__edvuaz_10 + 1 | 0;
      destination[_unary__edvuaz_10] = encodeMap[bits_0 & 63];
      if (shouldPadOnEncode(this)) {
        var _unary__edvuaz_11 = destinationIndex;
        destinationIndex = _unary__edvuaz_11 + 1 | 0;
        destination[_unary__edvuaz_11] = 61;
        var _unary__edvuaz_12 = destinationIndex;
        destinationIndex = _unary__edvuaz_12 + 1 | 0;
        destination[_unary__edvuaz_12] = 61;
      }
    } else if (tmp0_subject === 2) {
      var _unary__edvuaz_13 = sourceIndex;
      sourceIndex = _unary__edvuaz_13 + 1 | 0;
      var byte1_1 = source[_unary__edvuaz_13] & 255;
      var _unary__edvuaz_14 = sourceIndex;
      sourceIndex = _unary__edvuaz_14 + 1 | 0;
      var byte2_0 = source[_unary__edvuaz_14] & 255;
      var bits_1 = byte1_1 << 10 | byte2_0 << 2;
      var _unary__edvuaz_15 = destinationIndex;
      destinationIndex = _unary__edvuaz_15 + 1 | 0;
      destination[_unary__edvuaz_15] = encodeMap[bits_1 >>> 12 | 0];
      var _unary__edvuaz_16 = destinationIndex;
      destinationIndex = _unary__edvuaz_16 + 1 | 0;
      destination[_unary__edvuaz_16] = encodeMap[(bits_1 >>> 6 | 0) & 63];
      var _unary__edvuaz_17 = destinationIndex;
      destinationIndex = _unary__edvuaz_17 + 1 | 0;
      destination[_unary__edvuaz_17] = encodeMap[bits_1 & 63];
      if (shouldPadOnEncode(this)) {
        var _unary__edvuaz_18 = destinationIndex;
        destinationIndex = _unary__edvuaz_18 + 1 | 0;
        destination[_unary__edvuaz_18] = 61;
      }
    }
    // Inline function 'kotlin.check' call
    if (!(sourceIndex === endIndex)) {
      throw IllegalStateException_init_$Create$_0('Check failed.');
    }
    return destinationIndex - destinationOffset | 0;
  };
  protoOf(Base64).y7 = function (sourceSize) {
    var groups = sourceSize / 3 | 0;
    var trailingBytes = sourceSize % 3 | 0;
    var size = imul(groups, 4);
    if (!(trailingBytes === 0)) {
      size = size + (shouldPadOnEncode(this) ? 4 : trailingBytes + 1 | 0) | 0;
    }
    if (size < 0) {
      throw IllegalArgumentException_init_$Create$_0('Input is too big');
    }
    if (this.w6_1) {
      size = size + imul((size - 1 | 0) / this.x6_1 | 0, 2) | 0;
    }
    if (size < 0) {
      throw IllegalArgumentException_init_$Create$_0('Input is too big');
    }
    return size;
  };
  protoOf(Base64).z7 = function (source, startIndex, endIndex) {
    var symbols = endIndex - startIndex | 0;
    if (symbols === 0) {
      return 0;
    }
    if (symbols === 1) {
      throw IllegalArgumentException_init_$Create$_0('Input should have at least 2 symbols for Base64 decoding, startIndex: ' + startIndex + ', endIndex: ' + endIndex);
    }
    if (this.w6_1) {
      var inductionVariable = startIndex;
      if (inductionVariable < endIndex)
        $l$loop: do {
          var index = inductionVariable;
          inductionVariable = inductionVariable + 1 | 0;
          var symbol = source[index] & 255;
          var symbolBits = get_base64DecodeMap()[symbol];
          if (symbolBits < 0) {
            if (symbolBits === -2) {
              symbols = symbols - (endIndex - index | 0) | 0;
              break $l$loop;
            }
            symbols = symbols - 1 | 0;
          }
        }
         while (inductionVariable < endIndex);
    } else if (source[endIndex - 1 | 0] === 61) {
      symbols = symbols - 1 | 0;
      if (source[endIndex - 2 | 0] === 61) {
        symbols = symbols - 1 | 0;
      }
    }
    // Inline function 'kotlin.Long.times' call
    // Inline function 'kotlin.Long.div' call
    return toLong(symbols).j1(toLong(6)).k1(toLong(8)).n1();
  };
  protoOf(Base64).a8 = function (source, startIndex, endIndex) {
    this.c8(charSequenceLength(source), startIndex, endIndex);
    var byteArray = new Int8Array(endIndex - startIndex | 0);
    var length = 0;
    var inductionVariable = startIndex;
    if (inductionVariable < endIndex)
      do {
        var index = inductionVariable;
        inductionVariable = inductionVariable + 1 | 0;
        // Inline function 'kotlin.code' call
        var this_0 = charSequenceGet(source, index);
        var symbol = Char__toInt_impl_vasixd(this_0);
        if (symbol <= 255) {
          var _unary__edvuaz = length;
          length = _unary__edvuaz + 1 | 0;
          byteArray[_unary__edvuaz] = toByte(symbol);
        } else {
          var _unary__edvuaz_0 = length;
          length = _unary__edvuaz_0 + 1 | 0;
          byteArray[_unary__edvuaz_0] = 63;
        }
      }
       while (inductionVariable < endIndex);
    return byteArray;
  };
  protoOf(Base64).b8 = function (source) {
    var stringBuilder = StringBuilder_init_$Create$(source.length);
    var inductionVariable = 0;
    var last = source.length;
    while (inductionVariable < last) {
      var byte = source[inductionVariable];
      inductionVariable = inductionVariable + 1 | 0;
      stringBuilder.t5(numberToChar(byte));
    }
    return stringBuilder.toString();
  };
  protoOf(Base64).c8 = function (sourceSize, startIndex, endIndex) {
    Companion_instance_4.i6(startIndex, endIndex, sourceSize);
  };
  var properties_initialized_Base64_kt_5g824v;
  function _init_properties_Base64_kt__ymmsz3() {
    if (!properties_initialized_Base64_kt_5g824v) {
      properties_initialized_Base64_kt_5g824v = true;
      // Inline function 'kotlin.byteArrayOf' call
      base64EncodeMap = new Int8Array([65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 43, 47]);
      // Inline function 'kotlin.apply' call
      var this_0 = new Int32Array(256);
      fill(this_0, -1);
      this_0[61] = -2;
      // Inline function 'kotlin.collections.forEachIndexed' call
      var index = 0;
      var indexedObject = get_base64EncodeMap();
      var inductionVariable = 0;
      var last = indexedObject.length;
      while (inductionVariable < last) {
        var item = indexedObject[inductionVariable];
        inductionVariable = inductionVariable + 1 | 0;
        var _unary__edvuaz = index;
        index = _unary__edvuaz + 1 | 0;
        this_0[item] = _unary__edvuaz;
      }
      base64DecodeMap = this_0;
      // Inline function 'kotlin.byteArrayOf' call
      base64UrlEncodeMap = new Int8Array([65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 45, 95]);
      // Inline function 'kotlin.apply' call
      var this_1 = new Int32Array(256);
      fill(this_1, -1);
      this_1[61] = -2;
      // Inline function 'kotlin.collections.forEachIndexed' call
      var index_0 = 0;
      var indexedObject_0 = get_base64UrlEncodeMap();
      var inductionVariable_0 = 0;
      var last_0 = indexedObject_0.length;
      while (inductionVariable_0 < last_0) {
        var item_0 = indexedObject_0[inductionVariable_0];
        inductionVariable_0 = inductionVariable_0 + 1 | 0;
        var _unary__edvuaz_0 = index_0;
        index_0 = _unary__edvuaz_0 + 1 | 0;
        this_1[item_0] = _unary__edvuaz_0;
      }
      base64UrlDecodeMap = this_1;
    }
  }
  function Companion_7() {
    Companion_instance_7 = this;
    this.j_1 = new IntRange(1, 0);
  }
  var Companion_instance_7;
  function Companion_getInstance_7() {
    if (Companion_instance_7 == null)
      new Companion_7();
    return Companion_instance_7;
  }
  function IntRange(start, endInclusive) {
    Companion_getInstance_7();
    IntProgression.call(this, start, endInclusive, 1);
  }
  protoOf(IntRange).g8 = function () {
    return this.h8_1;
  };
  protoOf(IntRange).k8 = function () {
    return this.i8_1;
  };
  protoOf(IntRange).m = function () {
    return this.h8_1 > this.i8_1;
  };
  protoOf(IntRange).equals = function (other) {
    var tmp;
    if (other instanceof IntRange) {
      tmp = this.m() && other.m() || (this.h8_1 === other.h8_1 && this.i8_1 === other.i8_1);
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(IntRange).hashCode = function () {
    return this.m() ? -1 : imul(31, this.h8_1) + this.i8_1 | 0;
  };
  protoOf(IntRange).toString = function () {
    return '' + this.h8_1 + '..' + this.i8_1;
  };
  function IntProgressionIterator(first, last, step) {
    IntIterator.call(this);
    this.l8_1 = step;
    this.m8_1 = last;
    this.n8_1 = this.l8_1 > 0 ? first <= last : first >= last;
    this.o8_1 = this.n8_1 ? first : this.m8_1;
  }
  protoOf(IntProgressionIterator).f = function () {
    return this.n8_1;
  };
  protoOf(IntProgressionIterator).s6 = function () {
    var value = this.o8_1;
    if (value === this.m8_1) {
      if (!this.n8_1)
        throw NoSuchElementException_init_$Create$();
      this.n8_1 = false;
    } else {
      this.o8_1 = this.o8_1 + this.l8_1 | 0;
    }
    return value;
  };
  function Companion_8() {
  }
  protoOf(Companion_8).k = function (rangeStart, rangeEnd, step) {
    return new IntProgression(rangeStart, rangeEnd, step);
  };
  var Companion_instance_8;
  function Companion_getInstance_8() {
    return Companion_instance_8;
  }
  function IntProgression(start, endInclusive, step) {
    if (step === 0)
      throw IllegalArgumentException_init_$Create$_0('Step must be non-zero.');
    if (step === -2147483648)
      throw IllegalArgumentException_init_$Create$_0('Step must be greater than Int.MIN_VALUE to avoid overflow on negation.');
    this.h8_1 = start;
    this.i8_1 = getProgressionLastElement(start, endInclusive, step);
    this.j8_1 = step;
  }
  protoOf(IntProgression).e = function () {
    return new IntProgressionIterator(this.h8_1, this.i8_1, this.j8_1);
  };
  protoOf(IntProgression).m = function () {
    return this.j8_1 > 0 ? this.h8_1 > this.i8_1 : this.h8_1 < this.i8_1;
  };
  protoOf(IntProgression).equals = function (other) {
    var tmp;
    if (other instanceof IntProgression) {
      tmp = this.m() && other.m() || (this.h8_1 === other.h8_1 && this.i8_1 === other.i8_1 && this.j8_1 === other.j8_1);
    } else {
      tmp = false;
    }
    return tmp;
  };
  protoOf(IntProgression).hashCode = function () {
    return this.m() ? -1 : imul(31, imul(31, this.h8_1) + this.i8_1 | 0) + this.j8_1 | 0;
  };
  protoOf(IntProgression).toString = function () {
    return this.j8_1 > 0 ? '' + this.h8_1 + '..' + this.i8_1 + ' step ' + this.j8_1 : '' + this.h8_1 + ' downTo ' + this.i8_1 + ' step ' + (-this.j8_1 | 0);
  };
  function appendElement(_this__u8e3s4, element, transform) {
    if (!(transform == null))
      _this__u8e3s4.d(transform(element));
    else {
      if (element == null ? true : isCharSequence(element))
        _this__u8e3s4.d(element);
      else {
        if (element instanceof Char)
          _this__u8e3s4.t5(element.p8_1);
        else {
          _this__u8e3s4.d(toString_1(element));
        }
      }
    }
  }
  function equals_0(_this__u8e3s4, other, ignoreCase) {
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    if (_this__u8e3s4 === other)
      return true;
    if (!ignoreCase)
      return false;
    var thisUpper = uppercaseChar(_this__u8e3s4);
    var otherUpper = uppercaseChar(other);
    var tmp;
    if (thisUpper === otherUpper) {
      tmp = true;
    } else {
      // Inline function 'kotlin.text.lowercaseChar' call
      // Inline function 'kotlin.text.lowercase' call
      // Inline function 'kotlin.js.asDynamic' call
      // Inline function 'kotlin.js.unsafeCast' call
      var tmp$ret$2 = toString(thisUpper).toLowerCase();
      var tmp_0 = charCodeAt(tmp$ret$2, 0);
      // Inline function 'kotlin.text.lowercaseChar' call
      // Inline function 'kotlin.text.lowercase' call
      // Inline function 'kotlin.js.asDynamic' call
      // Inline function 'kotlin.js.unsafeCast' call
      var tmp$ret$6 = toString(otherUpper).toLowerCase();
      tmp = tmp_0 === charCodeAt(tmp$ret$6, 0);
    }
    return tmp;
  }
  function split(_this__u8e3s4, delimiters, ignoreCase, limit) {
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    limit = limit === VOID ? 0 : limit;
    if (delimiters.length === 1) {
      var delimiter = delimiters[0];
      // Inline function 'kotlin.text.isEmpty' call
      if (!(charSequenceLength(delimiter) === 0)) {
        return split_0(_this__u8e3s4, delimiter, ignoreCase, limit);
      }
    }
    // Inline function 'kotlin.collections.map' call
    var this_0 = asIterable(rangesDelimitedBy(_this__u8e3s4, delimiters, VOID, ignoreCase, limit));
    // Inline function 'kotlin.collections.mapTo' call
    var destination = ArrayList_init_$Create$_0(collectionSizeOrDefault(this_0, 10));
    var _iterator__ex2g4s = this_0.e();
    while (_iterator__ex2g4s.f()) {
      var item = _iterator__ex2g4s.g();
      var tmp$ret$1 = substring_1(_this__u8e3s4, item);
      destination.a2(tmp$ret$1);
    }
    return destination;
  }
  function split_0(_this__u8e3s4, delimiter, ignoreCase, limit) {
    requireNonNegativeLimit(limit);
    var currentOffset = 0;
    var nextIndex = indexOf_0(_this__u8e3s4, delimiter, currentOffset, ignoreCase);
    if (nextIndex === -1 || limit === 1) {
      return listOf(toString_1(_this__u8e3s4));
    }
    var isLimited = limit > 0;
    var result = ArrayList_init_$Create$_0(isLimited ? coerceAtMost(limit, 10) : 10);
    $l$loop: do {
      var tmp2 = currentOffset;
      // Inline function 'kotlin.text.substring' call
      var endIndex = nextIndex;
      var tmp$ret$0 = toString_1(charSequenceSubSequence(_this__u8e3s4, tmp2, endIndex));
      result.a2(tmp$ret$0);
      currentOffset = nextIndex + delimiter.length | 0;
      if (isLimited && result.h() === (limit - 1 | 0))
        break $l$loop;
      nextIndex = indexOf_0(_this__u8e3s4, delimiter, currentOffset, ignoreCase);
    }
     while (!(nextIndex === -1));
    var tmp2_0 = currentOffset;
    // Inline function 'kotlin.text.substring' call
    var endIndex_0 = charSequenceLength(_this__u8e3s4);
    var tmp$ret$1 = toString_1(charSequenceSubSequence(_this__u8e3s4, tmp2_0, endIndex_0));
    result.a2(tmp$ret$1);
    return result;
  }
  function rangesDelimitedBy(_this__u8e3s4, delimiters, startIndex, ignoreCase, limit) {
    startIndex = startIndex === VOID ? 0 : startIndex;
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    limit = limit === VOID ? 0 : limit;
    requireNonNegativeLimit(limit);
    var delimitersList = asList(delimiters);
    return new DelimitedRangesSequence(_this__u8e3s4, startIndex, limit, rangesDelimitedBy$lambda(delimitersList, ignoreCase));
  }
  function substring_1(_this__u8e3s4, range) {
    return toString_1(charSequenceSubSequence(_this__u8e3s4, range.g8(), range.k8() + 1 | 0));
  }
  function requireNonNegativeLimit(limit) {
    // Inline function 'kotlin.require' call
    if (!(limit >= 0)) {
      var message = 'Limit must be non-negative, but was ' + limit;
      throw IllegalArgumentException_init_$Create$_0(toString_1(message));
    }
    return Unit_instance;
  }
  function indexOf_0(_this__u8e3s4, string, startIndex, ignoreCase) {
    startIndex = startIndex === VOID ? 0 : startIndex;
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    var tmp;
    var tmp_0;
    if (ignoreCase) {
      tmp_0 = true;
    } else {
      tmp_0 = !(typeof _this__u8e3s4 === 'string');
    }
    if (tmp_0) {
      tmp = indexOf_1(_this__u8e3s4, string, startIndex, charSequenceLength(_this__u8e3s4), ignoreCase);
    } else {
      // Inline function 'kotlin.text.nativeIndexOf' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp = _this__u8e3s4.indexOf(string, startIndex);
    }
    return tmp;
  }
  function calcNext($this) {
    if ($this.s8_1 < 0) {
      $this.q8_1 = 0;
      $this.t8_1 = null;
    } else {
      var tmp;
      var tmp_0;
      if ($this.v8_1.y8_1 > 0) {
        $this.u8_1 = $this.u8_1 + 1 | 0;
        tmp_0 = $this.u8_1 >= $this.v8_1.y8_1;
      } else {
        tmp_0 = false;
      }
      if (tmp_0) {
        tmp = true;
      } else {
        tmp = $this.s8_1 > charSequenceLength($this.v8_1.w8_1);
      }
      if (tmp) {
        $this.t8_1 = numberRangeToNumber($this.r8_1, get_lastIndex_0($this.v8_1.w8_1));
        $this.s8_1 = -1;
      } else {
        var match = $this.v8_1.z8_1($this.v8_1.w8_1, $this.s8_1);
        if (match == null) {
          $this.t8_1 = numberRangeToNumber($this.r8_1, get_lastIndex_0($this.v8_1.w8_1));
          $this.s8_1 = -1;
        } else {
          var index = match.c9();
          var length = match.d9();
          $this.t8_1 = until($this.r8_1, index);
          $this.r8_1 = index + length | 0;
          $this.s8_1 = $this.r8_1 + (length === 0 ? 1 : 0) | 0;
        }
      }
      $this.q8_1 = 1;
    }
  }
  function DelimitedRangesSequence$iterator$1(this$0) {
    this.v8_1 = this$0;
    this.q8_1 = -1;
    this.r8_1 = coerceIn(this$0.x8_1, 0, charSequenceLength(this$0.w8_1));
    this.s8_1 = this.r8_1;
    this.t8_1 = null;
    this.u8_1 = 0;
  }
  protoOf(DelimitedRangesSequence$iterator$1).g = function () {
    if (this.q8_1 === -1) {
      calcNext(this);
    }
    if (this.q8_1 === 0)
      throw NoSuchElementException_init_$Create$();
    var tmp = this.t8_1;
    var result = tmp instanceof IntRange ? tmp : THROW_CCE();
    this.t8_1 = null;
    this.q8_1 = -1;
    return result;
  };
  protoOf(DelimitedRangesSequence$iterator$1).f = function () {
    if (this.q8_1 === -1) {
      calcNext(this);
    }
    return this.q8_1 === 1;
  };
  function DelimitedRangesSequence(input, startIndex, limit, getNextMatch) {
    this.w8_1 = input;
    this.x8_1 = startIndex;
    this.y8_1 = limit;
    this.z8_1 = getNextMatch;
  }
  protoOf(DelimitedRangesSequence).e = function () {
    return new DelimitedRangesSequence$iterator$1(this);
  };
  function findAnyOf(_this__u8e3s4, strings, startIndex, ignoreCase, last) {
    if (!ignoreCase && strings.h() === 1) {
      var string = single(strings);
      var index = !last ? indexOf_0(_this__u8e3s4, string, startIndex) : lastIndexOf(_this__u8e3s4, string, startIndex);
      return index < 0 ? null : to(index, string);
    }
    var indices = !last ? numberRangeToNumber(coerceAtLeast(startIndex, 0), charSequenceLength(_this__u8e3s4)) : downTo(coerceAtMost(startIndex, get_lastIndex_0(_this__u8e3s4)), 0);
    if (typeof _this__u8e3s4 === 'string') {
      var inductionVariable = indices.h8_1;
      var last_0 = indices.i8_1;
      var step = indices.j8_1;
      if (step > 0 && inductionVariable <= last_0 || (step < 0 && last_0 <= inductionVariable))
        do {
          var index_0 = inductionVariable;
          inductionVariable = inductionVariable + step | 0;
          var tmp$ret$1;
          $l$block: {
            // Inline function 'kotlin.collections.firstOrNull' call
            var _iterator__ex2g4s = strings.e();
            while (_iterator__ex2g4s.f()) {
              var element = _iterator__ex2g4s.g();
              if (regionMatches(element, 0, _this__u8e3s4, index_0, element.length, ignoreCase)) {
                tmp$ret$1 = element;
                break $l$block;
              }
            }
            tmp$ret$1 = null;
          }
          var matchingString = tmp$ret$1;
          if (!(matchingString == null))
            return to(index_0, matchingString);
        }
         while (!(index_0 === last_0));
    } else {
      var inductionVariable_0 = indices.h8_1;
      var last_1 = indices.i8_1;
      var step_0 = indices.j8_1;
      if (step_0 > 0 && inductionVariable_0 <= last_1 || (step_0 < 0 && last_1 <= inductionVariable_0))
        do {
          var index_1 = inductionVariable_0;
          inductionVariable_0 = inductionVariable_0 + step_0 | 0;
          var tmp$ret$3;
          $l$block_0: {
            // Inline function 'kotlin.collections.firstOrNull' call
            var _iterator__ex2g4s_0 = strings.e();
            while (_iterator__ex2g4s_0.f()) {
              var element_0 = _iterator__ex2g4s_0.g();
              if (regionMatchesImpl(element_0, 0, _this__u8e3s4, index_1, element_0.length, ignoreCase)) {
                tmp$ret$3 = element_0;
                break $l$block_0;
              }
            }
            tmp$ret$3 = null;
          }
          var matchingString_0 = tmp$ret$3;
          if (!(matchingString_0 == null))
            return to(index_1, matchingString_0);
        }
         while (!(index_1 === last_1));
    }
    return null;
  }
  function indexOf_1(_this__u8e3s4, other, startIndex, endIndex, ignoreCase, last) {
    last = last === VOID ? false : last;
    var indices = !last ? numberRangeToNumber(coerceAtLeast(startIndex, 0), coerceAtMost(endIndex, charSequenceLength(_this__u8e3s4))) : downTo(coerceAtMost(startIndex, get_lastIndex_0(_this__u8e3s4)), coerceAtLeast(endIndex, 0));
    var tmp;
    if (typeof _this__u8e3s4 === 'string') {
      tmp = typeof other === 'string';
    } else {
      tmp = false;
    }
    if (tmp) {
      var inductionVariable = indices.h8_1;
      var last_0 = indices.i8_1;
      var step = indices.j8_1;
      if (step > 0 && inductionVariable <= last_0 || (step < 0 && last_0 <= inductionVariable))
        do {
          var index = inductionVariable;
          inductionVariable = inductionVariable + step | 0;
          if (regionMatches(other, 0, _this__u8e3s4, index, other.length, ignoreCase))
            return index;
        }
         while (!(index === last_0));
    } else {
      var inductionVariable_0 = indices.h8_1;
      var last_1 = indices.i8_1;
      var step_0 = indices.j8_1;
      if (step_0 > 0 && inductionVariable_0 <= last_1 || (step_0 < 0 && last_1 <= inductionVariable_0))
        do {
          var index_0 = inductionVariable_0;
          inductionVariable_0 = inductionVariable_0 + step_0 | 0;
          if (regionMatchesImpl(other, 0, _this__u8e3s4, index_0, charSequenceLength(other), ignoreCase))
            return index_0;
        }
         while (!(index_0 === last_1));
    }
    return -1;
  }
  function get_lastIndex_0(_this__u8e3s4) {
    return charSequenceLength(_this__u8e3s4) - 1 | 0;
  }
  function lastIndexOf(_this__u8e3s4, string, startIndex, ignoreCase) {
    startIndex = startIndex === VOID ? get_lastIndex_0(_this__u8e3s4) : startIndex;
    ignoreCase = ignoreCase === VOID ? false : ignoreCase;
    var tmp;
    var tmp_0;
    if (ignoreCase) {
      tmp_0 = true;
    } else {
      tmp_0 = !(typeof _this__u8e3s4 === 'string');
    }
    if (tmp_0) {
      tmp = indexOf_1(_this__u8e3s4, string, startIndex, 0, ignoreCase, true);
    } else {
      // Inline function 'kotlin.text.nativeLastIndexOf' call
      // Inline function 'kotlin.js.asDynamic' call
      tmp = _this__u8e3s4.lastIndexOf(string, startIndex);
    }
    return tmp;
  }
  function regionMatchesImpl(_this__u8e3s4, thisOffset, other, otherOffset, length, ignoreCase) {
    if (otherOffset < 0 || thisOffset < 0 || thisOffset > (charSequenceLength(_this__u8e3s4) - length | 0) || otherOffset > (charSequenceLength(other) - length | 0)) {
      return false;
    }
    var inductionVariable = 0;
    if (inductionVariable < length)
      do {
        var index = inductionVariable;
        inductionVariable = inductionVariable + 1 | 0;
        if (!equals_0(charSequenceGet(_this__u8e3s4, thisOffset + index | 0), charSequenceGet(other, otherOffset + index | 0), ignoreCase))
          return false;
      }
       while (inductionVariable < length);
    return true;
  }
  function rangesDelimitedBy$lambda($delimitersList, $ignoreCase) {
    return function ($this$DelimitedRangesSequence, currentIndex) {
      var tmp0_safe_receiver = findAnyOf($this$DelimitedRangesSequence, $delimitersList, currentIndex, $ignoreCase, false);
      var tmp;
      if (tmp0_safe_receiver == null) {
        tmp = null;
      } else {
        // Inline function 'kotlin.let' call
        tmp = to(tmp0_safe_receiver.a9_1, tmp0_safe_receiver.b9_1.length);
      }
      return tmp;
    };
  }
  function Pair(first, second) {
    this.a9_1 = first;
    this.b9_1 = second;
  }
  protoOf(Pair).toString = function () {
    return '(' + toString_0(this.a9_1) + ', ' + toString_0(this.b9_1) + ')';
  };
  protoOf(Pair).c9 = function () {
    return this.a9_1;
  };
  protoOf(Pair).d9 = function () {
    return this.b9_1;
  };
  protoOf(Pair).hashCode = function () {
    var result = this.a9_1 == null ? 0 : hashCode(this.a9_1);
    result = imul(result, 31) + (this.b9_1 == null ? 0 : hashCode(this.b9_1)) | 0;
    return result;
  };
  protoOf(Pair).equals = function (other) {
    if (this === other)
      return true;
    if (!(other instanceof Pair))
      return false;
    var tmp0_other_with_cast = other instanceof Pair ? other : THROW_CCE();
    if (!equals(this.a9_1, tmp0_other_with_cast.a9_1))
      return false;
    if (!equals(this.b9_1, tmp0_other_with_cast.b9_1))
      return false;
    return true;
  };
  function to(_this__u8e3s4, that) {
    return new Pair(_this__u8e3s4, that);
  }
  function _UShort___init__impl__jigrne(data) {
    return data;
  }
  function _UShort___get_data__impl__g0245($this) {
    return $this;
  }
  //region block: post-declaration
  protoOf(AbstractMap).asJsReadonlyMapView = asJsReadonlyMapView;
  protoOf(InternalHashMap).w3 = containsAllEntries;
  protoOf(EmptyMap).asJsReadonlyMapView = asJsReadonlyMapView;
  //endregion
  //region block: init
  Companion_instance = new Companion();
  Companion_instance_0 = new Companion_0();
  Unit_instance = new Unit();
  Companion_instance_3 = new Companion_3();
  Companion_instance_4 = new Companion_4();
  Companion_instance_5 = new Companion_5();
  Companion_instance_6 = new Companion_6();
  EmptyIterator_instance = new EmptyIterator();
  Companion_instance_8 = new Companion_8();
  //endregion
  //region block: exports
  function $jsExportAll$(_) {
    var $kotlin = _.kotlin || (_.kotlin = {});
    var $kotlin$collections = $kotlin.collections || ($kotlin.collections = {});
    defineProp($kotlin$collections, 'KtMap', Companion_getInstance);
  }
  $jsExportAll$(_);
  _.$jsExportAll$ = $jsExportAll$;
  _.$_$ = _.$_$ || {};
  _.$_$.a = VOID;
  _.$_$.b = ArrayList_init_$Create$;
  _.$_$.c = LinkedHashMap_init_$Create$;
  _.$_$.d = Default_getInstance;
  _.$_$.e = Unit_instance;
  _.$_$.f = KtMap;
  _.$_$.g = copyToArray;
  _.$_$.h = emptyList;
  _.$_$.i = emptyMap;
  _.$_$.j = joinToString;
  _.$_$.k = removeFirst;
  _.$_$.l = toMutableMap;
  _.$_$.m = println;
  _.$_$.n = charSequenceLength;
  _.$_$.o = defineProp;
  _.$_$.p = equals;
  _.$_$.q = getBooleanHashCode;
  _.$_$.r = getStringHashCode;
  _.$_$.s = hashCode;
  _.$_$.t = initMetadataForClass;
  _.$_$.u = initMetadataForInterface;
  _.$_$.v = isArray;
  _.$_$.w = isInterface;
  _.$_$.x = protoOf;
  _.$_$.y = toString_1;
  _.$_$.z = decodeToString;
  _.$_$.a1 = encodeToByteArray;
  _.$_$.b1 = endsWith;
  _.$_$.c1 = split;
  _.$_$.d1 = startsWith;
  _.$_$.e1 = substring_0;
  _.$_$.f1 = substring;
  _.$_$.g1 = take;
  _.$_$.h1 = THROW_CCE;
  _.$_$.i1 = ensureNotNull;
  //endregion
  return _;
}));
